<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Polygenic Architectures are Difficult to Resolve Using GWAS</title>

<script src="site_libs/header-attrs-2.5/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">NemaScan: GWAS in C. elegans Wild Isolates</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sam-widmayer/NemaScan_Performance">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Polygenic Architectures are Difficult to Resolve Using GWAS</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-12-01
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>NemaScan_Performance/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200825code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200825)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200825code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200825)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
~/Documents/projects/NemaScan_Performance/
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsamwidmayerNemaScanPerformancetree68829fe7ff9d1f8846df2edf35882f6091f73db2targetblank68829fea"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sam-widmayer/NemaScan_Performance/tree/68829fe7ff9d1f8846df2edf35882f6091f73db2" target="_blank">68829fe</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsamwidmayerNemaScanPerformancetree68829fe7ff9d1f8846df2edf35882f6091f73db2targetblank68829fea" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sam-widmayer/NemaScan_Performance/tree/68829fe7ff9d1f8846df2edf35882f6091f73db2" target="_blank">68829fe</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/.Rhistory
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  code/2014_toxinphenotypes.R
    Untracked:  code/Find_GCTA_Intervals.R
    Untracked:  code/startup.R
    Untracked:  code/susie.R
    Untracked:  data/NemaScan_Performance.Algorithm.Sims.20201024.RData
    Untracked:  data/NemaScan_Performance.Algorithm.Sims.20201110_EMMA.RData
    Untracked:  data/NemaScan_Performance.Algorithm.Sims.v.2.20201104.RData
    Untracked:  data/NemaScan_Performance.Architecture.20201120.RData
    Untracked:  data/NemaScan_Performance.Architecture.CI50.20201130.RData
    Untracked:  data/NemaScan_Performance.Architecture.CI50.group500.20201201.RData
    Untracked:  data/NemaScan_Performance.Architecture.group500.20201124.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_GAMMA_EMMA_hardfiltered.20201020.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_PowerPrecision.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_PowerPrecision_GAMMA.20200909.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_PowerPrecision_GAMMA.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_PowerPrecision_GAMMA_EMMA.20200924.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_PowerPrecision_REDO.20200909.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_cendr_sets.20200929.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_population_subsampling.20200925.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_population_subsampling.20201018.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_population_subsampling.20201019.RData
    Untracked:  data/NemaScan_Performance.Variant.Set.HF.chr125.20201112.RData
    Untracked:  data/NemaScan_Performance.Variant.Set.Hard.Filtered.20201111.RData
    Untracked:  data/NemaScan_Performance.Variant.Set.Imputed.20201111.RData
    Untracked:  data/NemaScan_Performance.Variant.Set.Imputed.NonDivergent.20201112.RData
    Untracked:  data/NemaScan_Performance.Variant.Set.Imputed.chr125.20201112.RData
    Untracked:  data/TO_SIMS.bim
    Untracked:  data/all_susie_BF_hits.tsv
    Untracked:  data/all_susie_hits.tsv
    Untracked:  data/causal.variants.sim.100.txt
    Untracked:  data/causal.variants.sim.5.txt
    Untracked:  data/hard.filtered.snps.bim
    Untracked:  data/imputed.snps.bim
    Untracked:  output/2014_toxin_phenos.tsv
    Untracked:  output/5_104_0.8_0.05_0.5-5_hahnel.isotypes.manhattan.plot.png
    Untracked:  output/5_12_0.2_0.05_0.5-5_hahnel.isotypes.manhattan.plot.png
    Untracked:  output/5_13_0.7_0.05_0.5-5_hahnel.isotypes.manhattan.plot.png
    Untracked:  output/5_13_0.8_0.05_0.5-5_hahnel.isotypes.manhattan.plot.png
    Untracked:  output/5_140_0.8_0.05_0.5-5_hahnel.isotypes.manhattan.plot.png
    Untracked:  output/5_14_0.1_0.05_0.5-5_hahnel.isotypes.manhattan.plot.png
    Untracked:  output/5_158_0.8_0.05_0.5-5_hahnel.isotypes.manhattan.plot.png
    Untracked:  output/5_21_0.5_0.05_0.5-5_hahnel.isotypes.manhattan.plot.png
    Untracked:  output/5_21_0.9_0.05_0.5-5_hahnel.isotypes.manhattan.plot.png
    Untracked:  output/5_22_0.5_0.05_0.5-5_hahnel.isotypes.manhattan.plot.png
    Untracked:  output/5_22_0.9_0.05_0.5-5_hahnel.isotypes.manhattan.plot.png
    Untracked:  output/5_27_0.3_0.05_0.5-5_hahnel.isotypes.manhattan.plot.png
    Untracked:  output/5_27_0.9_0.05_0.5-5_hahnel.isotypes.manhattan.plot.png
    Untracked:  output/5_34_0.5_0.05_0.5-5_hahnel.isotypes.manhattan.plot.png
    Untracked:  output/5_42_0.5_0.05_0.5-5_hahnel.isotypes.manhattan.plot.png
    Untracked:  output/n.divergent.variants.in.variant.set.png
    Untracked:  output/perc.divergent.variants.in.variant.set.png
    Untracked:  output/prelim.fig.1.png
    Untracked:  output/susie.validation.csv
    Untracked:  power.plot.png

Unstaged changes:
    Modified:   analysis/algorithm.evaluation.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/architecture.Rmd</code>) and HTML (<code>docs/architecture.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/68829fe7ff9d1f8846df2edf35882f6091f73db2/analysis/architecture.Rmd" target="_blank">68829fe</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-12-01
</td>
<td>
integrate multiple CI and grouping params
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/c39b2011fae9980784fa5eae10c5ae4344ac1c68/analysis/architecture.Rmd" target="_blank">c39b201</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-12-01
</td>
<td>
integrate multiple CI and grouping params
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/f24b92958c68ffbab3e8afa43b345fa2aab9a69c/analysis/architecture.Rmd" target="_blank">f24b929</a>
</td>
<td>
Sam
</td>
<td>
2020-11-30
</td>
<td>
turkey day changes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/fcfac6a4c18bc150a0a338a12ee612b61b6514b4/docs/architecture.html" target="_blank">fcfac6a</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/385b5cbb02ca682a4e03d95e43e97ee7f291f4d9/analysis/architecture.Rmd" target="_blank">385b5cb</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
<td>
prelim figures
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/a07596f57a8cbd1751a9031ed99607b0312adcb9/docs/architecture.html" target="_blank">a07596f</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/b6ccbbf8f25af40a7fdab34e64d867b1af20f9ac/analysis/architecture.Rmd" target="_blank">b6ccbbf</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
<td>
silence summarise stuff
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/9114b11f64c5131678476846985b2cc2298bb004/analysis/architecture.Rmd" target="_blank">9114b11</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
<td>
update title
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/d4f156386b38dfc0d9ed09fc4d471ad95ba7d0d0/docs/architecture.html" target="_blank">d4f1563</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/30ce6a242ca70ce803a061aab07cad23693c8257/analysis/architecture.Rmd" target="_blank">30ce6a2</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
<td>
update plots
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="analysis-date-december-01-2020" class="section level3">
<h3>Analysis date: December 01, 2020</h3>
</div>
<div id="nemascan-simulation-performance-was-assessed-with-the-following-experimental-parameters" class="section level3">
<h3>NemaScan simulation performance was assessed with the following experimental parameters:</h3>
<ul>
<li><h6 id="number-of-simulated-qtl-1-5-10-25">Number of Simulated QTL: <code>1, 5, 10, 25</code></h6></li>
<li><h6 id="sample-populations-complete">Sample Population(s): <code>complete</code></h6></li>
<li><h6 id="heritabilityies-0.2-0.4-0.6-0.8">Heritability(ies): <code>0.2, 0.4, 0.6, 0.8</code></h6></li>
<li><h6 id="mafs-0.05">MAF(s): <code>0.05</code></h6></li>
<li><h6 id="number-of-replicates-per-regime-100">Number of Replicates per Regime: <code>100</code></h6></li>
<li><h6 id="qtl-effect-range-gamma">QTL Effect Range: <code>gamma</code></h6></li>
</ul>
<p><img src="figure/architecture.Rmd/simulated%20QTL%20locations-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-simulated-QTL-locations-1">
Past versions of “simulated QTL locations-1.png”
</button>
</p>
<div id="fig-no-spaces-simulated-QTL-locations-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/d4f156386b38dfc0d9ed09fc4d471ad95ba7d0d0/docs/figure/architecture.Rmd/simulated QTL locations-1.png" target="_blank">d4f1563</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/architecture.Rmd/simulated%20QTL%20locations-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-simulated-QTL-locations-2">
Past versions of “simulated QTL locations-2.png”
</button>
</p>
<div id="fig-no-spaces-simulated-QTL-locations-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/d4f156386b38dfc0d9ed09fc4d471ad95ba7d0d0/docs/figure/architecture.Rmd/simulated QTL locations-2.png" target="_blank">d4f1563</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/architecture.Rmd/simulated%20QTL%20locations-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-simulated-QTL-locations-3">
Past versions of “simulated QTL locations-3.png”
</button>
</p>
<div id="fig-no-spaces-simulated-QTL-locations-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/d4f156386b38dfc0d9ed09fc4d471ad95ba7d0d0/docs/figure/architecture.Rmd/simulated QTL locations-3.png" target="_blank">d4f1563</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>   h2        algorithm group.size  CI           1           5          10
1 0.2 LMM-EXACT-INBRED       1000 150 1.48 ± 1.32  1.29 ± 0.8 1.83 ± 1.75
2 0.2   LMM-EXACT-LOCO       1000 150 1.21 ± 0.46 1.27 ± 0.54 1.28 ± 0.68
3 0.4 LMM-EXACT-INBRED       1000 150 1.34 ± 0.98 1.56 ± 1.03 1.73 ± 0.74
4 0.4   LMM-EXACT-LOCO       1000 150 1.87 ± 1.58 1.94 ± 1.34 2.02 ± 1.01
5 0.6 LMM-EXACT-INBRED       1000 150 1.92 ± 1.89 2.14 ± 1.65 2.15 ± 1.06
6 0.6   LMM-EXACT-LOCO       1000 150 2.76 ± 2.55 2.66 ± 1.82  2.8 ± 1.43
7 0.8 LMM-EXACT-INBRED       1000 150 2.33 ± 2.16 2.51 ± 1.83 2.68 ± 1.38
8 0.8   LMM-EXACT-LOCO       1000 150 3.45 ± 2.86  3.2 ± 1.89 3.22 ± 1.63
           25
1 2.05 ± 1.99
2 1.12 ± 0.33
3 1.61 ± 0.78
4 1.95 ± 1.01
5 2.21 ± 1.24
6 2.59 ± 1.29
7 2.95 ± 1.75
8 3.44 ± 1.71</code></pre>
<pre><code>Joining, by = c(&quot;h2&quot;, &quot;algorithm&quot;, &quot;group.size&quot;, &quot;CI&quot;)</code></pre>
<p><img src="figure/architecture.Rmd/PR%20curve-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-PR-curve-1">
Past versions of “PR curve-1.png”
</button>
</p>
<div id="fig-no-spaces-PR-curve-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/fcfac6a4c18bc150a0a338a12ee612b61b6514b4/docs/figure/architecture.Rmd/PR curve-1.png" target="_blank">fcfac6a</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/d4f156386b38dfc0d9ed09fc4d471ad95ba7d0d0/docs/figure/architecture.Rmd/PR curve-1.png" target="_blank">d4f1563</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Joining, by = c(&quot;h2&quot;, &quot;algorithm&quot;, &quot;nQTL&quot;, &quot;group.size&quot;, &quot;CI&quot;, &quot;.metric&quot;, &quot;.estimator&quot;, &quot;.estimate&quot;)
Joining, by = c(&quot;h2&quot;, &quot;algorithm&quot;, &quot;nQTL&quot;, &quot;group.size&quot;, &quot;CI&quot;, &quot;.metric&quot;, &quot;.estimator&quot;, &quot;.estimate&quot;)</code></pre>
<pre><code>Joining, by = c(&quot;h2&quot;, &quot;algorithm&quot;, &quot;nQTL&quot;, &quot;group.size&quot;, &quot;CI&quot;)</code></pre>
<p><img src="figure/architecture.Rmd/metrics%20v%20nQTL-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-metrics-v-nQTL-1">
Past versions of “metrics v nQTL-1.png”
</button>
</p>
<div id="fig-no-spaces-metrics-v-nQTL-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/fcfac6a4c18bc150a0a338a12ee612b61b6514b4/docs/figure/architecture.Rmd/metrics v nQTL-1.png" target="_blank">fcfac6a</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/architecture.Rmd/metrics%20v%20nQTL-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-metrics-v-nQTL-2">
Past versions of “metrics v nQTL-2.png”
</button>
</p>
<div id="fig-no-spaces-metrics-v-nQTL-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/fcfac6a4c18bc150a0a338a12ee612b61b6514b4/docs/figure/architecture.Rmd/metrics v nQTL-2.png" target="_blank">fcfac6a</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/architecture.Rmd/metrics%20v%20nQTL-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-metrics-v-nQTL-3">
Past versions of “metrics v nQTL-3.png”
</button>
</p>
<div id="fig-no-spaces-metrics-v-nQTL-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/fcfac6a4c18bc150a0a338a12ee612b61b6514b4/docs/figure/architecture.Rmd/metrics v nQTL-3.png" target="_blank">fcfac6a</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Group Size: 500&quot;
[1] &quot;CI Extension: 50&quot;
# A tibble: 16 x 5
   h2    nQTL  `LMM-EXACT-INBRED` `LMM-EXACT-LOCO` winner
   &lt;fct&gt; &lt;fct&gt;              &lt;dbl&gt;            &lt;dbl&gt; &lt;chr&gt; 
 1 0.2   1                 0.878            0.898  LOCO  
 2 0.2   5                 0.201            0.207  LOCO  
 3 0.2   10                0.131            0.115  INBRED
 4 0.2   25                0.0703           0.0450 INBRED
 5 0.4   1                 0.960            0.960  INBRED
 6 0.4   5                 0.293            0.341  LOCO  
 7 0.4   10                0.165            0.199  LOCO  
 8 0.4   25                0.0750           0.108  LOCO  
 9 0.6   1                 0.959            1      LOCO  
10 0.6   5                 0.337            0.398  LOCO  
11 0.6   10                0.206            0.268  LOCO  
12 0.6   25                0.115            0.186  LOCO  
13 0.8   1                 0.960            1      LOCO  
14 0.8   5                 0.388            0.478  LOCO  
15 0.8   10                0.250            0.338  LOCO  
16 0.8   25                0.153            0.257  LOCO  
[1] &quot;Group Size: 500&quot;
[1] &quot;CI Extension: 150&quot;
# A tibble: 16 x 5
   h2    nQTL  `LMM-EXACT-INBRED` `LMM-EXACT-LOCO` winner
   &lt;fct&gt; &lt;fct&gt;              &lt;dbl&gt;            &lt;dbl&gt; &lt;chr&gt; 
 1 0.2   1                 0.838            0.859  LOCO  
 2 0.2   5                 0.239            0.229  INBRED
 3 0.2   10                0.189            0.114  INBRED
 4 0.2   25                0.121            0.0563 INBRED
 5 0.4   1                 0.89             0.93   LOCO  
 6 0.4   5                 0.271            0.333  LOCO  
 7 0.4   10                0.165            0.218  LOCO  
 8 0.4   25                0.0856           0.125  LOCO  
 9 0.6   1                 0.9              1      LOCO  
10 0.6   5                 0.323            0.418  LOCO  
11 0.6   10                0.226            0.321  LOCO  
12 0.6   25                0.123            0.192  LOCO  
13 0.8   1                 0.91             1      LOCO  
14 0.8   5                 0.369            0.510  LOCO  
15 0.8   10                0.284            0.392  LOCO  
16 0.8   25                0.173            0.282  LOCO  
[1] &quot;Group Size: 1000&quot;
[1] &quot;CI Extension: 50&quot;
# A tibble: 16 x 5
   h2    nQTL  `LMM-EXACT-INBRED` `LMM-EXACT-LOCO` winner
   &lt;fct&gt; &lt;fct&gt;              &lt;dbl&gt;            &lt;dbl&gt; &lt;chr&gt; 
 1 0.2   1                 0.806            0.867  LOCO  
 2 0.2   5                 0.247            0.183  INBRED
 3 0.2   10                0.113            0.108  INBRED
 4 0.2   25                0.0570           0.0558 INBRED
 5 0.4   1                 0.898            0.929  LOCO  
 6 0.4   5                 0.263            0.291  LOCO  
 7 0.4   10                0.146            0.198  LOCO  
 8 0.4   25                0.0767           0.124  LOCO  
 9 0.6   1                 0.918            1      LOCO  
10 0.6   5                 0.321            0.384  LOCO  
11 0.6   10                0.201            0.275  LOCO  
12 0.6   25                0.120            0.191  LOCO  
13 0.8   1                 0.918            1      LOCO  
14 0.8   5                 0.357            0.476  LOCO  
15 0.8   10                0.246            0.365  LOCO  
16 0.8   25                0.170            0.286  LOCO  
[1] &quot;Group Size: 1000&quot;
[1] &quot;CI Extension: 150&quot;
# A tibble: 16 x 5
   h2    nQTL  `LMM-EXACT-INBRED` `LMM-EXACT-LOCO` winner
   &lt;fct&gt; &lt;fct&gt;              &lt;dbl&gt;            &lt;dbl&gt; &lt;chr&gt; 
 1 0.2   1                 0.7              0.79   LOCO  
 2 0.2   5                 0.198            0.220  LOCO  
 3 0.2   10                0.139            0.111  INBRED
 4 0.2   25                0.0984           0.0510 INBRED
 5 0.4   1                 0.818            0.909  LOCO  
 6 0.4   5                 0.275            0.353  LOCO  
 7 0.4   10                0.178            0.224  LOCO  
 8 0.4   25                0.0859           0.134  LOCO  
 9 0.6   1                 0.88             1      LOCO  
10 0.6   5                 0.335            0.451  LOCO  
11 0.6   10                0.231            0.322  LOCO  
12 0.6   25                0.130            0.207  LOCO  
13 0.8   1                 0.88             1      LOCO  
14 0.8   5                 0.415            0.555  LOCO  
15 0.8   10                0.290            0.406  LOCO  
16 0.8   25                0.177            0.290  LOCO  </code></pre>
<pre><code>[1] &quot;CI Extension: 50&quot;
[1] &quot;Algorithm: LMM-EXACT-INBRED&quot;
# A tibble: 16 x 5
   h2    nQTL   `500` `1000` winner
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt; 
 1 0.2   1     0.878  0.806  500   
 2 0.4   1     0.960  0.898  500   
 3 0.6   1     0.959  0.918  500   
 4 0.8   1     0.960  0.918  500   
 5 0.2   5     0.201  0.247  1000  
 6 0.4   5     0.293  0.263  500   
 7 0.6   5     0.337  0.321  500   
 8 0.8   5     0.388  0.357  500   
 9 0.2   10    0.131  0.113  500   
10 0.4   10    0.165  0.146  500   
11 0.6   10    0.206  0.201  500   
12 0.8   10    0.250  0.246  500   
13 0.2   25    0.0703 0.0570 500   
14 0.4   25    0.0750 0.0767 1000  
15 0.6   25    0.115  0.120  1000  
16 0.8   25    0.153  0.170  1000  
[1] &quot;CI Extension: 150&quot;
[1] &quot;Algorithm: LMM-EXACT-INBRED&quot;
# A tibble: 16 x 5
   h2    nQTL   `500` `1000` winner
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt; 
 1 0.2   1     0.838  0.7    500   
 2 0.4   1     0.89   0.818  500   
 3 0.6   1     0.9    0.88   500   
 4 0.8   1     0.91   0.88   500   
 5 0.2   5     0.239  0.198  500   
 6 0.4   5     0.271  0.275  1000  
 7 0.6   5     0.323  0.335  1000  
 8 0.8   5     0.369  0.415  1000  
 9 0.2   10    0.189  0.139  500   
10 0.4   10    0.165  0.178  1000  
11 0.6   10    0.226  0.231  1000  
12 0.8   10    0.284  0.290  1000  
13 0.2   25    0.121  0.0984 500   
14 0.4   25    0.0856 0.0859 1000  
15 0.6   25    0.123  0.130  1000  
16 0.8   25    0.173  0.177  1000  
[1] &quot;CI Extension: 50&quot;
[1] &quot;Algorithm: LMM-EXACT-LOCO&quot;
# A tibble: 16 x 5
   h2    nQTL   `500` `1000` winner
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt; 
 1 0.2   1     0.898  0.867  500   
 2 0.4   1     0.960  0.929  500   
 3 0.6   1     1      1      1000  
 4 0.8   1     1      1      1000  
 5 0.2   5     0.207  0.183  500   
 6 0.4   5     0.341  0.291  500   
 7 0.6   5     0.398  0.384  500   
 8 0.8   5     0.478  0.476  500   
 9 0.2   10    0.115  0.108  500   
10 0.4   10    0.199  0.198  500   
11 0.6   10    0.268  0.275  1000  
12 0.8   10    0.338  0.365  1000  
13 0.2   25    0.0450 0.0558 1000  
14 0.4   25    0.108  0.124  1000  
15 0.6   25    0.186  0.191  1000  
16 0.8   25    0.257  0.286  1000  
[1] &quot;CI Extension: 150&quot;
[1] &quot;Algorithm: LMM-EXACT-LOCO&quot;
# A tibble: 16 x 5
   h2    nQTL   `500` `1000` winner
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt; 
 1 0.2   1     0.859  0.79   500   
 2 0.4   1     0.93   0.909  500   
 3 0.6   1     1      1      1000  
 4 0.8   1     1      1      1000  
 5 0.2   5     0.229  0.220  500   
 6 0.4   5     0.333  0.353  1000  
 7 0.6   5     0.418  0.451  1000  
 8 0.8   5     0.510  0.555  1000  
 9 0.2   10    0.114  0.111  500   
10 0.4   10    0.218  0.224  1000  
11 0.6   10    0.321  0.322  1000  
12 0.8   10    0.392  0.406  1000  
13 0.2   25    0.0563 0.0510 500   
14 0.4   25    0.125  0.134  1000  
15 0.6   25    0.192  0.207  1000  
16 0.8   25    0.282  0.290  1000  </code></pre>
<pre><code>[1] &quot;Group Size: 500&quot;
[1] &quot;Algorithm: LMM-EXACT-INBRED&quot;
# A tibble: 16 x 5
   h2    nQTL    `50`  `150` winner
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt; 
 1 0.2   1     0.878  0.838  50    
 2 0.4   1     0.960  0.89   50    
 3 0.6   1     0.959  0.9    50    
 4 0.8   1     0.960  0.91   50    
 5 0.2   5     0.201  0.239  150   
 6 0.4   5     0.293  0.271  50    
 7 0.6   5     0.337  0.323  50    
 8 0.8   5     0.388  0.369  50    
 9 0.2   10    0.131  0.189  150   
10 0.4   10    0.165  0.165  150   
11 0.6   10    0.206  0.226  150   
12 0.8   10    0.250  0.284  150   
13 0.2   25    0.0703 0.121  150   
14 0.4   25    0.0750 0.0856 150   
15 0.6   25    0.115  0.123  150   
16 0.8   25    0.153  0.173  150   
[1] &quot;Group Size: 1000&quot;
[1] &quot;Algorithm: LMM-EXACT-INBRED&quot;
# A tibble: 16 x 5
   h2    nQTL    `50`  `150` winner
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt; 
 1 0.2   1     0.806  0.7    50    
 2 0.4   1     0.898  0.818  50    
 3 0.6   1     0.918  0.88   50    
 4 0.8   1     0.918  0.88   50    
 5 0.2   5     0.247  0.198  50    
 6 0.4   5     0.263  0.275  150   
 7 0.6   5     0.321  0.335  150   
 8 0.8   5     0.357  0.415  150   
 9 0.2   10    0.113  0.139  150   
10 0.4   10    0.146  0.178  150   
11 0.6   10    0.201  0.231  150   
12 0.8   10    0.246  0.290  150   
13 0.2   25    0.0570 0.0984 150   
14 0.4   25    0.0767 0.0859 150   
15 0.6   25    0.120  0.130  150   
16 0.8   25    0.170  0.177  150   
[1] &quot;Group Size: 500&quot;
[1] &quot;Algorithm: LMM-EXACT-LOCO&quot;
# A tibble: 16 x 5
   h2    nQTL    `50`  `150` winner
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt; 
 1 0.2   1     0.898  0.859  50    
 2 0.4   1     0.960  0.93   50    
 3 0.6   1     1      1      150   
 4 0.8   1     1      1      150   
 5 0.2   5     0.207  0.229  150   
 6 0.4   5     0.341  0.333  50    
 7 0.6   5     0.398  0.418  150   
 8 0.8   5     0.478  0.510  150   
 9 0.2   10    0.115  0.114  50    
10 0.4   10    0.199  0.218  150   
11 0.6   10    0.268  0.321  150   
12 0.8   10    0.338  0.392  150   
13 0.2   25    0.0450 0.0563 150   
14 0.4   25    0.108  0.125  150   
15 0.6   25    0.186  0.192  150   
16 0.8   25    0.257  0.282  150   
[1] &quot;Group Size: 1000&quot;
[1] &quot;Algorithm: LMM-EXACT-LOCO&quot;
# A tibble: 16 x 5
   h2    nQTL    `50`  `150` winner
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt; 
 1 0.2   1     0.867  0.79   50    
 2 0.4   1     0.929  0.909  50    
 3 0.6   1     1      1      150   
 4 0.8   1     1      1      150   
 5 0.2   5     0.183  0.220  150   
 6 0.4   5     0.291  0.353  150   
 7 0.6   5     0.384  0.451  150   
 8 0.8   5     0.476  0.555  150   
 9 0.2   10    0.108  0.111  150   
10 0.4   10    0.198  0.224  150   
11 0.6   10    0.275  0.322  150   
12 0.8   10    0.365  0.406  150   
13 0.2   25    0.0558 0.0510 50    
14 0.4   25    0.124  0.134  150   
15 0.6   25    0.191  0.207  150   
16 0.8   25    0.286  0.290  150   </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.3 (2020-02-29)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] data.table_1.13.2        wesanderson_0.3.6.9000   viridis_0.5.1           
 [4] viridisLite_0.3.0        RColorBrewer_1.1-2       nationalparkcolors_0.1.0
 [7] yardstick_0.0.7          workflows_0.2.1          tune_0.1.1              
[10] rsample_0.0.8            recipes_0.1.13           parsnip_0.1.3           
[13] modeldata_0.0.2          infer_0.5.3              dials_0.0.9             
[16] scales_1.1.1             broom_0.7.2              tidymodels_0.1.1        
[19] forcats_0.5.0            stringr_1.4.0            dplyr_1.0.2             
[22] purrr_0.3.4              readr_1.4.0              tidyr_1.1.2             
[25] tibble_3.0.4             ggplot2_3.3.2            tidyverse_1.3.0         
[28] workflowr_1.6.2         

loaded via a namespace (and not attached):
 [1] colorspace_1.4-1   ellipsis_0.3.1     class_7.3-17       rprojroot_1.3-2   
 [5] fs_1.5.0           rstudioapi_0.11    farver_2.0.3       listenv_0.8.0     
 [9] furrr_0.1.0        prodlim_2019.11.13 fansi_0.4.1        lubridate_1.7.9   
[13] xml2_1.3.2         codetools_0.2-16   splines_3.6.3      knitr_1.30        
[17] jsonlite_1.7.1     pROC_1.16.2        dbplyr_1.4.4       compiler_3.6.3    
[21] httr_1.4.2         backports_1.1.10   assertthat_0.2.1   Matrix_1.2-18     
[25] cli_2.1.0          later_1.1.0.1      htmltools_0.5.0    tools_3.6.3       
[29] gtable_0.3.0       glue_1.4.2         Rcpp_1.0.5         cellranger_1.1.0  
[33] DiceDesign_1.8-1   vctrs_0.3.4        iterators_1.0.12   timeDate_3043.102 
[37] gower_0.2.2        xfun_0.18          globals_0.13.1     rvest_0.3.6       
[41] lifecycle_0.2.0    future_1.19.1      MASS_7.3-53        ipred_0.9-9       
[45] hms_0.5.3          promises_1.1.1     parallel_3.6.3     yaml_2.2.1        
[49] gridExtra_2.3      rpart_4.1-15       stringi_1.5.3      highr_0.8         
[53] foreach_1.5.0      lhs_1.1.1          lava_1.6.8         rlang_0.4.8       
[57] pkgconfig_2.0.3    evaluate_0.14      lattice_0.20-41    labeling_0.4.2    
[61] cowplot_1.1.0      tidyselect_1.1.0   plyr_1.8.6         magrittr_1.5      
[65] R6_2.4.1           generics_0.0.2     DBI_1.1.0          pillar_1.4.6      
[69] haven_2.3.1        whisker_0.4        withr_2.3.0        survival_3.2-7    
[73] nnet_7.3-14        modelr_0.1.8       crayon_1.3.4       utf8_1.1.4        
[77] rmarkdown_2.5      grid_3.6.3         readxl_1.3.1       blob_1.2.1        
[81] git2r_0.27.1       reprex_0.3.0       digest_0.6.27      httpuv_1.5.4      
[85] munsell_0.5.0      GPfit_1.0-8       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
