<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Polygenic Architectures are Difficult to Resolve Using GWAS</title>

<script src="site_libs/header-attrs-2.5/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">NemaScan: GWAS in C. elegans Wild Isolates</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sam-widmayer/NemaScan_Performance">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Polygenic Architectures are Difficult to Resolve Using GWAS</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-02-01
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>NemaScan_Performance/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200825code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200825)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200825code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200825)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
~/Documents/projects/NemaScan_Performance/
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsamwidmayerNemaScanPerformancetreec1206c6d48e6ac66101ffc940b39503364917f4atargetblankc1206c6a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sam-widmayer/NemaScan_Performance/tree/c1206c6d48e6ac66101ffc940b39503364917f4a" target="_blank">c1206c6</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsamwidmayerNemaScanPerformancetreec1206c6d48e6ac66101ffc940b39503364917f4atargetblankc1206c6a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sam-widmayer/NemaScan_Performance/tree/c1206c6d48e6ac66101ffc940b39503364917f4a" target="_blank">c1206c6</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/.Rhistory
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  analysis/genomic.landscape.Rmd
    Untracked:  code/2014_toxinphenotypes.R
    Untracked:  code/Aggregate_Mappings.R
    Untracked:  code/Find_Aggregate_Intervals.R
    Untracked:  code/Find_GCTA_Intervals.R
    Untracked:  code/pipeline.plotting.R
    Untracked:  code/startup.R
    Untracked:  code/susie.R
    Untracked:  data/Chr.1.ARM.divergent.regions.bed
    Untracked:  data/Chr.1.ARM.nondivergent.regions.bed
    Untracked:  data/Chr.1.CENTER.divergent.regions.bed
    Untracked:  data/Chr.1.CENTER.nondivergent.regions.bed
    Untracked:  data/Chr.2.ARM.divergent.regions.bed
    Untracked:  data/Chr.2.ARM.nondivergent.regions.bed
    Untracked:  data/Chr.2.CENTER.divergent.regions.bed
    Untracked:  data/Chr.2.CENTER.nondivergent.regions.bed
    Untracked:  data/Chr.3.ARM.divergent.regions.bed
    Untracked:  data/Chr.3.ARM.nondivergent.regions.bed
    Untracked:  data/Chr.3.CENTER.divergent.regions.bed
    Untracked:  data/Chr.3.CENTER.nondivergent.regions.bed
    Untracked:  data/Chr.4.ARM.divergent.regions.bed
    Untracked:  data/Chr.4.ARM.nondivergent.regions.bed
    Untracked:  data/Chr.4.CENTER.divergent.regions.bed
    Untracked:  data/Chr.4.CENTER.nondivergent.regions.bed
    Untracked:  data/Chr.5.ARM.divergent.regions.bed
    Untracked:  data/Chr.5.ARM.nondivergent.regions.bed
    Untracked:  data/Chr.5.CENTER.divergent.regions.bed
    Untracked:  data/Chr.5.CENTER.nondivergent.regions.bed
    Untracked:  data/Chr.6.ARM.divergent.regions.bed
    Untracked:  data/Chr.6.ARM.nondivergent.regions.bed
    Untracked:  data/Chr.6.CENTER.divergent.regions.bed
    Untracked:  data/Chr.6.CENTER.nondivergent.regions.bed
    Untracked:  data/NemaScan_Performance.Algorithm.Sims.20201024.RData
    Untracked:  data/NemaScan_Performance.Algorithm.Sims.20201110_EMMA.RData
    Untracked:  data/NemaScan_Performance.Algorithm.Sims.v.2.20201104.RData
    Untracked:  data/NemaScan_Performance.Architecture.20201120.RData
    Untracked:  data/NemaScan_Performance.Architecture.CI50.20201130.RData
    Untracked:  data/NemaScan_Performance.Architecture.CI50.group500.20201201.RData
    Untracked:  data/NemaScan_Performance.Architecture.Mixed.20201205.RData
    Untracked:  data/NemaScan_Performance.Architecture.Mixed.20201209.RData
    Untracked:  data/NemaScan_Performance.Architecture.Mixed.20201223.RData
    Untracked:  data/NemaScan_Performance.Architecture.Mixed.CI.50.20201224.RData
    Untracked:  data/NemaScan_Performance.Architecture.Mixed.CI50.group500.v2.20210131.RData
    Untracked:  data/NemaScan_Performance.Architecture.Mixed.CI50.v2.20210131.RData
    Untracked:  data/NemaScan_Performance.Architecture.Mixed.group500.20201223.RData
    Untracked:  data/NemaScan_Performance.Architecture.Mixed.group500.CI.50.20201224.RData
    Untracked:  data/NemaScan_Performance.Architecture.Mixed.group500.v2.20210131.RData
    Untracked:  data/NemaScan_Performance.Architecture.Mixed.v2.20210131.RData
    Untracked:  data/NemaScan_Performance.Architecture.group500.20201124.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_GAMMA_EMMA_hardfiltered.20201020.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_PowerPrecision.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_PowerPrecision_GAMMA.20200909.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_PowerPrecision_GAMMA.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_PowerPrecision_GAMMA_EMMA.20200924.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_PowerPrecision_REDO.20200909.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_cendr_sets.20200929.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_population_subsampling.20200925.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_population_subsampling.20201018.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_population_subsampling.20201019.RData
    Untracked:  data/NemaScan_Performance.Demography.SampleSize.Complete.Subsampled.20201230.RData
    Untracked:  data/NemaScan_Performance.Demography.SampleSize.Unswept.Subsampled.20210111.RData
    Untracked:  data/NemaScan_Performance.Genomic.Locations.Chr.1.ARM.divergent.20210126.RData
    Untracked:  data/NemaScan_Performance.Genomic.Locations.Chr.1.ARM.divergent.20210130.RData
    Untracked:  data/NemaScan_Performance.Genomic.Locations.Chr.1.ARM.nondivergent.20210129.RData
    Untracked:  data/NemaScan_Performance.Genomic.Locations.Chr.1.CENTER.divergent.20210129.RData
    Untracked:  data/NemaScan_Performance.Genomic.Locations.Chr.1.CENTER.nondivergent.20210129.RData
    Untracked:  data/NemaScan_Performance.Variant.Set.HF.chr125.20201112.RData
    Untracked:  data/NemaScan_Performance.Variant.Set.Hard.Filtered.20201111.RData
    Untracked:  data/NemaScan_Performance.Variant.Set.Imputed.20201111.RData
    Untracked:  data/NemaScan_Performance.Variant.Set.Imputed.NonDivergent.20201112.RData
    Untracked:  data/NemaScan_Performance.Variant.Set.Imputed.chr125.20201112.RData
    Untracked:  data/NemaScan_Performance.updated.MIPstrain.performance.20210119.RData
    Untracked:  data/NemaScan_Population_Metrics.Demography.SampleSize.Complete.Subsampled.20210111.RData
    Untracked:  data/NemaScan_Population_Metrics.Demography.SampleSize.Unswept.Subsampled.20210111.RData
    Untracked:  data/NemaScan_Strain_Summary.Demography.SampleSize.Complete.Subsampled.20210111.RData
    Untracked:  data/NemaScan_Strain_Summary.Demography.SampleSize.Unswept.Subsampled.20210111.RData
    Untracked:  data/TO_SIMS.bim
    Untracked:  data/all_susie_BF_hits.tsv
    Untracked:  data/all_susie_hits.tsv
    Untracked:  data/causal.variants.sim.100.txt
    Untracked:  data/causal.variants.sim.5.txt
    Untracked:  data/hard.filtered.snps.bim
    Untracked:  data/imputed.snps.bim
    Untracked:  output/10_1_0.6_0.05_gamma_fiscal.burro.manhattan.plot.png
    Untracked:  output/10_2_0.6_0.05_gamma_sublinear.cowrie.manhattan.plot.png
    Untracked:  output/2014_toxin_phenos.tsv
    Untracked:  output/5_1_0.8_CHRIII_5.12MB_effect.plot.png
    Untracked:  output/5_1_0.8_CHRII_4.44MB_effect.plot.png
    Untracked:  output/5_1_0.8_CHRV_5.44MB_effect.plot.png
    Untracked:  output/5_1_0.8_CHRX_1.83MB_effect.plot.png
    Untracked:  output/5_1_0.8_LD.plot.png
    Untracked:  output/5_1_0.8_manhattan.plot.png
    Untracked:  output/5_92_0.2_LD.plot.png
    Untracked:  output/complete.subsampled.strains.txt
    Untracked:  output/n.divergent.variants.in.variant.set.png
    Untracked:  output/perc.divergent.variants.in.variant.set.png
    Untracked:  output/pr.curve.variant.set.png
    Untracked:  output/prelim.fig.1.png
    Untracked:  output/prelim.fig.2.png
    Untracked:  output/prelim.fig.3.png
    Untracked:  output/prelim.supp.fig.A.png
    Untracked:  output/prelim.supp.fig.B.png
    Untracked:  output/subsampled.strains_20201225_n144_reps200.txt
    Untracked:  output/subsampled.strains_20201225_n192_reps200.txt
    Untracked:  output/subsampled.strains_20201225_n240_reps200.txt
    Untracked:  output/subsampled.strains_20201225_n288_reps200.txt
    Untracked:  output/subsampled.strains_20201225_n336_reps200.txt
    Untracked:  output/subsampled.strains_20201225_n384_reps200.txt
    Untracked:  output/subsampled.strains_20201225_n48_reps200.txt
    Untracked:  output/subsampled.strains_20201225_n96_reps200.txt
    Untracked:  output/subsampled.swept.strains_20201225_n120_reps200.txt
    Untracked:  output/subsampled.swept.strains_20201225_n144_reps200.txt
    Untracked:  output/subsampled.swept.strains_20201225_n168_reps200.txt
    Untracked:  output/subsampled.swept.strains_20201225_n24_reps200.txt
    Untracked:  output/subsampled.swept.strains_20201225_n48_reps200.txt
    Untracked:  output/subsampled.swept.strains_20201225_n72_reps200.txt
    Untracked:  output/subsampled.swept.strains_20201225_n96_reps200.txt
    Untracked:  output/subsampled.unswept.strains_20201225_n120_reps200.txt
    Untracked:  output/subsampled.unswept.strains_20201225_n144_reps200.txt
    Untracked:  output/subsampled.unswept.strains_20201225_n168_reps200.txt
    Untracked:  output/subsampled.unswept.strains_20201225_n192_reps200.txt
    Untracked:  output/subsampled.unswept.strains_20201225_n216_reps200.txt
    Untracked:  output/subsampled.unswept.strains_20201225_n24_reps200.txt
    Untracked:  output/subsampled.unswept.strains_20201225_n48_reps200.txt
    Untracked:  output/subsampled.unswept.strains_20201225_n72_reps200.txt
    Untracked:  output/subsampled.unswept.strains_20201225_n96_reps200.txt
    Untracked:  output/susie.validation.csv
    Untracked:  output/unswept.subsampled.strains.txt
    Untracked:  output/var.exp.comps.png

Unstaged changes:
    Modified:   analysis/population.comparison.Rmd
    Modified:   code/commonly.used.sim.locations.R
    Modified:   code/generate_strain_lists_swept.R
    Modified:   output/swept.by.chr.strain.lists.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/architecture.Rmd</code>) and HTML (<code>docs/architecture.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/c1206c6d48e6ac66101ffc940b39503364917f4a/analysis/architecture.Rmd" target="_blank">c1206c6</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-02-01
</td>
<td>
update with correct var.exp in detected QTL
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/81a1cf48b9007aae8aedb895e58a90a0304324b9/docs/architecture.html" target="_blank">81a1cf4</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/cfd024e80bb1a766011640020b61868e72d2f892/analysis/architecture.Rmd" target="_blank">cfd024e</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-19
</td>
<td>
update plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/588afcb45fb6f118bc830511d9a3b7261f7d0ee5/docs/architecture.html" target="_blank">588afcb</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/2956df6ee6ed303d9b61dec31a8299dac5b3b75e/analysis/architecture.Rmd" target="_blank">2956df6</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-05
</td>
<td>
percent diff for grouping, extension
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/d2833ef3a47726ca7263cb6b9edfd5a3ec759491/docs/architecture.html" target="_blank">d2833ef</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/dd4da2f9f0590f77e1f45620f9bbe13c22636f37/analysis/architecture.Rmd" target="_blank">dd4da2f</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-05
</td>
<td>
fix group size tables
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/d1dc9d8308579615bd1d1724e2cc6061faf081ef/docs/architecture.html" target="_blank">d1dc9d8</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/abcb1dd41ab626806f7e5211113e683f9e362482/analysis/architecture.Rmd" target="_blank">abcb1dd</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-04
</td>
<td>
reintegrate tables for grouping and CI
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/8688742d60cc957d9bcb2a93a2fbdd004a2de1dd/docs/architecture.html" target="_blank">8688742</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/66d05e54c3e33000cca7b56d35bd9530c4adc201/analysis/architecture.Rmd" target="_blank">66d05e5</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-04
</td>
<td>
fix fig2A jitter
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/38e8395da3f74c32635e01408170487cf5f63705/analysis/architecture.Rmd" target="_blank">38e8395</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-04
</td>
<td>
wflow_git_commit(files = “analysis/architecture.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/842ad5b04e027ee220b87f6560a87d0dac7270eb/docs/architecture.html" target="_blank">842ad5b</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/b4ff99dafccff5bc379162295813d939e4af372a/analysis/architecture.Rmd" target="_blank">b4ff99d</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-04
</td>
<td>
sd for FDR and power
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/437068532ecf3e9d049ab85bb318ded5fa743fa8/docs/architecture.html" target="_blank">4370685</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/7c6f1265bd9c9e3712b00d0b057937c70ebe588a/analysis/architecture.Rmd" target="_blank">7c6f126</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
<td>
simplify plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/777ef170737613e948f5e7edd003807b67307e08/docs/architecture.html" target="_blank">777ef17</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/2bf68e35243817816f287a6247e2134214726fa6/analysis/architecture.Rmd" target="_blank">2bf68e3</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
<td>
wflow_publish(“analysis/architecture.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/2423b37efaae901b653c973c856d1cc6b2aeedf0/docs/architecture.html" target="_blank">2423b37</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/ed561fc1f5d15e6439afd08e3e2a35a1df1907c2/analysis/architecture.Rmd" target="_blank">ed561fc</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
<td>
adjust pr curve for main result
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/a0cbc46ebdc3711526fe36076abf87b1152f11cc/analysis/architecture.Rmd" target="_blank">a0cbc46</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
<td>
adjust pr curve for main result
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/03660cc13e021b39173f15faf11594a8d9bcdcd3/docs/architecture.html" target="_blank">03660cc</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/8a6a1eb1f341774473a1dee4217c29dbb04967e7/analysis/architecture.Rmd" target="_blank">8a6a1eb</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
<td>
comment out supplementary tables
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/5ad923569f9055ce3d127bbe2dd40f74507ea07d/analysis/architecture.Rmd" target="_blank">5ad9235</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
<td>
update with new aggregated runs, group sizes, and CIs
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/77ecc62128790056a493e281ea421fb10d5ca90b/analysis/architecture.Rmd" target="_blank">77ecc62</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-12-18
</td>
<td>
xmas push
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/f5961126a1ae6bb94da128b2ab9367093cad10c8/docs/architecture.html" target="_blank">f596112</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-12-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/de6baedf907967b14303c5608041f159480ffc59/analysis/architecture.Rmd" target="_blank">de6baed</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-12-02
</td>
<td>
add FPR evaluation for CIs and groupings
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/c69064b8e481f2c1e9ed48f16e0ffca40e1ffeab/docs/architecture.html" target="_blank">c69064b</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-12-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/68829fe7ff9d1f8846df2edf35882f6091f73db2/analysis/architecture.Rmd" target="_blank">68829fe</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-12-01
</td>
<td>
integrate multiple CI and grouping params
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/c39b2011fae9980784fa5eae10c5ae4344ac1c68/analysis/architecture.Rmd" target="_blank">c39b201</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-12-01
</td>
<td>
integrate multiple CI and grouping params
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/f24b92958c68ffbab3e8afa43b345fa2aab9a69c/analysis/architecture.Rmd" target="_blank">f24b929</a>
</td>
<td>
Sam
</td>
<td>
2020-11-30
</td>
<td>
turkey day changes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/fcfac6a4c18bc150a0a338a12ee612b61b6514b4/docs/architecture.html" target="_blank">fcfac6a</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/385b5cbb02ca682a4e03d95e43e97ee7f291f4d9/analysis/architecture.Rmd" target="_blank">385b5cb</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
<td>
prelim figures
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/a07596f57a8cbd1751a9031ed99607b0312adcb9/docs/architecture.html" target="_blank">a07596f</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/b6ccbbf8f25af40a7fdab34e64d867b1af20f9ac/analysis/architecture.Rmd" target="_blank">b6ccbbf</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
<td>
silence summarise stuff
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/9114b11f64c5131678476846985b2cc2298bb004/analysis/architecture.Rmd" target="_blank">9114b11</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
<td>
update title
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/d4f156386b38dfc0d9ed09fc4d471ad95ba7d0d0/docs/architecture.html" target="_blank">d4f1563</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/30ce6a242ca70ce803a061aab07cad23693c8257/analysis/architecture.Rmd" target="_blank">30ce6a2</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
<td>
update plots
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="analysis-date-february-01-2021" class="section level3">
<h3>Analysis date: February 01, 2021</h3>
</div>
<div id="nemascan-simulation-performance-was-assessed-with-the-following-experimental-parameters" class="section level3">
<h3>NemaScan simulation performance was assessed with the following experimental parameters:</h3>
<ul>
<li><h6 id="number-of-simulated-qtl-1-5-10-25">Number of Simulated QTL: <code>1, 5, 10, 25</code></h6></li>
<li><h6 id="sample-populations-complete">Sample Population(s): <code>complete</code></h6></li>
<li><h6 id="heritabilityies-0.2-0.4-0.6-0.8">Heritability(ies): <code>0.2, 0.4, 0.6, 0.8</code></h6></li>
<li><h6 id="mafs-0.05">MAF(s): <code>0.05</code></h6></li>
<li><h6 id="number-of-replicates-per-regime-100">Number of Replicates per Regime: <code>100</code></h6></li>
<li><h6 id="qtl-effect-range-gamma">QTL Effect Range: <code>gamma</code></h6></li>
</ul>
<p><img src="figure/architecture.Rmd/simulated%20QTL%20locations-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-simulated-QTL-locations-1">
Past versions of “simulated QTL locations-1.png”
</button>
</p>
<div id="fig-no-spaces-simulated-QTL-locations-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/03660cc13e021b39173f15faf11594a8d9bcdcd3/docs/figure/architecture.Rmd/simulated QTL locations-1.png" target="_blank">03660cc</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/c69064b8e481f2c1e9ed48f16e0ffca40e1ffeab/docs/figure/architecture.Rmd/simulated QTL locations-1.png" target="_blank">c69064b</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-12-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/d4f156386b38dfc0d9ed09fc4d471ad95ba7d0d0/docs/figure/architecture.Rmd/simulated QTL locations-1.png" target="_blank">d4f1563</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/architecture.Rmd/simulated%20QTL%20locations-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-simulated-QTL-locations-2">
Past versions of “simulated QTL locations-2.png”
</button>
</p>
<div id="fig-no-spaces-simulated-QTL-locations-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/03660cc13e021b39173f15faf11594a8d9bcdcd3/docs/figure/architecture.Rmd/simulated QTL locations-2.png" target="_blank">03660cc</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/c69064b8e481f2c1e9ed48f16e0ffca40e1ffeab/docs/figure/architecture.Rmd/simulated QTL locations-2.png" target="_blank">c69064b</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-12-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/d4f156386b38dfc0d9ed09fc4d471ad95ba7d0d0/docs/figure/architecture.Rmd/simulated QTL locations-2.png" target="_blank">d4f1563</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/architecture.Rmd/simulated%20QTL%20locations-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-simulated-QTL-locations-3">
Past versions of “simulated QTL locations-3.png”
</button>
</p>
<div id="fig-no-spaces-simulated-QTL-locations-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/03660cc13e021b39173f15faf11594a8d9bcdcd3/docs/figure/architecture.Rmd/simulated QTL locations-3.png" target="_blank">03660cc</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/c69064b8e481f2c1e9ed48f16e0ffca40e1ffeab/docs/figure/architecture.Rmd/simulated QTL locations-3.png" target="_blank">c69064b</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-12-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/d4f156386b38dfc0d9ed09fc4d471ad95ba7d0d0/docs/figure/architecture.Rmd/simulated QTL locations-3.png" target="_blank">d4f1563</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/architecture.Rmd/simulated%20QTL%20locations-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-simulated-QTL-locations-4">
Past versions of “simulated QTL locations-4.png”
</button>
</p>
<div id="fig-no-spaces-simulated-QTL-locations-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/fcfac6a4c18bc150a0a338a12ee612b61b6514b4/docs/figure/architecture.Rmd/simulated QTL locations-4.png" target="_blank">fcfac6a</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/d4f156386b38dfc0d9ed09fc4d471ad95ba7d0d0/docs/figure/architecture.Rmd/simulated QTL locations-4.png" target="_blank">d4f1563</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/architecture.Rmd/simulated%20QTL%20locations-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code># A tibble: 16 x 4
# Groups:   nQTL, h2 [16]
   nQTL  h2    beavis  good
   &lt;fct&gt; &lt;fct&gt;  &lt;int&gt; &lt;int&gt;
 1 1     0.2       54    28
 2 1     0.4       55    35
 3 1     0.6       59    40
 4 1     0.8       55    43
 5 5     0.2      105    26
 6 5     0.4      116    47
 7 5     0.6      161    58
 8 5     0.8      189    82
 9 10    0.2      157    32
10 10    0.4      144    71
11 10    0.6      194    90
12 10    0.8      261   125
13 25    0.2      257    25
14 25    0.4      303    58
15 25    0.6      428   106
16 25    0.8      568   176</code></pre>
<p><img src="figure/architecture.Rmd/default%20settings%20PR%20curve-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-default-settings-PR-curve-1">
Past versions of “default settings PR curve-1.png”
</button>
</p>
<div id="fig-no-spaces-default-settings-PR-curve-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/842ad5b04e027ee220b87f6560a87d0dac7270eb/docs/figure/architecture.Rmd/default settings PR curve-1.png" target="_blank">842ad5b</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Joining, by = c(&quot;h2&quot;, &quot;algorithm&quot;, &quot;nQTL&quot;, &quot;group.size&quot;, &quot;CI&quot;, &quot;.metric&quot;, &quot;.estimator&quot;, &quot;.estimate&quot;)
Joining, by = c(&quot;h2&quot;, &quot;algorithm&quot;, &quot;nQTL&quot;, &quot;group.size&quot;, &quot;CI&quot;, &quot;.metric&quot;, &quot;.estimator&quot;, &quot;.estimate&quot;)</code></pre>
<pre><code>Joining, by = c(&quot;h2&quot;, &quot;algorithm&quot;, &quot;nQTL&quot;, &quot;group.size&quot;, &quot;CI&quot;)</code></pre>
<p><img src="figure/architecture.Rmd/plots-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plots-1">
Past versions of plots-1.png
</button>
</p>
<div id="fig-plots-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/81a1cf48b9007aae8aedb895e58a90a0304324b9/docs/figure/architecture.Rmd/plots-1.png" target="_blank">81a1cf4</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/8688742d60cc957d9bcb2a93a2fbdd004a2de1dd/docs/figure/architecture.Rmd/plots-1.png" target="_blank">8688742</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/842ad5b04e027ee220b87f6560a87d0dac7270eb/docs/figure/architecture.Rmd/plots-1.png" target="_blank">842ad5b</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/437068532ecf3e9d049ab85bb318ded5fa743fa8/docs/figure/architecture.Rmd/plots-1.png" target="_blank">4370685</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/2423b37efaae901b653c973c856d1cc6b2aeedf0/docs/figure/architecture.Rmd/plots-1.png" target="_blank">2423b37</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/03660cc13e021b39173f15faf11594a8d9bcdcd3/docs/figure/architecture.Rmd/plots-1.png" target="_blank">03660cc</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/architecture.Rmd/plots-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plots-2">
Past versions of plots-2.png
</button>
</p>
<div id="fig-plots-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/81a1cf48b9007aae8aedb895e58a90a0304324b9/docs/figure/architecture.Rmd/plots-2.png" target="_blank">81a1cf4</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/8688742d60cc957d9bcb2a93a2fbdd004a2de1dd/docs/figure/architecture.Rmd/plots-2.png" target="_blank">8688742</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/842ad5b04e027ee220b87f6560a87d0dac7270eb/docs/figure/architecture.Rmd/plots-2.png" target="_blank">842ad5b</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/architecture.Rmd/plots-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plots-3">
Past versions of plots-3.png
</button>
</p>
<div id="fig-plots-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/81a1cf48b9007aae8aedb895e58a90a0304324b9/docs/figure/architecture.Rmd/plots-3.png" target="_blank">81a1cf4</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/8688742d60cc957d9bcb2a93a2fbdd004a2de1dd/docs/figure/architecture.Rmd/plots-3.png" target="_blank">8688742</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/842ad5b04e027ee220b87f6560a87d0dac7270eb/docs/figure/architecture.Rmd/plots-3.png" target="_blank">842ad5b</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/architecture.Rmd/plots-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plots-4">
Past versions of plots-4.png
</button>
</p>
<div id="fig-plots-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/81a1cf48b9007aae8aedb895e58a90a0304324b9/docs/figure/architecture.Rmd/plots-4.png" target="_blank">81a1cf4</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/architecture.Rmd/plots-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plots-5">
Past versions of plots-5.png
</button>
</p>
<div id="fig-plots-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/81a1cf48b9007aae8aedb895e58a90a0304324b9/docs/figure/architecture.Rmd/plots-5.png" target="_blank">81a1cf4</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/architecture.Rmd/plots-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plots-6">
Past versions of plots-6.png
</button>
</p>
<div id="fig-plots-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/81a1cf48b9007aae8aedb895e58a90a0304324b9/docs/figure/architecture.Rmd/plots-6.png" target="_blank">81a1cf4</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/architecture.Rmd/plots-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plots-7">
Past versions of plots-7.png
</button>
</p>
<div id="fig-plots-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/81a1cf48b9007aae8aedb895e58a90a0304324b9/docs/figure/architecture.Rmd/plots-7.png" target="_blank">81a1cf4</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="tables" class="section level2">
<h2>TABLES</h2>
<pre><code>[1] &quot;POWER EVALUATION&quot;</code></pre>
<pre><code>[1] &quot;Group Size: 500&quot;
[1] &quot;CI Extension: 50&quot;
# A tibble: 16 x 5
   h2    nQTL  `LMM-EXACT-INBRED` `LMM-EXACT-LOCO`  MIXED
   &lt;fct&gt; &lt;fct&gt;              &lt;dbl&gt;            &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1                 0.808            0.838  0.828 
 2 0.2   5                 0.224            0.2    0.24  
 3 0.2   10                0.167            0.103  0.178 
 4 0.2   25                0.0871           0.0442 0.0988
 5 0.4   1                 0.869            0.909  0.909 
 6 0.4   5                 0.258            0.306  0.308 
 7 0.4   10                0.168            0.198  0.200 
 8 0.4   25                0.0735           0.119  0.120 
 9 0.6   1                 0.899            1      1     
10 0.6   5                 0.324            0.406  0.41  
11 0.6   10                0.202            0.264  0.267 
12 0.6   25                0.110            0.184  0.187 
13 0.8   1                 0.918            1      1     
14 0.8   5                 0.364            0.502  0.506 
15 0.8   10                0.255            0.348  0.356 
16 0.8   25                0.151            0.257  0.259 
[1] &quot;Group Size: 500&quot;
[1] &quot;CI Extension: 150&quot;
# A tibble: 16 x 5
   h2    nQTL  `LMM-EXACT-INBRED` `LMM-EXACT-LOCO` MIXED
   &lt;fct&gt; &lt;fct&gt;              &lt;dbl&gt;            &lt;dbl&gt; &lt;dbl&gt;
 1 0.2   1                 0.808            0.838  0.828
 2 0.2   5                 0.238            0.21   0.258
 3 0.2   10                0.173            0.111  0.187
 4 0.2   25                0.0955           0.0534 0.111
 5 0.4   1                 0.879            0.909  0.909
 6 0.4   5                 0.268            0.322  0.324
 7 0.4   10                0.179            0.211  0.215
 8 0.4   25                0.0899           0.137  0.138
 9 0.6   1                 0.899            1      1    
10 0.6   5                 0.336            0.43   0.434
11 0.6   10                0.224            0.276  0.281
12 0.6   25                0.128            0.207  0.211
13 0.8   1                 0.918            1      1    
14 0.8   5                 0.384            0.518  0.53 
15 0.8   10                0.276            0.364  0.380
16 0.8   25                0.175            0.285  0.291
[1] &quot;Group Size: 1000&quot;
[1] &quot;CI Extension: 50&quot;
# A tibble: 16 x 5
   h2    nQTL  `LMM-EXACT-INBRED` `LMM-EXACT-LOCO` MIXED
   &lt;fct&gt; &lt;fct&gt;              &lt;dbl&gt;            &lt;dbl&gt; &lt;dbl&gt;
 1 0.2   1                 0.808            0.838  0.828
 2 0.2   5                 0.234            0.2    0.248
 3 0.2   10                0.170            0.104  0.182
 4 0.2   25                0.0899           0.0450 0.102
 5 0.4   1                 0.869            0.909  0.909
 6 0.4   5                 0.26             0.31   0.312
 7 0.4   10                0.170            0.202  0.204
 8 0.4   25                0.0767           0.126  0.127
 9 0.6   1                 0.899            1      1    
10 0.6   5                 0.326            0.412  0.416
11 0.6   10                0.207            0.273  0.276
12 0.6   25                0.113            0.189  0.192
13 0.8   1                 0.918            1      1    
14 0.8   5                 0.374            0.516  0.522
15 0.8   10                0.262            0.357  0.366
16 0.8   25                0.156            0.263  0.268
[1] &quot;Group Size: 1000&quot;
[1] &quot;CI Extension: 150&quot;
# A tibble: 16 x 5
   h2    nQTL  `LMM-EXACT-INBRED` `LMM-EXACT-LOCO` MIXED
   &lt;fct&gt; &lt;fct&gt;              &lt;dbl&gt;            &lt;dbl&gt; &lt;dbl&gt;
 1 0.2   1                 0.808            0.838  0.828
 2 0.2   5                 0.244            0.21   0.262
 3 0.2   10                0.176            0.111  0.190
 4 0.2   25                0.0971           0.0542 0.113
 5 0.4   1                 0.879            0.909  0.909
 6 0.4   5                 0.27             0.324  0.326
 7 0.4   10                0.180            0.212  0.216
 8 0.4   25                0.0915           0.144  0.145
 9 0.6   1                 0.899            1      1    
10 0.6   5                 0.338            0.434  0.438
11 0.6   10                0.225            0.280  0.285
12 0.6   25                0.130            0.210  0.214
13 0.8   1                 0.918            1      1    
14 0.8   5                 0.394            0.528  0.542
15 0.8   10                0.280            0.370  0.388
16 0.8   25                0.179            0.290  0.299</code></pre>
<pre><code>[1] &quot;CI Extension: 50&quot;
[1] &quot;Algorithm: LMM-EXACT-INBRED&quot;
# A tibble: 16 x 5
   h2    nQTL   `500` `1000` grouping.diff
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;        
 1 0.2   1     0.808  0.808  0%           
 2 0.4   1     0.869  0.869  0%           
 3 0.6   1     0.899  0.899  0%           
 4 0.8   1     0.918  0.918  0%           
 5 0.2   5     0.224  0.234  1%           
 6 0.4   5     0.258  0.26   0.2%         
 7 0.6   5     0.324  0.326  0.2%         
 8 0.8   5     0.364  0.374  1%           
 9 0.2   10    0.167  0.170  0.3%         
10 0.4   10    0.168  0.170  0.2%         
11 0.6   10    0.202  0.207  0.5%         
12 0.8   10    0.255  0.262  0.7%         
13 0.2   25    0.0871 0.0899 0.28%        
14 0.4   25    0.0735 0.0767 0.32%        
15 0.6   25    0.110  0.113  0.28%        
16 0.8   25    0.151  0.156  0.56%        
[1] &quot;CI Extension: 150&quot;
[1] &quot;Algorithm: LMM-EXACT-INBRED&quot;
# A tibble: 16 x 5
   h2    nQTL   `500` `1000` grouping.diff
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;        
 1 0.2   1     0.808  0.808  0%           
 2 0.4   1     0.879  0.879  0%           
 3 0.6   1     0.899  0.899  0%           
 4 0.8   1     0.918  0.918  0%           
 5 0.2   5     0.238  0.244  0.6%         
 6 0.4   5     0.268  0.27   0.2%         
 7 0.6   5     0.336  0.338  0.2%         
 8 0.8   5     0.384  0.394  1%           
 9 0.2   10    0.173  0.176  0.3%         
10 0.4   10    0.179  0.180  0.1%         
11 0.6   10    0.224  0.225  0.1%         
12 0.8   10    0.276  0.280  0.4%         
13 0.2   25    0.0955 0.0971 0.16%        
14 0.4   25    0.0899 0.0915 0.16%        
15 0.6   25    0.128  0.130  0.2%         
16 0.8   25    0.175  0.179  0.44%        
[1] &quot;CI Extension: 50&quot;
[1] &quot;Algorithm: LMM-EXACT-LOCO&quot;
# A tibble: 16 x 5
   h2    nQTL   `500` `1000` grouping.diff
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;        
 1 0.2   1     0.838  0.838  0%           
 2 0.4   1     0.909  0.909  0%           
 3 0.6   1     1      1      0%           
 4 0.8   1     1      1      0%           
 5 0.2   5     0.2    0.2    0%           
 6 0.4   5     0.306  0.31   0.4%         
 7 0.6   5     0.406  0.412  0.6%         
 8 0.8   5     0.502  0.516  1.4%         
 9 0.2   10    0.103  0.104  0.1%         
10 0.4   10    0.198  0.202  0.4%         
11 0.6   10    0.264  0.273  0.9%         
12 0.8   10    0.348  0.357  0.9%         
13 0.2   25    0.0442 0.0450 0.08%        
14 0.4   25    0.119  0.126  0.72%        
15 0.6   25    0.184  0.189  0.48%        
16 0.8   25    0.257  0.263  0.68%        
[1] &quot;CI Extension: 150&quot;
[1] &quot;Algorithm: LMM-EXACT-LOCO&quot;
# A tibble: 16 x 5
   h2    nQTL   `500` `1000` grouping.diff
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;        
 1 0.2   1     0.838  0.838  0%           
 2 0.4   1     0.909  0.909  0%           
 3 0.6   1     1      1      0%           
 4 0.8   1     1      1      0%           
 5 0.2   5     0.21   0.21   0%           
 6 0.4   5     0.322  0.324  0.2%         
 7 0.6   5     0.43   0.434  0.4%         
 8 0.8   5     0.518  0.528  1%           
 9 0.2   10    0.111  0.111  0%           
10 0.4   10    0.211  0.212  0.1%         
11 0.6   10    0.276  0.280  0.4%         
12 0.8   10    0.364  0.370  0.6%         
13 0.2   25    0.0534 0.0542 0.08%        
14 0.4   25    0.137  0.144  0.64%        
15 0.6   25    0.207  0.210  0.28%        
16 0.8   25    0.285  0.290  0.52%        
[1] &quot;CI Extension: 50&quot;
[1] &quot;Algorithm: MIXED&quot;
# A tibble: 16 x 5
   h2    nQTL   `500` `1000` grouping.diff
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;        
 1 0.2   1     0.828   0.828 0%           
 2 0.4   1     0.909   0.909 0%           
 3 0.6   1     1       1     0%           
 4 0.8   1     1       1     0%           
 5 0.2   5     0.24    0.248 0.8%         
 6 0.4   5     0.308   0.312 0.4%         
 7 0.6   5     0.41    0.416 0.6%         
 8 0.8   5     0.506   0.522 1.6%         
 9 0.2   10    0.178   0.182 0.4%         
10 0.4   10    0.200   0.204 0.4%         
11 0.6   10    0.267   0.276 0.9%         
12 0.8   10    0.356   0.366 1%           
13 0.2   25    0.0988  0.102 0.32%        
14 0.4   25    0.120   0.127 0.72%        
15 0.6   25    0.187   0.192 0.52%        
16 0.8   25    0.259   0.268 0.96%        
[1] &quot;CI Extension: 150&quot;
[1] &quot;Algorithm: MIXED&quot;
# A tibble: 16 x 5
   h2    nQTL  `500` `1000` grouping.diff
   &lt;fct&gt; &lt;fct&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;        
 1 0.2   1     0.828  0.828 0%           
 2 0.4   1     0.909  0.909 0%           
 3 0.6   1     1      1     0%           
 4 0.8   1     1      1     0%           
 5 0.2   5     0.258  0.262 0.4%         
 6 0.4   5     0.324  0.326 0.2%         
 7 0.6   5     0.434  0.438 0.4%         
 8 0.8   5     0.53   0.542 1.2%         
 9 0.2   10    0.187  0.190 0.3%         
10 0.4   10    0.215  0.216 0.1%         
11 0.6   10    0.281  0.285 0.4%         
12 0.8   10    0.380  0.388 0.8%         
13 0.2   25    0.111  0.113 0.2%         
14 0.4   25    0.138  0.145 0.64%        
15 0.6   25    0.211  0.214 0.32%        
16 0.8   25    0.291  0.299 0.72%        </code></pre>
<pre><code>[1] &quot;Group Size: 500&quot;
[1] &quot;Algorithm: LMM-EXACT-INBRED&quot;
# A tibble: 16 x 5
   h2    nQTL    `50`  `150` CI.diff
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;  
 1 0.2   1     0.808  0.808  0%     
 2 0.4   1     0.869  0.879  1.01%  
 3 0.6   1     0.899  0.899  0%     
 4 0.8   1     0.918  0.918  0%     
 5 0.2   5     0.224  0.238  1.4%   
 6 0.4   5     0.258  0.268  1%     
 7 0.6   5     0.324  0.336  1.2%   
 8 0.8   5     0.364  0.384  2%     
 9 0.2   10    0.167  0.173  0.6%   
10 0.4   10    0.168  0.179  1.1%   
11 0.6   10    0.202  0.224  2.21%  
12 0.8   10    0.255  0.276  2.11%  
13 0.2   25    0.0871 0.0955 0.84%  
14 0.4   25    0.0735 0.0899 1.65%  
15 0.6   25    0.110  0.128  1.73%  
16 0.8   25    0.151  0.175  2.41%  
[1] &quot;Group Size: 1000&quot;
[1] &quot;Algorithm: LMM-EXACT-INBRED&quot;
# A tibble: 16 x 5
   h2    nQTL    `50`  `150` CI.diff
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;  
 1 0.2   1     0.808  0.808  0%     
 2 0.4   1     0.869  0.879  1.01%  
 3 0.6   1     0.899  0.899  0%     
 4 0.8   1     0.918  0.918  0%     
 5 0.2   5     0.234  0.244  1%     
 6 0.4   5     0.26   0.27   1%     
 7 0.6   5     0.326  0.338  1.2%   
 8 0.8   5     0.374  0.394  2%     
 9 0.2   10    0.170  0.176  0.6%   
10 0.4   10    0.170  0.180  1%     
11 0.6   10    0.207  0.225  1.81%  
12 0.8   10    0.262  0.280  1.81%  
13 0.2   25    0.0899 0.0971 0.72%  
14 0.4   25    0.0767 0.0915 1.49%  
15 0.6   25    0.113  0.130  1.65%  
16 0.8   25    0.156  0.179  2.29%  
[1] &quot;Group Size: 500&quot;
[1] &quot;Algorithm: LMM-EXACT-LOCO&quot;
# A tibble: 16 x 5
   h2    nQTL    `50`  `150` CI.diff
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;  
 1 0.2   1     0.838  0.838  0%     
 2 0.4   1     0.909  0.909  0%     
 3 0.6   1     1      1      0%     
 4 0.8   1     1      1      0%     
 5 0.2   5     0.2    0.21   1%     
 6 0.4   5     0.306  0.322  1.6%   
 7 0.6   5     0.406  0.43   2.4%   
 8 0.8   5     0.502  0.518  1.6%   
 9 0.2   10    0.103  0.111  0.8%   
10 0.4   10    0.198  0.211  1.31%  
11 0.6   10    0.264  0.276  1.2%   
12 0.8   10    0.348  0.364  1.61%  
13 0.2   25    0.0442 0.0534 0.92%  
14 0.4   25    0.119  0.137  1.81%  
15 0.6   25    0.184  0.207  2.33%  
16 0.8   25    0.257  0.285  2.81%  
[1] &quot;Group Size: 1000&quot;
[1] &quot;Algorithm: LMM-EXACT-LOCO&quot;
# A tibble: 16 x 5
   h2    nQTL    `50`  `150` CI.diff
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;  
 1 0.2   1     0.838  0.838  0%     
 2 0.4   1     0.909  0.909  0%     
 3 0.6   1     1      1      0%     
 4 0.8   1     1      1      0%     
 5 0.2   5     0.2    0.21   1%     
 6 0.4   5     0.31   0.324  1.4%   
 7 0.6   5     0.412  0.434  2.2%   
 8 0.8   5     0.516  0.528  1.2%   
 9 0.2   10    0.104  0.111  0.7%   
10 0.4   10    0.202  0.212  1%     
11 0.6   10    0.273  0.280  0.7%   
12 0.8   10    0.357  0.370  1.31%  
13 0.2   25    0.0450 0.0542 0.92%  
14 0.4   25    0.126  0.144  1.73%  
15 0.6   25    0.189  0.210  2.13%  
16 0.8   25    0.263  0.290  2.65%  
[1] &quot;Group Size: 500&quot;
[1] &quot;Algorithm: MIXED&quot;
# A tibble: 16 x 5
   h2    nQTL    `50` `150` CI.diff
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;  
 1 0.2   1     0.828  0.828 0%     
 2 0.4   1     0.909  0.909 0%     
 3 0.6   1     1      1     0%     
 4 0.8   1     1      1     0%     
 5 0.2   5     0.24   0.258 1.8%   
 6 0.4   5     0.308  0.324 1.6%   
 7 0.6   5     0.41   0.434 2.4%   
 8 0.8   5     0.506  0.53  2.4%   
 9 0.2   10    0.178  0.187 0.9%   
10 0.4   10    0.200  0.215 1.51%  
11 0.6   10    0.267  0.281 1.41%  
12 0.8   10    0.356  0.380 2.31%  
13 0.2   25    0.0988 0.111 1.24%  
14 0.4   25    0.120  0.138 1.89%  
15 0.6   25    0.187  0.211 2.41%  
16 0.8   25    0.259  0.291 3.29%  
[1] &quot;Group Size: 1000&quot;
[1] &quot;Algorithm: MIXED&quot;
# A tibble: 16 x 5
   h2    nQTL   `50` `150` CI.diff
   &lt;fct&gt; &lt;fct&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;  
 1 0.2   1     0.828 0.828 0%     
 2 0.4   1     0.909 0.909 0%     
 3 0.6   1     1     1     0%     
 4 0.8   1     1     1     0%     
 5 0.2   5     0.248 0.262 1.4%   
 6 0.4   5     0.312 0.326 1.4%   
 7 0.6   5     0.416 0.438 2.2%   
 8 0.8   5     0.522 0.542 2%     
 9 0.2   10    0.182 0.190 0.8%   
10 0.4   10    0.204 0.216 1.2%   
11 0.6   10    0.276 0.285 0.9%   
12 0.8   10    0.366 0.388 2.11%  
13 0.2   25    0.102 0.113 1.12%  
14 0.4   25    0.127 0.145 1.81%  
15 0.6   25    0.192 0.214 2.21%  
16 0.8   25    0.268 0.299 3.05%  </code></pre>
<pre><code>[1] &quot;FDR EVALUATION&quot;</code></pre>
<pre><code>[1] &quot;Group Size: 500&quot;
[1] &quot;CI Extension: 50&quot;
# A tibble: 16 x 5
   h2    nQTL  `LMM-EXACT-INBRED` `LMM-EXACT-LOCO`  MIXED
   &lt;fct&gt; &lt;fct&gt;              &lt;dbl&gt;            &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1                 0.370            0.117  0.401 
 2 0.2   5                 0.367            0.107  0.362 
 3 0.2   10                0.131            0.104  0.153 
 4 0.2   25                0.0882           0.0984 0.102 
 5 0.4   1                 0.259            0.458  0.474 
 6 0.4   5                 0.204            0.271  0.274 
 7 0.4   10                0.0874           0.147  0.164 
 8 0.4   25                0.0804           0.0945 0.0942
 9 0.6   1                 0.522            0.637  0.666 
10 0.6   5                 0.299            0.364  0.379 
11 0.6   10                0.193            0.201  0.218 
12 0.6   25                0.0646           0.0653 0.0717
13 0.8   1                 0.628            0.703  0.729 
14 0.8   5                 0.413            0.426  0.456 
15 0.8   10                0.282            0.286  0.319 
16 0.8   25                0.101            0.0766 0.0968
[1] &quot;Group Size: 500&quot;
[1] &quot;CI Extension: 150&quot;
# A tibble: 16 x 5
   h2    nQTL  `LMM-EXACT-INBRED` `LMM-EXACT-LOCO`  MIXED
   &lt;fct&gt; &lt;fct&gt;              &lt;dbl&gt;            &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1                 0.370            0.117  0.401 
 2 0.2   5                 0.339            0.0948 0.332 
 3 0.2   10                0.113            0.0750 0.131 
 4 0.2   25                0.0703           0.0567 0.0705
 5 0.4   1                 0.25             0.458  0.474 
 6 0.4   5                 0.188            0.255  0.257 
 7 0.4   10                0.0729           0.125  0.137 
 8 0.4   25                0.0468           0.0630 0.0625
 9 0.6   1                 0.522            0.637  0.666 
10 0.6   5                 0.285            0.336  0.352 
11 0.6   10                0.155            0.186  0.198 
12 0.6   25                0.0479           0.0462 0.0488
13 0.8   1                 0.628            0.703  0.729 
14 0.8   5                 0.385            0.409  0.431 
15 0.8   10                0.251            0.265  0.291 
16 0.8   25                0.0705           0.0646 0.0752
[1] &quot;Group Size: 1000&quot;
[1] &quot;CI Extension: 50&quot;
# A tibble: 16 x 5
   h2    nQTL  `LMM-EXACT-INBRED` `LMM-EXACT-LOCO`  MIXED
   &lt;fct&gt; &lt;fct&gt;              &lt;dbl&gt;            &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1                 0.328            0.0879 0.349 
 2 0.2   5                 0.278            0.0909 0.279 
 3 0.2   10                0.101            0.0714 0.113 
 4 0.2   25                0.0549           0.0667 0.0662
 5 0.4   1                 0.232            0.388  0.404 
 6 0.4   5                 0.156            0.217  0.220 
 7 0.4   10                0.0714           0.0946 0.114 
 8 0.4   25                0.0498           0.0541 0.0539
 9 0.6   1                 0.451            0.570  0.599 
10 0.6   5                 0.249            0.285  0.302 
11 0.6   10                0.159            0.153  0.169 
12 0.6   25                0.0375           0.0369 0.042 
13 0.8   1                 0.548            0.622  0.646 
14 0.8   5                 0.310            0.314  0.339 
15 0.8   10                0.207            0.202  0.220 
16 0.8   25                0.0760           0.0479 0.0592
[1] &quot;Group Size: 1000&quot;
[1] &quot;CI Extension: 150&quot;
# A tibble: 16 x 5
   h2    nQTL  `LMM-EXACT-INBRED` `LMM-EXACT-LOCO`  MIXED
   &lt;fct&gt; &lt;fct&gt;              &lt;dbl&gt;            &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1                 0.328            0.0879 0.349 
 2 0.2   5                 0.261            0.0789 0.260 
 3 0.2   10                0.0838           0.0513 0.0957
 4 0.2   25                0.0435           0.0288 0.0408
 5 0.4   1                 0.223            0.388  0.404 
 6 0.4   5                 0.140            0.202  0.205 
 7 0.4   10                0.0628           0.0786 0.0928
 8 0.4   25                0.0298           0.0298 0.0296
 9 0.6   1                 0.451            0.570  0.599 
10 0.6   5                 0.235            0.257  0.275 
11 0.6   10                0.132            0.144  0.155 
12 0.6   25                0.0242           0.0261 0.0273
13 0.8   1                 0.548            0.622  0.646 
14 0.8   5                 0.278            0.300  0.316 
15 0.8   10                0.182            0.191  0.201 
16 0.8   25                0.0551           0.0399 0.0449</code></pre>
<pre><code>[1] &quot;CI Extension: 50&quot;
[1] &quot;Algorithm: LMM-EXACT-INBRED&quot;
# A tibble: 16 x 5
   h2    nQTL   `500` `1000` grouping.diff
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;        
 1 0.2   1     0.370  0.328  -4.23%       
 2 0.2   5     0.367  0.278  -8.95%       
 3 0.2   10    0.131  0.101  -2.98%       
 4 0.2   25    0.0882 0.0549 -3.34%       
 5 0.4   1     0.259  0.232  -2.65%       
 6 0.4   5     0.204  0.156  -4.79%       
 7 0.4   10    0.0874 0.0714 -1.6%        
 8 0.4   25    0.0804 0.0498 -3.07%       
 9 0.6   1     0.522  0.451  -7.09%       
10 0.6   5     0.299  0.249  -4.99%       
11 0.6   10    0.193  0.159  -3.36%       
12 0.6   25    0.0646 0.0375 -2.71%       
13 0.8   1     0.628  0.548  -8.04%       
14 0.8   5     0.413  0.310  -10.29%      
15 0.8   10    0.282  0.207  -7.58%       
16 0.8   25    0.101  0.0760 -2.47%       
[1] &quot;CI Extension: 150&quot;
[1] &quot;Algorithm: LMM-EXACT-INBRED&quot;
# A tibble: 16 x 5
   h2    nQTL   `500` `1000` grouping.diff
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;        
 1 0.2   1     0.370  0.328  -4.23%       
 2 0.2   5     0.339  0.261  -7.83%       
 3 0.2   10    0.113  0.0838 -2.96%       
 4 0.2   25    0.0703 0.0435 -2.68%       
 5 0.4   1     0.25   0.223  -2.68%       
 6 0.4   5     0.188  0.140  -4.78%       
 7 0.4   10    0.0729 0.0628 -1.01%       
 8 0.4   25    0.0468 0.0298 -1.7%        
 9 0.6   1     0.522  0.451  -7.09%       
10 0.6   5     0.285  0.235  -4.98%       
11 0.6   10    0.155  0.132  -2.35%       
12 0.6   25    0.0479 0.0242 -2.37%       
13 0.8   1     0.628  0.548  -8.04%       
14 0.8   5     0.385  0.278  -10.62%      
15 0.8   10    0.251  0.182  -6.89%       
16 0.8   25    0.0705 0.0551 -1.54%       
[1] &quot;CI Extension: 50&quot;
[1] &quot;Algorithm: LMM-EXACT-LOCO&quot;
# A tibble: 16 x 5
   h2    nQTL   `500` `1000` grouping.diff
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;        
 1 0.2   1     0.117  0.0879 -2.91%       
 2 0.2   5     0.107  0.0909 -1.62%       
 3 0.2   10    0.104  0.0714 -3.29%       
 4 0.2   25    0.0984 0.0667 -3.17%       
 5 0.4   1     0.458  0.388  -7.01%       
 6 0.4   5     0.271  0.217  -5.43%       
 7 0.4   10    0.147  0.0946 -5.26%       
 8 0.4   25    0.0945 0.0541 -4.05%       
 9 0.6   1     0.637  0.570  -6.78%       
10 0.6   5     0.364  0.285  -7.89%       
11 0.6   10    0.201  0.153  -4.8%        
12 0.6   25    0.0653 0.0369 -2.84%       
13 0.8   1     0.703  0.622  -8.14%       
14 0.8   5     0.426  0.314  -11.18%      
15 0.8   10    0.286  0.202  -8.42%       
16 0.8   25    0.0766 0.0479 -2.87%       
[1] &quot;CI Extension: 150&quot;
[1] &quot;Algorithm: LMM-EXACT-LOCO&quot;
# A tibble: 16 x 5
   h2    nQTL   `500` `1000` grouping.diff
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;        
 1 0.2   1     0.117  0.0879 -2.91%       
 2 0.2   5     0.0948 0.0789 -1.59%       
 3 0.2   10    0.0750 0.0513 -2.37%       
 4 0.2   25    0.0567 0.0288 -2.8%        
 5 0.4   1     0.458  0.388  -7.01%       
 6 0.4   5     0.255  0.202  -5.27%       
 7 0.4   10    0.125  0.0786 -4.64%       
 8 0.4   25    0.0630 0.0298 -3.32%       
 9 0.6   1     0.637  0.570  -6.78%       
10 0.6   5     0.336  0.257  -7.96%       
11 0.6   10    0.186  0.144  -4.22%       
12 0.6   25    0.0462 0.0261 -2.01%       
13 0.8   1     0.703  0.622  -8.14%       
14 0.8   5     0.409  0.300  -10.89%      
15 0.8   10    0.265  0.191  -7.44%       
16 0.8   25    0.0646 0.0399 -2.48%       
[1] &quot;CI Extension: 50&quot;
[1] &quot;Algorithm: MIXED&quot;
# A tibble: 16 x 5
   h2    nQTL   `500` `1000` grouping.diff
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;        
 1 0.2   1     0.401  0.349  -5.23%       
 2 0.2   5     0.362  0.279  -8.26%       
 3 0.2   10    0.153  0.113  -4.04%       
 4 0.2   25    0.102  0.0662 -3.6%        
 5 0.4   1     0.474  0.404  -6.97%       
 6 0.4   5     0.274  0.220  -5.36%       
 7 0.4   10    0.164  0.114  -5.03%       
 8 0.4   25    0.0942 0.0539 -4.03%       
 9 0.6   1     0.666  0.599  -6.64%       
10 0.6   5     0.379  0.302  -7.68%       
11 0.6   10    0.218  0.169  -4.85%       
12 0.6   25    0.0717 0.042  -2.97%       
13 0.8   1     0.729  0.646  -8.31%       
14 0.8   5     0.456  0.339  -11.67%      
15 0.8   10    0.319  0.220  -9.85%       
16 0.8   25    0.0968 0.0592 -3.76%       
[1] &quot;CI Extension: 150&quot;
[1] &quot;Algorithm: MIXED&quot;
# A tibble: 16 x 5
   h2    nQTL   `500` `1000` grouping.diff
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;        
 1 0.2   1     0.401  0.349  -5.23%       
 2 0.2   5     0.332  0.260  -7.17%       
 3 0.2   10    0.131  0.0957 -3.51%       
 4 0.2   25    0.0705 0.0408 -2.97%       
 5 0.4   1     0.474  0.404  -6.97%       
 6 0.4   5     0.257  0.205  -5.2%        
 7 0.4   10    0.137  0.0928 -4.43%       
 8 0.4   25    0.0625 0.0296 -3.29%       
 9 0.6   1     0.666  0.599  -6.64%       
10 0.6   5     0.352  0.275  -7.74%       
11 0.6   10    0.198  0.155  -4.29%       
12 0.6   25    0.0488 0.0273 -2.15%       
13 0.8   1     0.729  0.646  -8.31%       
14 0.8   5     0.431  0.316  -11.57%      
15 0.8   10    0.291  0.201  -9%          
16 0.8   25    0.0752 0.0449 -3.02%       </code></pre>
<pre><code>[1] &quot;Group Size: 500&quot;
[1] &quot;Algorithm: LMM-EXACT-INBRED&quot;
# A tibble: 16 x 5
   h2    nQTL    `50`  `150` CI.diff
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;  
 1 0.2   1     0.370  0.370  0%     
 2 0.2   5     0.367  0.339  -2.83% 
 3 0.2   10    0.131  0.113  -1.75% 
 4 0.2   25    0.0882 0.0703 -1.79% 
 5 0.4   1     0.259  0.25   -0.86% 
 6 0.4   5     0.204  0.188  -1.58% 
 7 0.4   10    0.0874 0.0729 -1.45% 
 8 0.4   25    0.0804 0.0468 -3.36% 
 9 0.6   1     0.522  0.522  0%     
10 0.6   5     0.299  0.285  -1.36% 
11 0.6   10    0.193  0.155  -3.75% 
12 0.6   25    0.0646 0.0479 -1.67% 
13 0.8   1     0.628  0.628  0%     
14 0.8   5     0.413  0.385  -2.83% 
15 0.8   10    0.282  0.251  -3.18% 
16 0.8   25    0.101  0.0705 -3.02% 
[1] &quot;Group Size: 1000&quot;
[1] &quot;Algorithm: LMM-EXACT-INBRED&quot;
# A tibble: 16 x 5
   h2    nQTL    `50`  `150` CI.diff
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;  
 1 0.2   1     0.328  0.328  0%     
 2 0.2   5     0.278  0.261  -1.72% 
 3 0.2   10    0.101  0.0838 -1.73% 
 4 0.2   25    0.0549 0.0435 -1.14% 
 5 0.4   1     0.232  0.223  -0.89% 
 6 0.4   5     0.156  0.140  -1.57% 
 7 0.4   10    0.0714 0.0628 -0.86% 
 8 0.4   25    0.0498 0.0298 -2%    
 9 0.6   1     0.451  0.451  0%     
10 0.6   5     0.249  0.235  -1.36% 
11 0.6   10    0.159  0.132  -2.74% 
12 0.6   25    0.0375 0.0242 -1.34% 
13 0.8   1     0.548  0.548  0%     
14 0.8   5     0.310  0.278  -3.16% 
15 0.8   10    0.207  0.182  -2.49% 
16 0.8   25    0.0760 0.0551 -2.09% 
[1] &quot;Group Size: 500&quot;
[1] &quot;Algorithm: LMM-EXACT-LOCO&quot;
# A tibble: 16 x 5
   h2    nQTL    `50`  `150` CI.diff
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;  
 1 0.2   1     0.117  0.117  0%     
 2 0.2   5     0.107  0.0948 -1.23% 
 3 0.2   10    0.104  0.0750 -2.93% 
 4 0.2   25    0.0984 0.0567 -4.16% 
 5 0.4   1     0.458  0.458  0%     
 6 0.4   5     0.271  0.255  -1.68% 
 7 0.4   10    0.147  0.125  -2.22% 
 8 0.4   25    0.0945 0.0630 -3.15% 
 9 0.6   1     0.637  0.637  0%     
10 0.6   5     0.364  0.336  -2.72% 
11 0.6   10    0.201  0.186  -1.42% 
12 0.6   25    0.0653 0.0462 -1.91% 
13 0.8   1     0.703  0.703  0%     
14 0.8   5     0.426  0.409  -1.7%  
15 0.8   10    0.286  0.265  -2.08% 
16 0.8   25    0.0766 0.0646 -1.19% 
[1] &quot;Group Size: 1000&quot;
[1] &quot;Algorithm: LMM-EXACT-LOCO&quot;
# A tibble: 16 x 5
   h2    nQTL    `50`  `150` CI.diff
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;  
 1 0.2   1     0.0879 0.0879 0%     
 2 0.2   5     0.0909 0.0789 -1.2%  
 3 0.2   10    0.0714 0.0513 -2.01% 
 4 0.2   25    0.0667 0.0288 -3.79% 
 5 0.4   1     0.388  0.388  0%     
 6 0.4   5     0.217  0.202  -1.52% 
 7 0.4   10    0.0946 0.0786 -1.6%  
 8 0.4   25    0.0541 0.0298 -2.42% 
 9 0.6   1     0.570  0.570  0%     
10 0.6   5     0.285  0.257  -2.79% 
11 0.6   10    0.153  0.144  -0.85% 
12 0.6   25    0.0369 0.0261 -1.08% 
13 0.8   1     0.622  0.622  0%     
14 0.8   5     0.314  0.300  -1.41% 
15 0.8   10    0.202  0.191  -1.1%  
16 0.8   25    0.0479 0.0399 -0.8%  
[1] &quot;Group Size: 500&quot;
[1] &quot;Algorithm: MIXED&quot;
# A tibble: 16 x 5
   h2    nQTL    `50`  `150` CI.diff
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;  
 1 0.2   1     0.401  0.401  0%     
 2 0.2   5     0.362  0.332  -3.01% 
 3 0.2   10    0.153  0.131  -2.23% 
 4 0.2   25    0.102  0.0705 -3.17% 
 5 0.4   1     0.474  0.474  0%     
 6 0.4   5     0.274  0.257  -1.67% 
 7 0.4   10    0.164  0.137  -2.68% 
 8 0.4   25    0.0942 0.0625 -3.17% 
 9 0.6   1     0.666  0.666  0%     
10 0.6   5     0.379  0.352  -2.65% 
11 0.6   10    0.218  0.198  -1.99% 
12 0.6   25    0.0717 0.0488 -2.29% 
13 0.8   1     0.729  0.729  0%     
14 0.8   5     0.456  0.431  -2.46% 
15 0.8   10    0.319  0.291  -2.78% 
16 0.8   25    0.0968 0.0752 -2.16% 
[1] &quot;Group Size: 1000&quot;
[1] &quot;Algorithm: MIXED&quot;
# A tibble: 16 x 5
   h2    nQTL    `50`  `150` CI.diff
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;  
 1 0.2   1     0.349  0.349  0%     
 2 0.2   5     0.279  0.260  -1.92% 
 3 0.2   10    0.113  0.0957 -1.71% 
 4 0.2   25    0.0662 0.0408 -2.54% 
 5 0.4   1     0.404  0.404  0%     
 6 0.4   5     0.220  0.205  -1.51% 
 7 0.4   10    0.114  0.0928 -2.07% 
 8 0.4   25    0.0539 0.0296 -2.43% 
 9 0.6   1     0.599  0.599  0%     
10 0.6   5     0.302  0.275  -2.72% 
11 0.6   10    0.169  0.155  -1.44% 
12 0.6   25    0.042  0.0273 -1.47% 
13 0.8   1     0.646  0.646  0%     
14 0.8   5     0.339  0.316  -2.36% 
15 0.8   10    0.220  0.201  -1.93% 
16 0.8   25    0.0592 0.0449 -1.42% </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.3 (2020-02-29)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] data.table_1.13.2        wesanderson_0.3.6.9000   viridis_0.5.1           
 [4] viridisLite_0.3.0        RColorBrewer_1.1-2       nationalparkcolors_0.1.0
 [7] yardstick_0.0.7          workflows_0.2.1          tune_0.1.1              
[10] rsample_0.0.8            recipes_0.1.13           parsnip_0.1.3           
[13] modeldata_0.0.2          infer_0.5.3              dials_0.0.9             
[16] scales_1.1.1             broom_0.7.2              tidymodels_0.1.1        
[19] forcats_0.5.0            stringr_1.4.0            dplyr_1.0.2             
[22] purrr_0.3.4              readr_1.4.0              tidyr_1.1.2             
[25] tibble_3.0.4             ggplot2_3.3.2            tidyverse_1.3.0         
[28] workflowr_1.6.2         

loaded via a namespace (and not attached):
 [1] colorspace_1.4-1   ellipsis_0.3.1     class_7.3-17       rprojroot_1.3-2   
 [5] fs_1.5.0           rstudioapi_0.11    farver_2.0.3       listenv_0.8.0     
 [9] furrr_0.1.0        prodlim_2019.11.13 fansi_0.4.1        lubridate_1.7.9   
[13] xml2_1.3.2         codetools_0.2-16   splines_3.6.3      knitr_1.30        
[17] jsonlite_1.7.1     pROC_1.16.2        dbplyr_1.4.4       compiler_3.6.3    
[21] httr_1.4.2         backports_1.1.10   assertthat_0.2.1   Matrix_1.2-18     
[25] cli_2.1.0          later_1.1.0.1      htmltools_0.5.0    tools_3.6.3       
[29] gtable_0.3.0       glue_1.4.2         Rcpp_1.0.6         cellranger_1.1.0  
[33] DiceDesign_1.8-1   vctrs_0.3.4        iterators_1.0.12   timeDate_3043.102 
[37] gower_0.2.2        xfun_0.18          globals_0.13.1     rvest_0.3.6       
[41] lifecycle_0.2.0    future_1.19.1      MASS_7.3-53        ipred_0.9-9       
[45] hms_0.5.3          promises_1.1.1     parallel_3.6.3     yaml_2.2.1        
[49] gridExtra_2.3      rpart_4.1-15       stringi_1.5.3      highr_0.8         
[53] foreach_1.5.0      lhs_1.1.1          lava_1.6.8         rlang_0.4.10      
[57] pkgconfig_2.0.3    evaluate_0.14      lattice_0.20-41    labeling_0.4.2    
[61] cowplot_1.1.0      tidyselect_1.1.0   plyr_1.8.6         magrittr_2.0.1    
[65] R6_2.4.1           generics_0.0.2     DBI_1.1.0          pillar_1.4.6      
[69] haven_2.3.1        whisker_0.4        withr_2.3.0        survival_3.2-7    
[73] nnet_7.3-14        modelr_0.1.8       crayon_1.3.4       utf8_1.1.4        
[77] rmarkdown_2.5      grid_3.6.3         readxl_1.3.1       blob_1.2.1        
[81] git2r_0.27.1       reprex_0.3.0       digest_0.6.27      httpuv_1.5.4      
[85] munsell_0.5.0      GPfit_1.0-8       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
