<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Polygenic Architectures are Difficult to Resolve Using GWAS</title>

<script src="site_libs/header-attrs-2.5/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">NemaScan: GWAS in C. elegans Wild Isolates</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sam-widmayer/NemaScan_Performance">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Polygenic Architectures are Difficult to Resolve Using GWAS</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-01-05
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>NemaScan_Performance/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200825code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200825)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200825code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200825)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
~/Documents/projects/NemaScan_Performance/
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsamwidmayerNemaScanPerformancetreedd4da2f9f0590f77e1f45620f9bbe13c22636f37targetblankdd4da2fa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sam-widmayer/NemaScan_Performance/tree/dd4da2f9f0590f77e1f45620f9bbe13c22636f37" target="_blank">dd4da2f</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsamwidmayerNemaScanPerformancetreedd4da2f9f0590f77e1f45620f9bbe13c22636f37targetblankdd4da2fa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sam-widmayer/NemaScan_Performance/tree/dd4da2f9f0590f77e1f45620f9bbe13c22636f37" target="_blank">dd4da2f</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/.Rhistory
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  code/2014_toxinphenotypes.R
    Untracked:  code/Aggregate_Mappings.R
    Untracked:  code/Find_Aggregate_Intervals.R
    Untracked:  code/Find_GCTA_Intervals.R
    Untracked:  code/pipeline.plotting.R
    Untracked:  code/startup.R
    Untracked:  code/susie.R
    Untracked:  data/NemaScan_Performance.Algorithm.Sims.20201024.RData
    Untracked:  data/NemaScan_Performance.Algorithm.Sims.20201110_EMMA.RData
    Untracked:  data/NemaScan_Performance.Algorithm.Sims.v.2.20201104.RData
    Untracked:  data/NemaScan_Performance.Architecture.20201120.RData
    Untracked:  data/NemaScan_Performance.Architecture.CI50.20201130.RData
    Untracked:  data/NemaScan_Performance.Architecture.CI50.group500.20201201.RData
    Untracked:  data/NemaScan_Performance.Architecture.Mixed.20201205.RData
    Untracked:  data/NemaScan_Performance.Architecture.Mixed.20201209.RData
    Untracked:  data/NemaScan_Performance.Architecture.Mixed.20201223.RData
    Untracked:  data/NemaScan_Performance.Architecture.Mixed.CI.50.20201224.RData
    Untracked:  data/NemaScan_Performance.Architecture.Mixed.group500.20201223.RData
    Untracked:  data/NemaScan_Performance.Architecture.Mixed.group500.CI.50.20201224.RData
    Untracked:  data/NemaScan_Performance.Architecture.group500.20201124.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_GAMMA_EMMA_hardfiltered.20201020.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_PowerPrecision.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_PowerPrecision_GAMMA.20200909.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_PowerPrecision_GAMMA.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_PowerPrecision_GAMMA_EMMA.20200924.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_PowerPrecision_REDO.20200909.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_cendr_sets.20200929.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_population_subsampling.20200925.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_population_subsampling.20201018.RData
    Untracked:  data/NemaScan_Performance.CeNDR2020_population_subsampling.20201019.RData
    Untracked:  data/NemaScan_Performance.Demography.SampleSize.Complete.Subsampled.20201230.RData
    Untracked:  data/NemaScan_Performance.Variant.Set.HF.chr125.20201112.RData
    Untracked:  data/NemaScan_Performance.Variant.Set.Hard.Filtered.20201111.RData
    Untracked:  data/NemaScan_Performance.Variant.Set.Imputed.20201111.RData
    Untracked:  data/NemaScan_Performance.Variant.Set.Imputed.NonDivergent.20201112.RData
    Untracked:  data/NemaScan_Performance.Variant.Set.Imputed.chr125.20201112.RData
    Untracked:  data/NemaScan_Population_Metrics.Demography.SampleSize.Complete.Subsampled.20201229.RData
    Untracked:  data/TO_SIMS.bim
    Untracked:  data/all_susie_BF_hits.tsv
    Untracked:  data/all_susie_hits.tsv
    Untracked:  data/causal.variants.sim.100.txt
    Untracked:  data/causal.variants.sim.5.txt
    Untracked:  data/hard.filtered.snps.bim
    Untracked:  data/imputed.snps.bim
    Untracked:  output/2014_toxin_phenos.tsv
    Untracked:  output/complete.subsampled.strains.txt
    Untracked:  output/n.divergent.variants.in.variant.set.png
    Untracked:  output/perc.divergent.variants.in.variant.set.png
    Untracked:  output/pr.curve.variant.set.png
    Untracked:  output/prelim.fig.1.png
    Untracked:  output/prelim.fig.2.png
    Untracked:  output/subsampled.strains_20201225_n144_reps200.txt
    Untracked:  output/subsampled.strains_20201225_n192_reps200.txt
    Untracked:  output/subsampled.strains_20201225_n240_reps200.txt
    Untracked:  output/subsampled.strains_20201225_n288_reps200.txt
    Untracked:  output/subsampled.strains_20201225_n336_reps200.txt
    Untracked:  output/subsampled.strains_20201225_n384_reps200.txt
    Untracked:  output/subsampled.strains_20201225_n48_reps200.txt
    Untracked:  output/subsampled.strains_20201225_n96_reps200.txt
    Untracked:  output/subsampled.swept.strains_20201225_n120_reps200.txt
    Untracked:  output/subsampled.swept.strains_20201225_n144_reps200.txt
    Untracked:  output/subsampled.swept.strains_20201225_n168_reps200.txt
    Untracked:  output/subsampled.swept.strains_20201225_n24_reps200.txt
    Untracked:  output/subsampled.swept.strains_20201225_n48_reps200.txt
    Untracked:  output/subsampled.swept.strains_20201225_n72_reps200.txt
    Untracked:  output/subsampled.swept.strains_20201225_n96_reps200.txt
    Untracked:  output/subsampled.unswept.strains_20201225_n120_reps200.txt
    Untracked:  output/subsampled.unswept.strains_20201225_n144_reps200.txt
    Untracked:  output/subsampled.unswept.strains_20201225_n168_reps200.txt
    Untracked:  output/subsampled.unswept.strains_20201225_n192_reps200.txt
    Untracked:  output/subsampled.unswept.strains_20201225_n216_reps200.txt
    Untracked:  output/subsampled.unswept.strains_20201225_n24_reps200.txt
    Untracked:  output/subsampled.unswept.strains_20201225_n48_reps200.txt
    Untracked:  output/subsampled.unswept.strains_20201225_n72_reps200.txt
    Untracked:  output/subsampled.unswept.strains_20201225_n96_reps200.txt
    Untracked:  output/susie.validation.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/architecture.Rmd</code>) and HTML (<code>docs/architecture.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/dd4da2f9f0590f77e1f45620f9bbe13c22636f37/analysis/architecture.Rmd" target="_blank">dd4da2f</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-05
</td>
<td>
fix group size tables
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/d1dc9d8308579615bd1d1724e2cc6061faf081ef/docs/architecture.html" target="_blank">d1dc9d8</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/abcb1dd41ab626806f7e5211113e683f9e362482/analysis/architecture.Rmd" target="_blank">abcb1dd</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-04
</td>
<td>
reintegrate tables for grouping and CI
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/8688742d60cc957d9bcb2a93a2fbdd004a2de1dd/docs/architecture.html" target="_blank">8688742</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/66d05e54c3e33000cca7b56d35bd9530c4adc201/analysis/architecture.Rmd" target="_blank">66d05e5</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-04
</td>
<td>
fix fig2A jitter
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/38e8395da3f74c32635e01408170487cf5f63705/analysis/architecture.Rmd" target="_blank">38e8395</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-04
</td>
<td>
wflow_git_commit(files = “analysis/architecture.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/842ad5b04e027ee220b87f6560a87d0dac7270eb/docs/architecture.html" target="_blank">842ad5b</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/b4ff99dafccff5bc379162295813d939e4af372a/analysis/architecture.Rmd" target="_blank">b4ff99d</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-04
</td>
<td>
sd for FDR and power
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/437068532ecf3e9d049ab85bb318ded5fa743fa8/docs/architecture.html" target="_blank">4370685</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/7c6f1265bd9c9e3712b00d0b057937c70ebe588a/analysis/architecture.Rmd" target="_blank">7c6f126</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
<td>
simplify plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/777ef170737613e948f5e7edd003807b67307e08/docs/architecture.html" target="_blank">777ef17</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/2bf68e35243817816f287a6247e2134214726fa6/analysis/architecture.Rmd" target="_blank">2bf68e3</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
<td>
wflow_publish(“analysis/architecture.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/2423b37efaae901b653c973c856d1cc6b2aeedf0/docs/architecture.html" target="_blank">2423b37</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/ed561fc1f5d15e6439afd08e3e2a35a1df1907c2/analysis/architecture.Rmd" target="_blank">ed561fc</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
<td>
adjust pr curve for main result
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/a0cbc46ebdc3711526fe36076abf87b1152f11cc/analysis/architecture.Rmd" target="_blank">a0cbc46</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
<td>
adjust pr curve for main result
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/03660cc13e021b39173f15faf11594a8d9bcdcd3/docs/architecture.html" target="_blank">03660cc</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/8a6a1eb1f341774473a1dee4217c29dbb04967e7/analysis/architecture.Rmd" target="_blank">8a6a1eb</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
<td>
comment out supplementary tables
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/5ad923569f9055ce3d127bbe2dd40f74507ea07d/analysis/architecture.Rmd" target="_blank">5ad9235</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
<td>
update with new aggregated runs, group sizes, and CIs
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/77ecc62128790056a493e281ea421fb10d5ca90b/analysis/architecture.Rmd" target="_blank">77ecc62</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-12-18
</td>
<td>
xmas push
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/f5961126a1ae6bb94da128b2ab9367093cad10c8/docs/architecture.html" target="_blank">f596112</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-12-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/de6baedf907967b14303c5608041f159480ffc59/analysis/architecture.Rmd" target="_blank">de6baed</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-12-02
</td>
<td>
add FPR evaluation for CIs and groupings
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/c69064b8e481f2c1e9ed48f16e0ffca40e1ffeab/docs/architecture.html" target="_blank">c69064b</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-12-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/68829fe7ff9d1f8846df2edf35882f6091f73db2/analysis/architecture.Rmd" target="_blank">68829fe</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-12-01
</td>
<td>
integrate multiple CI and grouping params
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/c39b2011fae9980784fa5eae10c5ae4344ac1c68/analysis/architecture.Rmd" target="_blank">c39b201</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-12-01
</td>
<td>
integrate multiple CI and grouping params
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/f24b92958c68ffbab3e8afa43b345fa2aab9a69c/analysis/architecture.Rmd" target="_blank">f24b929</a>
</td>
<td>
Sam
</td>
<td>
2020-11-30
</td>
<td>
turkey day changes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/fcfac6a4c18bc150a0a338a12ee612b61b6514b4/docs/architecture.html" target="_blank">fcfac6a</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/385b5cbb02ca682a4e03d95e43e97ee7f291f4d9/analysis/architecture.Rmd" target="_blank">385b5cb</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
<td>
prelim figures
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/a07596f57a8cbd1751a9031ed99607b0312adcb9/docs/architecture.html" target="_blank">a07596f</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/b6ccbbf8f25af40a7fdab34e64d867b1af20f9ac/analysis/architecture.Rmd" target="_blank">b6ccbbf</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
<td>
silence summarise stuff
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/9114b11f64c5131678476846985b2cc2298bb004/analysis/architecture.Rmd" target="_blank">9114b11</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
<td>
update title
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/d4f156386b38dfc0d9ed09fc4d471ad95ba7d0d0/docs/architecture.html" target="_blank">d4f1563</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/30ce6a242ca70ce803a061aab07cad23693c8257/analysis/architecture.Rmd" target="_blank">30ce6a2</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
<td>
update plots
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="analysis-date-january-05-2021" class="section level3">
<h3>Analysis date: January 05, 2021</h3>
</div>
<div id="nemascan-simulation-performance-was-assessed-with-the-following-experimental-parameters" class="section level3">
<h3>NemaScan simulation performance was assessed with the following experimental parameters:</h3>
<ul>
<li><h6 id="number-of-simulated-qtl-1-5-10-25">Number of Simulated QTL: <code>1, 5, 10, 25</code></h6></li>
<li><h6 id="sample-populations-complete">Sample Population(s): <code>complete</code></h6></li>
<li><h6 id="heritabilityies-0.2-0.4-0.6-0.8">Heritability(ies): <code>0.2, 0.4, 0.6, 0.8</code></h6></li>
<li><h6 id="mafs-0.05">MAF(s): <code>0.05</code></h6></li>
<li><h6 id="number-of-replicates-per-regime-100">Number of Replicates per Regime: <code>100</code></h6></li>
<li><h6 id="qtl-effect-range-gamma">QTL Effect Range: <code>gamma</code></h6></li>
</ul>
<p><img src="figure/architecture.Rmd/simulated%20QTL%20locations-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-simulated-QTL-locations-1">
Past versions of “simulated QTL locations-1.png”
</button>
</p>
<div id="fig-no-spaces-simulated-QTL-locations-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/03660cc13e021b39173f15faf11594a8d9bcdcd3/docs/figure/architecture.Rmd/simulated QTL locations-1.png" target="_blank">03660cc</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/c69064b8e481f2c1e9ed48f16e0ffca40e1ffeab/docs/figure/architecture.Rmd/simulated QTL locations-1.png" target="_blank">c69064b</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-12-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/d4f156386b38dfc0d9ed09fc4d471ad95ba7d0d0/docs/figure/architecture.Rmd/simulated QTL locations-1.png" target="_blank">d4f1563</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/architecture.Rmd/simulated%20QTL%20locations-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-simulated-QTL-locations-2">
Past versions of “simulated QTL locations-2.png”
</button>
</p>
<div id="fig-no-spaces-simulated-QTL-locations-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/03660cc13e021b39173f15faf11594a8d9bcdcd3/docs/figure/architecture.Rmd/simulated QTL locations-2.png" target="_blank">03660cc</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/c69064b8e481f2c1e9ed48f16e0ffca40e1ffeab/docs/figure/architecture.Rmd/simulated QTL locations-2.png" target="_blank">c69064b</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-12-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/d4f156386b38dfc0d9ed09fc4d471ad95ba7d0d0/docs/figure/architecture.Rmd/simulated QTL locations-2.png" target="_blank">d4f1563</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/architecture.Rmd/simulated%20QTL%20locations-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-simulated-QTL-locations-3">
Past versions of “simulated QTL locations-3.png”
</button>
</p>
<div id="fig-no-spaces-simulated-QTL-locations-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/03660cc13e021b39173f15faf11594a8d9bcdcd3/docs/figure/architecture.Rmd/simulated QTL locations-3.png" target="_blank">03660cc</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/c69064b8e481f2c1e9ed48f16e0ffca40e1ffeab/docs/figure/architecture.Rmd/simulated QTL locations-3.png" target="_blank">c69064b</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-12-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/d4f156386b38dfc0d9ed09fc4d471ad95ba7d0d0/docs/figure/architecture.Rmd/simulated QTL locations-3.png" target="_blank">d4f1563</a>
</td>
<td>
Sam
</td>
<td>
2020-11-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/architecture.Rmd/default%20settings%20PR%20curve-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-default-settings-PR-curve-1">
Past versions of “default settings PR curve-1.png”
</button>
</p>
<div id="fig-no-spaces-default-settings-PR-curve-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/842ad5b04e027ee220b87f6560a87d0dac7270eb/docs/figure/architecture.Rmd/default settings PR curve-1.png" target="_blank">842ad5b</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Joining, by = c(&quot;h2&quot;, &quot;algorithm&quot;, &quot;nQTL&quot;, &quot;group.size&quot;, &quot;CI&quot;, &quot;.metric&quot;, &quot;.estimator&quot;, &quot;.estimate&quot;)
Joining, by = c(&quot;h2&quot;, &quot;algorithm&quot;, &quot;nQTL&quot;, &quot;group.size&quot;, &quot;CI&quot;, &quot;.metric&quot;, &quot;.estimator&quot;, &quot;.estimate&quot;)</code></pre>
<pre><code>Joining, by = c(&quot;h2&quot;, &quot;algorithm&quot;, &quot;nQTL&quot;, &quot;group.size&quot;, &quot;CI&quot;)</code></pre>
<pre><code>Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 1 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 1 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 1 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 1 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 1 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 1 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 1 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 1 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 1 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 1 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 1 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 1 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 1 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 1 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 1 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.</code></pre>
<pre><code>Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 5 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 5 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 5 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 5 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 5 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 5 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.</code></pre>
<pre><code>Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 4 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.</code></pre>
<pre><code>Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 5 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 5 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 5 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 5 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 5 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 5 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 5 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 5 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 5 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 5 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 5 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 5 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.</code></pre>
<pre><code>Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 4 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 4 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.</code></pre>
<pre><code>Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 5 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.</code></pre>
<pre><code>Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 10 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 10 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 10 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 10 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 10 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 10 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 10 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 10 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 10 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 10 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 10 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 10 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 10 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 10 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 10 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 10 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 10 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 10 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 10 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 10 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.</code></pre>
<pre><code>Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 25 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 25 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 25 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 25 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 25 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 25 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 25 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 25 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 25 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 25 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 25 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 25 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 25 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 25 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.</code></pre>
<pre><code>Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 24 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.</code></pre>
<pre><code>Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 25 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 25 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 25 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 25 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 25 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 25 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 25 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 25 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 25 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 25 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 25 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.</code></pre>
<pre><code>Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 24 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.</code></pre>
<pre><code>Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 25 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 25 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.</code></pre>
<pre><code>Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 1 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 1 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 1 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.</code></pre>
<pre><code>Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 5 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 5 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 5 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.</code></pre>
<pre><code>Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 10 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 10 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 10 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.</code></pre>
<pre><code>Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 25 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.

Warning: While computing binary `precision()`, no predicted events were detected (i.e. `true_positive + false_positive = 0`). 
Precision is undefined in this case, and `NA` will be returned.
Note that 25 true event(s) actually occured for the problematic event level, &#39;TRUE&#39;.</code></pre>
<p><img src="figure/architecture.Rmd/plots-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plots-1">
Past versions of plots-1.png
</button>
</p>
<div id="fig-plots-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/8688742d60cc957d9bcb2a93a2fbdd004a2de1dd/docs/figure/architecture.Rmd/plots-1.png" target="_blank">8688742</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/842ad5b04e027ee220b87f6560a87d0dac7270eb/docs/figure/architecture.Rmd/plots-1.png" target="_blank">842ad5b</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2021-01-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/437068532ecf3e9d049ab85bb318ded5fa743fa8/docs/figure/architecture.Rmd/plots-1.png" target="_blank">4370685</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/2423b37efaae901b653c973c856d1cc6b2aeedf0/docs/figure/architecture.Rmd/plots-1.png" target="_blank">2423b37</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/03660cc13e021b39173f15faf11594a8d9bcdcd3/docs/figure/architecture.Rmd/plots-1.png" target="_blank">03660cc</a>
</td>
<td>
Sam
</td>
<td>
2020-12-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="tables" class="section level2">
<h2>TABLES</h2>
<pre><code>[1] &quot;POWER EVALUATION&quot;</code></pre>
<pre><code>[1] &quot;Group Size: 500&quot;
[1] &quot;CI Extension: 50&quot;
# A tibble: 16 x 5
   h2    nQTL  `LMM-EXACT-INBRED` `LMM-EXACT-LOCO`  MIXED
   &lt;fct&gt; &lt;fct&gt;              &lt;dbl&gt;            &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1                 0.79             0.82   0.81  
 2 0.2   5                 0.213            0.191  0.227 
 3 0.2   10                0.146            0.101  0.159 
 4 0.2   25                0.0746           0.0377 0.0851
 5 0.4   1                 0.86             0.96   0.96  
 6 0.4   5                 0.264            0.312  0.312 
 7 0.4   10                0.153            0.203  0.204 
 8 0.4   25                0.0750           0.113  0.114 
 9 0.6   1                 0.89             1      1     
10 0.6   5                 0.294            0.398  0.398 
11 0.6   10                0.200            0.288  0.289 
12 0.6   25                0.110            0.174  0.175 
13 0.8   1                 0.94             1      1     
14 0.8   5                 0.332            0.449  0.449 
15 0.8   10                0.249            0.380  0.383 
16 0.8   25                0.146            0.255  0.257 
[1] &quot;Group Size: 500&quot;
[1] &quot;CI Extension: 150&quot;
# A tibble: 16 x 5
   h2    nQTL  `LMM-EXACT-INBRED` `LMM-EXACT-LOCO`  MIXED
   &lt;fct&gt; &lt;fct&gt;              &lt;dbl&gt;            &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1                 0.79             0.82   0.81  
 2 0.2   5                 0.231            0.209  0.249 
 3 0.2   10                0.159            0.111  0.174 
 4 0.2   25                0.0811           0.0449 0.0931
 5 0.4   1                 0.86             0.96   0.96  
 6 0.4   5                 0.282            0.336  0.336 
 7 0.4   10                0.169            0.221  0.222 
 8 0.4   25                0.0903           0.129  0.132 
 9 0.6   1                 0.89             1      1     
10 0.6   5                 0.318            0.419  0.419 
11 0.6   10                0.219            0.304  0.305 
12 0.6   25                0.126            0.199  0.201 
13 0.8   1                 0.94             1      1     
14 0.8   5                 0.354            0.477  0.479 
15 0.8   10                0.270            0.402  0.406 
16 0.8   25                0.171            0.287  0.291 
[1] &quot;Group Size: 1000&quot;
[1] &quot;CI Extension: 50&quot;
# A tibble: 16 x 5
   h2    nQTL  `LMM-EXACT-INBRED` `LMM-EXACT-LOCO`  MIXED
   &lt;fct&gt; &lt;fct&gt;              &lt;dbl&gt;            &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1                 0.79             0.82   0.81  
 2 0.2   5                 0.225            0.193  0.239 
 3 0.2   10                0.152            0.102  0.166 
 4 0.2   25                0.0758           0.0381 0.0867
 5 0.4   1                 0.86             0.96   0.96  
 6 0.4   5                 0.270            0.316  0.318 
 7 0.4   10                0.155            0.205  0.206 
 8 0.4   25                0.0762           0.116  0.117 
 9 0.6   1                 0.89             1      1     
10 0.6   5                 0.300            0.408  0.408 
11 0.6   10                0.201            0.292  0.293 
12 0.6   25                0.111            0.180  0.181 
13 0.8   1                 0.94             1      1     
14 0.8   5                 0.340            0.457  0.457 
15 0.8   10                0.252            0.384  0.388 
16 0.8   25                0.149            0.262  0.265 
[1] &quot;Group Size: 1000&quot;
[1] &quot;CI Extension: 150&quot;
# A tibble: 16 x 5
   h2    nQTL  `LMM-EXACT-INBRED` `LMM-EXACT-LOCO`  MIXED
   &lt;fct&gt; &lt;fct&gt;              &lt;dbl&gt;            &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1                 0.79             0.82   0.81  
 2 0.2   5                 0.239            0.209  0.258 
 3 0.2   10                0.163            0.111  0.178 
 4 0.2   25                0.0823           0.0453 0.0947
 5 0.4   1                 0.86             0.96   0.96  
 6 0.4   5                 0.286            0.338  0.340 
 7 0.4   10                0.170            0.222  0.223 
 8 0.4   25                0.0911           0.132  0.134 
 9 0.6   1                 0.89             1      1     
10 0.6   5                 0.322            0.425  0.425 
11 0.6   10                0.220            0.306  0.307 
12 0.6   25                0.127            0.204  0.206 
13 0.8   1                 0.94             1      1     
14 0.8   5                 0.360            0.483  0.485 
15 0.8   10                0.272            0.405  0.410 
16 0.8   25                0.173            0.292  0.296 </code></pre>
<pre><code>[1] &quot;CI Extension: 50&quot;
[1] &quot;Algorithm: LMM-EXACT-INBRED&quot;
# A tibble: 16 x 4
   h2    nQTL   `500` `1000`
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1     0.79   0.79  
 2 0.4   1     0.86   0.86  
 3 0.6   1     0.89   0.89  
 4 0.8   1     0.94   0.94  
 5 0.2   5     0.213  0.225 
 6 0.4   5     0.264  0.270 
 7 0.6   5     0.294  0.300 
 8 0.8   5     0.332  0.340 
 9 0.2   10    0.146  0.152 
10 0.4   10    0.153  0.155 
11 0.6   10    0.200  0.201 
12 0.8   10    0.249  0.252 
13 0.2   25    0.0746 0.0758
14 0.4   25    0.0750 0.0762
15 0.6   25    0.110  0.111 
16 0.8   25    0.146  0.149 
[1] &quot;CI Extension: 150&quot;
[1] &quot;Algorithm: LMM-EXACT-INBRED&quot;
# A tibble: 16 x 4
   h2    nQTL   `500` `1000`
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1     0.79   0.79  
 2 0.4   1     0.86   0.86  
 3 0.6   1     0.89   0.89  
 4 0.8   1     0.94   0.94  
 5 0.2   5     0.231  0.239 
 6 0.4   5     0.282  0.286 
 7 0.6   5     0.318  0.322 
 8 0.8   5     0.354  0.360 
 9 0.2   10    0.159  0.163 
10 0.4   10    0.169  0.170 
11 0.6   10    0.219  0.220 
12 0.8   10    0.270  0.272 
13 0.2   25    0.0811 0.0823
14 0.4   25    0.0903 0.0911
15 0.6   25    0.126  0.127 
16 0.8   25    0.171  0.173 
[1] &quot;CI Extension: 50&quot;
[1] &quot;Algorithm: LMM-EXACT-LOCO&quot;
# A tibble: 16 x 4
   h2    nQTL   `500` `1000`
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1     0.82   0.82  
 2 0.4   1     0.96   0.96  
 3 0.6   1     1      1     
 4 0.8   1     1      1     
 5 0.2   5     0.191  0.193 
 6 0.4   5     0.312  0.316 
 7 0.6   5     0.398  0.408 
 8 0.8   5     0.449  0.457 
 9 0.2   10    0.101  0.102 
10 0.4   10    0.203  0.205 
11 0.6   10    0.288  0.292 
12 0.8   10    0.380  0.384 
13 0.2   25    0.0377 0.0381
14 0.4   25    0.113  0.116 
15 0.6   25    0.174  0.180 
16 0.8   25    0.255  0.262 
[1] &quot;CI Extension: 150&quot;
[1] &quot;Algorithm: LMM-EXACT-LOCO&quot;
# A tibble: 16 x 4
   h2    nQTL   `500` `1000`
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1     0.82   0.82  
 2 0.4   1     0.96   0.96  
 3 0.6   1     1      1     
 4 0.8   1     1      1     
 5 0.2   5     0.209  0.209 
 6 0.4   5     0.336  0.338 
 7 0.6   5     0.419  0.425 
 8 0.8   5     0.477  0.483 
 9 0.2   10    0.111  0.111 
10 0.4   10    0.221  0.222 
11 0.6   10    0.304  0.306 
12 0.8   10    0.402  0.405 
13 0.2   25    0.0449 0.0453
14 0.4   25    0.129  0.132 
15 0.6   25    0.199  0.204 
16 0.8   25    0.287  0.292 
[1] &quot;CI Extension: 50&quot;
[1] &quot;Algorithm: MIXED&quot;
# A tibble: 16 x 4
   h2    nQTL   `500` `1000`
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1     0.81   0.81  
 2 0.4   1     0.96   0.96  
 3 0.6   1     1      1     
 4 0.8   1     1      1     
 5 0.2   5     0.227  0.239 
 6 0.4   5     0.312  0.318 
 7 0.6   5     0.398  0.408 
 8 0.8   5     0.449  0.457 
 9 0.2   10    0.159  0.166 
10 0.4   10    0.204  0.206 
11 0.6   10    0.289  0.293 
12 0.8   10    0.383  0.388 
13 0.2   25    0.0851 0.0867
14 0.4   25    0.114  0.117 
15 0.6   25    0.175  0.181 
16 0.8   25    0.257  0.265 
[1] &quot;CI Extension: 150&quot;
[1] &quot;Algorithm: MIXED&quot;
# A tibble: 16 x 4
   h2    nQTL   `500` `1000`
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1     0.81   0.81  
 2 0.4   1     0.96   0.96  
 3 0.6   1     1      1     
 4 0.8   1     1      1     
 5 0.2   5     0.249  0.258 
 6 0.4   5     0.336  0.340 
 7 0.6   5     0.419  0.425 
 8 0.8   5     0.479  0.485 
 9 0.2   10    0.174  0.178 
10 0.4   10    0.222  0.223 
11 0.6   10    0.305  0.307 
12 0.8   10    0.406  0.410 
13 0.2   25    0.0931 0.0947
14 0.4   25    0.132  0.134 
15 0.6   25    0.201  0.206 
16 0.8   25    0.291  0.296 </code></pre>
<pre><code>[1] &quot;Group Size: 500&quot;
[1] &quot;Algorithm: LMM-EXACT-INBRED&quot;
# A tibble: 16 x 4
   h2    nQTL    `50`  `150`
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1     0.79   0.79  
 2 0.4   1     0.86   0.86  
 3 0.6   1     0.89   0.89  
 4 0.8   1     0.94   0.94  
 5 0.2   5     0.213  0.231 
 6 0.4   5     0.264  0.282 
 7 0.6   5     0.294  0.318 
 8 0.8   5     0.332  0.354 
 9 0.2   10    0.146  0.159 
10 0.4   10    0.153  0.169 
11 0.6   10    0.200  0.219 
12 0.8   10    0.249  0.270 
13 0.2   25    0.0746 0.0811
14 0.4   25    0.0750 0.0903
15 0.6   25    0.110  0.126 
16 0.8   25    0.146  0.171 
[1] &quot;Group Size: 1000&quot;
[1] &quot;Algorithm: LMM-EXACT-INBRED&quot;
# A tibble: 16 x 4
   h2    nQTL    `50`  `150`
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1     0.79   0.79  
 2 0.4   1     0.86   0.86  
 3 0.6   1     0.89   0.89  
 4 0.8   1     0.94   0.94  
 5 0.2   5     0.225  0.239 
 6 0.4   5     0.270  0.286 
 7 0.6   5     0.300  0.322 
 8 0.8   5     0.340  0.360 
 9 0.2   10    0.152  0.163 
10 0.4   10    0.155  0.170 
11 0.6   10    0.201  0.220 
12 0.8   10    0.252  0.272 
13 0.2   25    0.0758 0.0823
14 0.4   25    0.0762 0.0911
15 0.6   25    0.111  0.127 
16 0.8   25    0.149  0.173 
[1] &quot;Group Size: 500&quot;
[1] &quot;Algorithm: LMM-EXACT-LOCO&quot;
# A tibble: 16 x 4
   h2    nQTL    `50`  `150`
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1     0.82   0.82  
 2 0.4   1     0.96   0.96  
 3 0.6   1     1      1     
 4 0.8   1     1      1     
 5 0.2   5     0.191  0.209 
 6 0.4   5     0.312  0.336 
 7 0.6   5     0.398  0.419 
 8 0.8   5     0.449  0.477 
 9 0.2   10    0.101  0.111 
10 0.4   10    0.203  0.221 
11 0.6   10    0.288  0.304 
12 0.8   10    0.380  0.402 
13 0.2   25    0.0377 0.0449
14 0.4   25    0.113  0.129 
15 0.6   25    0.174  0.199 
16 0.8   25    0.255  0.287 
[1] &quot;Group Size: 1000&quot;
[1] &quot;Algorithm: LMM-EXACT-LOCO&quot;
# A tibble: 16 x 4
   h2    nQTL    `50`  `150`
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1     0.82   0.82  
 2 0.4   1     0.96   0.96  
 3 0.6   1     1      1     
 4 0.8   1     1      1     
 5 0.2   5     0.193  0.209 
 6 0.4   5     0.316  0.338 
 7 0.6   5     0.408  0.425 
 8 0.8   5     0.457  0.483 
 9 0.2   10    0.102  0.111 
10 0.4   10    0.205  0.222 
11 0.6   10    0.292  0.306 
12 0.8   10    0.384  0.405 
13 0.2   25    0.0381 0.0453
14 0.4   25    0.116  0.132 
15 0.6   25    0.180  0.204 
16 0.8   25    0.262  0.292 
[1] &quot;Group Size: 500&quot;
[1] &quot;Algorithm: MIXED&quot;
# A tibble: 16 x 4
   h2    nQTL    `50`  `150`
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1     0.81   0.81  
 2 0.4   1     0.96   0.96  
 3 0.6   1     1      1     
 4 0.8   1     1      1     
 5 0.2   5     0.227  0.249 
 6 0.4   5     0.312  0.336 
 7 0.6   5     0.398  0.419 
 8 0.8   5     0.449  0.479 
 9 0.2   10    0.159  0.174 
10 0.4   10    0.204  0.222 
11 0.6   10    0.289  0.305 
12 0.8   10    0.383  0.406 
13 0.2   25    0.0851 0.0931
14 0.4   25    0.114  0.132 
15 0.6   25    0.175  0.201 
16 0.8   25    0.257  0.291 
[1] &quot;Group Size: 1000&quot;
[1] &quot;Algorithm: MIXED&quot;
# A tibble: 16 x 4
   h2    nQTL    `50`  `150`
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1     0.81   0.81  
 2 0.4   1     0.96   0.96  
 3 0.6   1     1      1     
 4 0.8   1     1      1     
 5 0.2   5     0.239  0.258 
 6 0.4   5     0.318  0.340 
 7 0.6   5     0.408  0.425 
 8 0.8   5     0.457  0.485 
 9 0.2   10    0.166  0.178 
10 0.4   10    0.206  0.223 
11 0.6   10    0.293  0.307 
12 0.8   10    0.388  0.410 
13 0.2   25    0.0867 0.0947
14 0.4   25    0.117  0.134 
15 0.6   25    0.181  0.206 
16 0.8   25    0.265  0.296 </code></pre>
<pre><code>[1] &quot;FDR EVALUATION&quot;</code></pre>
<pre><code>[1] &quot;Group Size: 500&quot;
[1] &quot;CI Extension: 50&quot;
# A tibble: 16 x 5
   h2    nQTL  `LMM-EXACT-INBRED` `LMM-EXACT-LOCO`  MIXED
   &lt;fct&gt; &lt;fct&gt;              &lt;dbl&gt;            &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1                 0.342            0.18   0.386 
 2 0.2   5                 0.391            0.104  0.386 
 3 0.2   10                0.223            0.0734 0.224 
 4 0.2   25                0.0927           0.0693 0.0940
 5 0.4   1                 0.333            0.513  0.534 
 6 0.4   5                 0.201            0.233  0.255 
 7 0.4   10                0.195            0.207  0.215 
 8 0.4   25                0.0311           0.0570 0.0596
 9 0.6   1                 0.483            0.697  0.704 
10 0.6   5                 0.311            0.4    0.412 
11 0.6   10                0.194            0.211  0.225 
12 0.6   25                0.0351           0.0648 0.0745
13 0.8   1                 0.582            0.727  0.739 
14 0.8   5                 0.391            0.451  0.479 
15 0.8   10                0.291            0.283  0.307 
16 0.8   25                0.0854           0.0797 0.0935
[1] &quot;Group Size: 500&quot;
[1] &quot;CI Extension: 150&quot;
# A tibble: 16 x 5
   h2    nQTL  `LMM-EXACT-INBRED` `LMM-EXACT-LOCO`  MIXED
   &lt;fct&gt; &lt;fct&gt;              &lt;dbl&gt;            &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1                 0.342            0.18   0.386 
 2 0.2   5                 0.354            0.0714 0.347 
 3 0.2   10                0.185            0.0431 0.179 
 4 0.2   25                0.0734           0.0345 0.0757
 5 0.4   1                 0.333            0.513  0.534 
 6 0.4   5                 0.181            0.205  0.227 
 7 0.4   10                0.151            0.172  0.181 
 8 0.4   25                0.0175           0.0388 0.0381
 9 0.6   1                 0.483            0.697  0.704 
10 0.6   5                 0.279            0.381  0.394 
11 0.6   10                0.170            0.194  0.208 
12 0.6   25                0.0278           0.0407 0.0476
13 0.8   1                 0.582            0.727  0.739 
14 0.8   5                 0.355            0.423  0.450 
15 0.8   10                0.260            0.262  0.285 
16 0.8   25                0.0574           0.0542 0.0634
[1] &quot;Group Size: 1000&quot;
[1] &quot;CI Extension: 50&quot;
# A tibble: 16 x 5
   h2    nQTL  `LMM-EXACT-INBRED` `LMM-EXACT-LOCO`  MIXED
   &lt;fct&gt; &lt;fct&gt;              &lt;dbl&gt;            &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1                 0.301            0.163  0.347 
 2 0.2   5                 0.282            0.0680 0.287 
 3 0.2   10                0.141            0.0642 0.149 
 4 0.2   25                0.0503           0.0686 0.0568
 5 0.4   1                 0.312            0.455  0.475 
 6 0.4   5                 0.152            0.191  0.206 
 7 0.4   10                0.167            0.163  0.173 
 8 0.4   25                0.0256           0.0334 0.0364
 9 0.6   1                 0.440            0.625  0.635 
10 0.6   5                 0.236            0.310  0.316 
11 0.6   10                0.169            0.159  0.175 
12 0.6   25                0.0350           0.0507 0.0604
13 0.8   1                 0.510            0.660  0.666 
14 0.8   5                 0.313            0.350  0.378 
15 0.8   10                0.208            0.205  0.221 
16 0.8   25                0.0702           0.0522 0.0651
[1] &quot;Group Size: 1000&quot;
[1] &quot;CI Extension: 150&quot;
# A tibble: 16 x 5
   h2    nQTL  `LMM-EXACT-INBRED` `LMM-EXACT-LOCO`  MIXED
   &lt;fct&gt; &lt;fct&gt;              &lt;dbl&gt;            &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1                 0.301            0.163  0.347 
 2 0.2   5                 0.256            0.0545 0.260 
 3 0.2   10                0.124            0.0348 0.123 
 4 0.2   25                0.0330           0.0342 0.0407
 5 0.4   1                 0.312            0.455  0.475 
 6 0.4   5                 0.139            0.168  0.184 
 7 0.4   10                0.128            0.133  0.142 
 8 0.4   25                0.0130           0.0208 0.0205
 9 0.6   1                 0.440            0.625  0.635 
10 0.6   5                 0.208            0.294  0.301 
11 0.6   10                0.151            0.148  0.163 
12 0.6   25                0.0277           0.0324 0.0393
13 0.8   1                 0.510            0.660  0.666 
14 0.8   5                 0.281            0.324  0.350 
15 0.8   10                0.183            0.185  0.201 
16 0.8   25                0.0486           0.0358 0.0441</code></pre>
<pre><code>[1] &quot;CI Extension: 50&quot;
[1] &quot;Algorithm: LMM-EXACT-INBRED&quot;
# A tibble: 16 x 4
   h2    nQTL   `500` `1000`
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1     0.342  0.301 
 2 0.2   5     0.391  0.282 
 3 0.2   10    0.223  0.141 
 4 0.2   25    0.0927 0.0503
 5 0.4   1     0.333  0.312 
 6 0.4   5     0.201  0.152 
 7 0.4   10    0.195  0.167 
 8 0.4   25    0.0311 0.0256
 9 0.6   1     0.483  0.440 
10 0.6   5     0.311  0.236 
11 0.6   10    0.194  0.169 
12 0.6   25    0.0351 0.0350
13 0.8   1     0.582  0.510 
14 0.8   5     0.391  0.313 
15 0.8   10    0.291  0.208 
16 0.8   25    0.0854 0.0702
[1] &quot;CI Extension: 150&quot;
[1] &quot;Algorithm: LMM-EXACT-INBRED&quot;
# A tibble: 16 x 4
   h2    nQTL   `500` `1000`
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1     0.342  0.301 
 2 0.2   5     0.354  0.256 
 3 0.2   10    0.185  0.124 
 4 0.2   25    0.0734 0.0330
 5 0.4   1     0.333  0.312 
 6 0.4   5     0.181  0.139 
 7 0.4   10    0.151  0.128 
 8 0.4   25    0.0175 0.0130
 9 0.6   1     0.483  0.440 
10 0.6   5     0.279  0.208 
11 0.6   10    0.170  0.151 
12 0.6   25    0.0278 0.0277
13 0.8   1     0.582  0.510 
14 0.8   5     0.355  0.281 
15 0.8   10    0.260  0.183 
16 0.8   25    0.0574 0.0486
[1] &quot;CI Extension: 50&quot;
[1] &quot;Algorithm: LMM-EXACT-LOCO&quot;
# A tibble: 16 x 4
   h2    nQTL   `500` `1000`
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1     0.18   0.163 
 2 0.2   5     0.104  0.0680
 3 0.2   10    0.0734 0.0642
 4 0.2   25    0.0693 0.0686
 5 0.4   1     0.513  0.455 
 6 0.4   5     0.233  0.191 
 7 0.4   10    0.207  0.163 
 8 0.4   25    0.0570 0.0334
 9 0.6   1     0.697  0.625 
10 0.6   5     0.4    0.310 
11 0.6   10    0.211  0.159 
12 0.6   25    0.0648 0.0507
13 0.8   1     0.727  0.660 
14 0.8   5     0.451  0.350 
15 0.8   10    0.283  0.205 
16 0.8   25    0.0797 0.0522
[1] &quot;CI Extension: 150&quot;
[1] &quot;Algorithm: LMM-EXACT-LOCO&quot;
# A tibble: 16 x 4
   h2    nQTL   `500` `1000`
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1     0.18   0.163 
 2 0.2   5     0.0714 0.0545
 3 0.2   10    0.0431 0.0348
 4 0.2   25    0.0345 0.0342
 5 0.4   1     0.513  0.455 
 6 0.4   5     0.205  0.168 
 7 0.4   10    0.172  0.133 
 8 0.4   25    0.0388 0.0208
 9 0.6   1     0.697  0.625 
10 0.6   5     0.381  0.294 
11 0.6   10    0.194  0.148 
12 0.6   25    0.0407 0.0324
13 0.8   1     0.727  0.660 
14 0.8   5     0.423  0.324 
15 0.8   10    0.262  0.185 
16 0.8   25    0.0542 0.0358
[1] &quot;CI Extension: 50&quot;
[1] &quot;Algorithm: MIXED&quot;
# A tibble: 16 x 4
   h2    nQTL   `500` `1000`
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1     0.386  0.347 
 2 0.2   5     0.386  0.287 
 3 0.2   10    0.224  0.149 
 4 0.2   25    0.0940 0.0568
 5 0.4   1     0.534  0.475 
 6 0.4   5     0.255  0.206 
 7 0.4   10    0.215  0.173 
 8 0.4   25    0.0596 0.0364
 9 0.6   1     0.704  0.635 
10 0.6   5     0.412  0.316 
11 0.6   10    0.225  0.175 
12 0.6   25    0.0745 0.0604
13 0.8   1     0.739  0.666 
14 0.8   5     0.479  0.378 
15 0.8   10    0.307  0.221 
16 0.8   25    0.0935 0.0651
[1] &quot;CI Extension: 150&quot;
[1] &quot;Algorithm: MIXED&quot;
# A tibble: 16 x 4
   h2    nQTL   `500` `1000`
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1     0.386  0.347 
 2 0.2   5     0.347  0.260 
 3 0.2   10    0.179  0.123 
 4 0.2   25    0.0757 0.0407
 5 0.4   1     0.534  0.475 
 6 0.4   5     0.227  0.184 
 7 0.4   10    0.181  0.142 
 8 0.4   25    0.0381 0.0205
 9 0.6   1     0.704  0.635 
10 0.6   5     0.394  0.301 
11 0.6   10    0.208  0.163 
12 0.6   25    0.0476 0.0393
13 0.8   1     0.739  0.666 
14 0.8   5     0.450  0.350 
15 0.8   10    0.285  0.201 
16 0.8   25    0.0634 0.0441</code></pre>
<pre><code>[1] &quot;Group Size: 500&quot;
[1] &quot;Algorithm: LMM-EXACT-INBRED&quot;
# A tibble: 16 x 4
   h2    nQTL    `50`  `150`
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1     0.342  0.342 
 2 0.2   5     0.391  0.354 
 3 0.2   10    0.223  0.185 
 4 0.2   25    0.0927 0.0734
 5 0.4   1     0.333  0.333 
 6 0.4   5     0.201  0.181 
 7 0.4   10    0.195  0.151 
 8 0.4   25    0.0311 0.0175
 9 0.6   1     0.483  0.483 
10 0.6   5     0.311  0.279 
11 0.6   10    0.194  0.170 
12 0.6   25    0.0351 0.0278
13 0.8   1     0.582  0.582 
14 0.8   5     0.391  0.355 
15 0.8   10    0.291  0.260 
16 0.8   25    0.0854 0.0574
[1] &quot;Group Size: 1000&quot;
[1] &quot;Algorithm: LMM-EXACT-INBRED&quot;
# A tibble: 16 x 4
   h2    nQTL    `50`  `150`
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1     0.301  0.301 
 2 0.2   5     0.282  0.256 
 3 0.2   10    0.141  0.124 
 4 0.2   25    0.0503 0.0330
 5 0.4   1     0.312  0.312 
 6 0.4   5     0.152  0.139 
 7 0.4   10    0.167  0.128 
 8 0.4   25    0.0256 0.0130
 9 0.6   1     0.440  0.440 
10 0.6   5     0.236  0.208 
11 0.6   10    0.169  0.151 
12 0.6   25    0.0350 0.0277
13 0.8   1     0.510  0.510 
14 0.8   5     0.313  0.281 
15 0.8   10    0.208  0.183 
16 0.8   25    0.0702 0.0486
[1] &quot;Group Size: 500&quot;
[1] &quot;Algorithm: LMM-EXACT-LOCO&quot;
# A tibble: 16 x 4
   h2    nQTL    `50`  `150`
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1     0.18   0.18  
 2 0.2   5     0.104  0.0714
 3 0.2   10    0.0734 0.0431
 4 0.2   25    0.0693 0.0345
 5 0.4   1     0.513  0.513 
 6 0.4   5     0.233  0.205 
 7 0.4   10    0.207  0.172 
 8 0.4   25    0.0570 0.0388
 9 0.6   1     0.697  0.697 
10 0.6   5     0.4    0.381 
11 0.6   10    0.211  0.194 
12 0.6   25    0.0648 0.0407
13 0.8   1     0.727  0.727 
14 0.8   5     0.451  0.423 
15 0.8   10    0.283  0.262 
16 0.8   25    0.0797 0.0542
[1] &quot;Group Size: 1000&quot;
[1] &quot;Algorithm: LMM-EXACT-LOCO&quot;
# A tibble: 16 x 4
   h2    nQTL    `50`  `150`
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1     0.163  0.163 
 2 0.2   5     0.0680 0.0545
 3 0.2   10    0.0642 0.0348
 4 0.2   25    0.0686 0.0342
 5 0.4   1     0.455  0.455 
 6 0.4   5     0.191  0.168 
 7 0.4   10    0.163  0.133 
 8 0.4   25    0.0334 0.0208
 9 0.6   1     0.625  0.625 
10 0.6   5     0.310  0.294 
11 0.6   10    0.159  0.148 
12 0.6   25    0.0507 0.0324
13 0.8   1     0.660  0.660 
14 0.8   5     0.350  0.324 
15 0.8   10    0.205  0.185 
16 0.8   25    0.0522 0.0358
[1] &quot;Group Size: 500&quot;
[1] &quot;Algorithm: MIXED&quot;
# A tibble: 16 x 4
   h2    nQTL    `50`  `150`
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1     0.386  0.386 
 2 0.2   5     0.386  0.347 
 3 0.2   10    0.224  0.179 
 4 0.2   25    0.0940 0.0757
 5 0.4   1     0.534  0.534 
 6 0.4   5     0.255  0.227 
 7 0.4   10    0.215  0.181 
 8 0.4   25    0.0596 0.0381
 9 0.6   1     0.704  0.704 
10 0.6   5     0.412  0.394 
11 0.6   10    0.225  0.208 
12 0.6   25    0.0745 0.0476
13 0.8   1     0.739  0.739 
14 0.8   5     0.479  0.450 
15 0.8   10    0.307  0.285 
16 0.8   25    0.0935 0.0634
[1] &quot;Group Size: 1000&quot;
[1] &quot;Algorithm: MIXED&quot;
# A tibble: 16 x 4
   h2    nQTL    `50`  `150`
   &lt;fct&gt; &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 0.2   1     0.347  0.347 
 2 0.2   5     0.287  0.260 
 3 0.2   10    0.149  0.123 
 4 0.2   25    0.0568 0.0407
 5 0.4   1     0.475  0.475 
 6 0.4   5     0.206  0.184 
 7 0.4   10    0.173  0.142 
 8 0.4   25    0.0364 0.0205
 9 0.6   1     0.635  0.635 
10 0.6   5     0.316  0.301 
11 0.6   10    0.175  0.163 
12 0.6   25    0.0604 0.0393
13 0.8   1     0.666  0.666 
14 0.8   5     0.378  0.350 
15 0.8   10    0.221  0.201 
16 0.8   25    0.0651 0.0441</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.3 (2020-02-29)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] data.table_1.13.2        wesanderson_0.3.6.9000   viridis_0.5.1           
 [4] viridisLite_0.3.0        RColorBrewer_1.1-2       nationalparkcolors_0.1.0
 [7] yardstick_0.0.7          workflows_0.2.1          tune_0.1.1              
[10] rsample_0.0.8            recipes_0.1.13           parsnip_0.1.3           
[13] modeldata_0.0.2          infer_0.5.3              dials_0.0.9             
[16] scales_1.1.1             broom_0.7.2              tidymodels_0.1.1        
[19] forcats_0.5.0            stringr_1.4.0            dplyr_1.0.2             
[22] purrr_0.3.4              readr_1.4.0              tidyr_1.1.2             
[25] tibble_3.0.4             ggplot2_3.3.2            tidyverse_1.3.0         
[28] workflowr_1.6.2         

loaded via a namespace (and not attached):
 [1] colorspace_1.4-1   ellipsis_0.3.1     class_7.3-17       rprojroot_1.3-2   
 [5] fs_1.5.0           rstudioapi_0.11    farver_2.0.3       listenv_0.8.0     
 [9] furrr_0.1.0        prodlim_2019.11.13 fansi_0.4.1        lubridate_1.7.9   
[13] xml2_1.3.2         codetools_0.2-16   splines_3.6.3      knitr_1.30        
[17] jsonlite_1.7.1     pROC_1.16.2        dbplyr_1.4.4       compiler_3.6.3    
[21] httr_1.4.2         backports_1.1.10   assertthat_0.2.1   Matrix_1.2-18     
[25] cli_2.1.0          later_1.1.0.1      htmltools_0.5.0    tools_3.6.3       
[29] gtable_0.3.0       glue_1.4.2         Rcpp_1.0.5         cellranger_1.1.0  
[33] DiceDesign_1.8-1   vctrs_0.3.4        iterators_1.0.12   timeDate_3043.102 
[37] gower_0.2.2        xfun_0.18          globals_0.13.1     rvest_0.3.6       
[41] lifecycle_0.2.0    future_1.19.1      MASS_7.3-53        ipred_0.9-9       
[45] hms_0.5.3          promises_1.1.1     parallel_3.6.3     yaml_2.2.1        
[49] gridExtra_2.3      rpart_4.1-15       stringi_1.5.3      highr_0.8         
[53] foreach_1.5.0      lhs_1.1.1          lava_1.6.8         rlang_0.4.8       
[57] pkgconfig_2.0.3    evaluate_0.14      lattice_0.20-41    labeling_0.4.2    
[61] cowplot_1.1.0      tidyselect_1.1.0   plyr_1.8.6         magrittr_1.5      
[65] R6_2.4.1           generics_0.0.2     DBI_1.1.0          pillar_1.4.6      
[69] haven_2.3.1        whisker_0.4        withr_2.3.0        survival_3.2-7    
[73] nnet_7.3-14        modelr_0.1.8       crayon_1.3.4       utf8_1.1.4        
[77] rmarkdown_2.5      grid_3.6.3         readxl_1.3.1       blob_1.2.1        
[81] git2r_0.27.1       reprex_0.3.0       digest_0.6.27      httpuv_1.5.4      
[85] munsell_0.5.0      GPfit_1.0-8       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
