<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Home</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">NemaScan: GWAS in C. elegans Wild Isolates</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sam-widmayer/NemaScan_Performance">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Home</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-01-01
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>NemaScan_Performance/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsamwidmayerNemaScanPerformancetreedfbfcd6a8a258b6865c5f2ff0ce8cfdf7a910959targetblankdfbfcd6a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sam-widmayer/NemaScan_Performance/tree/dfbfcd6a8a258b6865c5f2ff0ce8cfdf7a910959" target="_blank">dfbfcd6</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsamwidmayerNemaScanPerformancetreedfbfcd6a8a258b6865c5f2ff0ce8cfdf7a910959targetblankdfbfcd6a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sam-widmayer/NemaScan_Performance/tree/dfbfcd6a8a258b6865c5f2ff0ce8cfdf7a910959" target="_blank">dfbfcd6</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  Emodepside_cv.EXT_LD.plot.png
    Untracked:  code/Gather_Population_Metrics.R
    Untracked:  code/praziquantel.processing.R
    Untracked:  data/NemaScan_Performance.Algorithm.Sims.20201110_EMMA.RData
    Untracked:  data/NemaScan_Performance.Architecture.20201120.RData
    Untracked:  data/NemaScan_Performance.Architecture.Mixed.20201223.RData
    Untracked:  data/NemaScan_Performance.Architecture.Mixed.CI.50.20201224.RData
    Untracked:  data/NemaScan_Performance.Architecture.Mixed.group500.20201223.RData
    Untracked:  data/NemaScan_Performance.Architecture.Mixed.group500.CI.50.20201224.RData
    Untracked:  data/NemaScan_Performance.Architecture.group500.20201124.RData
    Untracked:  data/NemaScan_Performance.Demography.SampleSize.Complete.Subsampled.20201230.RData
    Untracked:  data/NemaScan_Performance.Variant.Set.HF.chr125.20201112.RData
    Untracked:  data/NemaScan_Performance.Variant.Set.Hard.Filtered.20201111.RData
    Untracked:  data/NemaScan_Performance.Variant.Set.Imputed.20201111.RData
    Untracked:  data/NemaScan_Performance.Variant.Set.Imputed.chr125.20201112.RData
    Untracked:  data/NemaScan_Population_Metrics.Demography.SampleSize.Complete.Subsampled.20201229.RData
    Untracked:  output/WB.figure.png
    Untracked:  output/complete.subsampled.strains.txt
    Untracked:  output/pr.curve.variant.set.png
    Untracked:  output/prelim.fig.1.png
    Untracked:  output/prelim.fig.2.png
    Untracked:  output/report.html
    Untracked:  output/subsampled.strains_20201225_n144_reps200.txt
    Untracked:  output/subsampled.strains_20201225_n192_reps200.txt
    Untracked:  output/subsampled.strains_20201225_n240_reps200.txt
    Untracked:  output/subsampled.strains_20201225_n288_reps200.txt
    Untracked:  output/subsampled.strains_20201225_n336_reps200.txt
    Untracked:  output/subsampled.strains_20201225_n384_reps200.txt
    Untracked:  output/subsampled.strains_20201225_n48_reps200.txt
    Untracked:  output/subsampled.strains_20201225_n96_reps200.txt
    Untracked:  output/subsampled.swept.strains_20201225_n120_reps200.txt
    Untracked:  output/subsampled.swept.strains_20201225_n144_reps200.txt
    Untracked:  output/subsampled.swept.strains_20201225_n168_reps200.txt
    Untracked:  output/subsampled.swept.strains_20201225_n24_reps200.txt
    Untracked:  output/subsampled.swept.strains_20201225_n48_reps200.txt
    Untracked:  output/subsampled.swept.strains_20201225_n72_reps200.txt
    Untracked:  output/subsampled.swept.strains_20201225_n96_reps200.txt
    Untracked:  output/subsampled.unswept.strains_20201225_n120_reps200.txt
    Untracked:  output/subsampled.unswept.strains_20201225_n144_reps200.txt
    Untracked:  output/subsampled.unswept.strains_20201225_n168_reps200.txt
    Untracked:  output/subsampled.unswept.strains_20201225_n192_reps200.txt
    Untracked:  output/subsampled.unswept.strains_20201225_n216_reps200.txt
    Untracked:  output/subsampled.unswept.strains_20201225_n24_reps200.txt
    Untracked:  output/subsampled.unswept.strains_20201225_n48_reps200.txt
    Untracked:  output/subsampled.unswept.strains_20201225_n72_reps200.txt
    Untracked:  output/subsampled.unswept.strains_20201225_n96_reps200.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML (<code>docs/index.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/dfbfcd6a8a258b6865c5f2ff0ce8cfdf7a910959/analysis/index.Rmd" target="_blank">dfbfcd6</a>
</td>
<td>
Sam
</td>
<td>
2021-01-01
</td>
<td>
remove CeNDR set analysis from page
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/ed09ecc0d980252ea53e3663b427b60953b5f86c/docs/index.html" target="_blank">ed09ecc</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-10-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/7aa8f62dd944c9fe508e9d5a9c7b58db00cc2bd0/analysis/index.Rmd" target="_blank">7aa8f62</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-10-01
</td>
<td>
update with cendr strain set website link
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/4f70f25b20189fd56b36166ab56b04e80e240693/docs/index.html" target="_blank">4f70f25</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-10-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/a9fb7317cc6a0ef1a96cb3b9218a1282431c52b6/analysis/index.Rmd" target="_blank">a9fb731</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-10-01
</td>
<td>
update with cendr strain set analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/783f7088b7c35a47100383ce7d245513c2ff5245/docs/index.html" target="_blank">783f708</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-10-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/85d963d7f2a6f8bc0606fa9417d504540ac243b3/analysis/index.Rmd" target="_blank">85d963d</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-10-01
</td>
<td>
update links to new variant set analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/6076e8516f54c3d13761acb54cf3d8b2db95322d/docs/index.html" target="_blank">6076e85</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-09-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/9330d3ba5c72ae351d703d5afab68f8f09f37b1f/analysis/index.Rmd" target="_blank">9330d3b</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-09-26
</td>
<td>
update with population composition analysis link
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/a1188e30b4caeee136c561d137d0c0c3e37df435/docs/index.html" target="_blank">a1188e3</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-09-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/85df9fcaddcf2c2f7b10af5a632cbf6b55edb0f0/analysis/index.Rmd" target="_blank">85df9fc</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-09-09
</td>
<td>
adding CI analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/a845122f20826140da5def1f4eb103ddad3b1003/docs/index.html" target="_blank">a845122</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-08-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/d40a28d70b3312b1889ef4299f5c3f1e12940605/analysis/index.Rmd" target="_blank">d40a28d</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-08-25
</td>
<td>
add gamma project to index
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sam-widmayer/NemaScan_Performance/60c204fcbfd35d2812b8d0001da3bcaee389d61f/docs/index.html" target="_blank">60c204f</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-08-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/3f540de4c3e29adb12abe5b7b79fbfc8fd4ee3b4/analysis/index.Rmd" target="_blank">3f540de</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-08-25
</td>
<td>
Publish the initial files for project
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sam-widmayer/NemaScan_Performance/blob/4a092ceffc50825b04c04ca7c41f6f27eda4ecfa/analysis/index.Rmd" target="_blank">4a092ce</a>
</td>
<td>
Sam Widmayer
</td>
<td>
2020-08-25
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p><a href="https://www.elegansvariation.org/">CeNDR</a> is a resource maintained by the Andersen Lab which enables the collection and organization of <em>C. elegans</em> wild isolates for quantitative genetics and gene discovery. Recently, we have re-vamped our genome-wide association (GWA) mapping pipeline and expanded set of strains and genetic variants available for exploration. Analyses using simulations are currently ongoing to determine the power and utility of <em>C. elegans</em> for quantitative genetics.</p>
<ol style="list-style-type: decimal">
<li>A <a href="powerprecisionh2.html">small overview</a> of mapping performance with a single QTL with a defined effect.</li>
<li>A <a href="gamma.effects.html">more nuanced example</a> where there are a small collection of QTLs with effects that are distributed more closely to what we expect to see in nature.</li>
<li>We assessed false-positive rates in this GWA pipeline, with a specific focus on how they can help us conceptualize <a href="FPR_CI.html">confidence intervals</a> and divergent regions and <a href="variant.set.comp.html">different types of variant sets</a> can affect mapping results</li>
<li><em>C. elegans</em> strains from around the world have vastly different evolutionary histories, shaping genomic variation in different ways. We interrogated how <a href="population.comparison.html">population demography</a> affects GWA performance given a finite sampling capacity.</li>
</ol>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
