---
title: "Exact Linear Mixed Model Approaches are Most Effective for C. elegans GWAS"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
editor_options:
  chunk_output_type: console
---
```{r setup, message=FALSE, warning=FALSE, include=FALSE}
require(tidyverse)
require(tidymodels)
require(nationalparkcolors)
require(RColorBrewer)
require(workflowr)
require(viridis)
require(wesanderson)
require(data.table)
```

### Analysis date: `r format(Sys.time(), '%B %d, %Y')`

```{r message=FALSE, warning=FALSE, include=FALSE}
setwd("~/Documents/projects/NemaScan_Performance/") # Lab Computer
# setwd("~/Documents/AndersenLab/NemaScan_Performance/") # Laptop
# load(file = "data/NemaScan_Performance.Algorithm.Sims.20201110_EMMA.RData") # EMMA, Algorithm Sims 
load(file = "data/NemaScan_Performance.Algorithm.Sims.20210510.RData") # EMMA, Algorithm Sims
dat <- simulation.metrics.df %>%
  tidyr::separate(col = sim,
                  into = c("nQTL","Rep","h2","MAF","effect_range","strain_set"), 
                  sep = "_", remove = F) %>%
  tidyr::separate(col = QTL,
                  into = c("CHROM","POS"), 
                  sep = ":", remove = F) %>%
  dplyr::mutate(h2 = as.factor(h2),
                nQTL = as.factor(nQTL),
                POS = as.numeric(POS),
                startPOS = as.numeric(startPOS),
                peakPOS = as.numeric(peakPOS),
                endPOS = as.numeric(endPOS),
                interval.var.exp  = as.numeric(interval.var.exp),
                Simulated.QTL.VarExp = as.numeric(Simulated.QTL.VarExp), 
                peak_id = as.numeric(peak_id),
                BETA = as.numeric(BETA),
                Effect = as.numeric(Effect),
                Frequency = as.numeric(Frequency),
                log10p = dplyr::if_else(Simulated == FALSE, 
                                        true = interval.log10p, 
                                        false = log10p), # false discoveries inherit the log10p value of the peak marker for the interval
                log10p = as.numeric(log10p),
                interval_size = as.numeric(interval_size),
                aboveBF = dplyr::case_when(aboveBF == 1 ~ TRUE, 
                                           aboveBF == 0 ~ FALSE,
                                           is.na(aboveBF) ~ TRUE), # false discoveries by definition exceed significance threshold
                aboveBF = as.factor(aboveBF)) %>%
  dplyr::filter(!c(Detected == FALSE & aboveBF == TRUE),
                CHROM != 7, 
                algorithm != "LMM-EXACT-INBRED-LOCO")
```

### NemaScan simulation performance was assessed with the following experimental parameters:
* ###### Number of Simulated QTL: ```r levels(as.factor(dat$nQTL))```
* ###### Sample Population(s): ```r levels(as.factor(dat$strain_set))```
* ###### Heritability(ies): ```r levels(as.factor(dat$h2))```
* ###### MAF(s): ```r levels(as.factor(dat$MAF))```
* ###### Number of Replicates per Regime: ```r max(as.numeric(dat$Rep))```
* ###### QTL Effect Range: ```r levels(as.factor(dat$effect_range))```

```{r simulated QTL locations, echo=FALSE}
dat %>%
  dplyr::filter(Simulated == TRUE,
                !duplicated(QTL)) %>%
  ggplot(mapping = aes(x = POS/1000000, y = Effect, fill = nQTL)) + 
  theme_bw() + 
  geom_point(shape = 21, alpha = 0.75) + 
  facet_grid(.~CHROM, drop = TRUE) + 
  scale_fill_manual(values = park_palettes$Hawaii[c(2,3)]) + 
  labs(x = "Genomic position (Mb)",
       y = "Causal QTL Effect",
       title = "Where were QTL Simulated?")

dat %>%
  dplyr::filter(Simulated == TRUE,
                !duplicated(QTL)) %>%
  ggplot(., mapping = aes(x = Frequency, fill = nQTL)) +
  theme_bw() + 
  geom_density() + 
  scale_fill_manual(values = nationalparkcolors::park_palettes$RockyMountains[c(1,2)]) + 
  theme(legend.position = "none") + 
  facet_grid(nQTL~.) + 
  xlim(c(0.05,0.5)) + 
  labs(x = "Minor Allele Frequency",
       y = "Frequency of Simulated QTL (Smoothed Density)")

dat %>%
  dplyr::filter(Simulated == TRUE,
                !duplicated(QTL)) %>%
  ggplot(., mapping = aes(x = Simulated.QTL.VarExp, fill = nQTL)) +
  theme_bw() + 
  geom_density() + 
  scale_fill_manual(values = nationalparkcolors::park_palettes$RockyMountains[c(1,2)]) + 
  facet_grid(nQTL~.) + 
  theme(legend.position = "none")+
  labs(x = "Variance Explained by Simulated QTL",
       y = "Frequency of Simulated QTL (Smoothed Density)")
```

```{r PR curve, eval=FALSE, fig.height=6, fig.width=9, include=FALSE}
pr.curve <- dat %>%
  dplyr::group_by(h2, algorithm, nQTL) %>%
  yardstick::pr_curve(Simulated, log10p) %>%
  dplyr::filter(!is.infinite(.threshold)) %>%
  dplyr::mutate(nQTL = as.factor(nQTL))
levels(pr.curve$nQTL) <- c("1 Simulated QTL", "5 Simulated QTL")

h2.pal <- wes_palette("BottleRocket2", 9, type = "continuous")
algorithm.B <- ggplot(pr.curve, mapping = aes(x = recall, y = precision, group = h2, colour = h2)) + 
  theme_bw() + 
  geom_path() +
  theme(strip.text = element_text(size = 8),
        axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
        panel.grid.minor = element_blank()) +
  theme(legend.position = "top") + 
  ylim(c(0,1)) + 
  scale_colour_manual(values = h2.pal, name = expression(italic(h^2))) +
  facet_grid(nQTL~algorithm, scales = "free", space = "free") + 
  labs(x = "Recall", y = "Precision")
algorithm.B
```

```{r metrics v heritability, echo=FALSE}
algorithm.palette <- c("#E3B505", # EMMAx
                       "#DCB8CB", # LMM EXACT
                       "#5B2E48", # LMM EXACT INBRED
                       "#87E5FD") # LMM EXACT LOCO

designations <- dat %>%
  dplyr::mutate(designation = case_when(Simulated == TRUE & Detected == TRUE & aboveBF == TRUE ~ "Detected.CV",
                                        Simulated == TRUE & Detected == FALSE & aboveBF == FALSE ~ "Missed.CV",
                                        Simulated == TRUE & Detected == TRUE & aboveBF == FALSE ~ "CV.Not.Significant.In.Interval",
                                        Simulated == FALSE & Detected == TRUE & aboveBF == TRUE ~ "False.Discovery")) %>%
  tidyr::separate(col = detected.peak,
                  into = c("peak.CHROM","peak.POS"), 
                  sep = ":", remove = F) %>%
  dplyr::mutate(QTL.v.peak = abs(as.numeric(POS)-as.numeric(peak.POS))) %>%
  dplyr::group_by(designation, algorithm, nQTL, h2, Rep) %>%
  dplyr::summarise(n = n()) %>%
  tidyr::pivot_wider(names_from = designation, values_from = n)
designations[is.na(designations)] <- 0
designations$Detected <- rowSums(designations[,5:7])

Power <- designations %>%
  dplyr::mutate(Simulated = as.numeric(as.character(nQTL)),
                Power = Detected.CV/Simulated,
                Artefact.Rate = False.Discovery/Detected,
                Detected.CV.NS.Rate = CV.Not.Significant.In.Interval/Detected) %>%
  dplyr::ungroup() %>%
  dplyr::group_by(h2, nQTL, algorithm) %>%
  dplyr::summarise(mean.Power = mean(Power),
                   sd.Power = sd(Power),
                   n = n())
Power$h2 <- as.numeric(as.character(Power$h2))
algorithm.A <- Power %>%
  ggplot(., mapping = aes(x = h2, y = mean.Power, colour = algorithm, group = algorithm )) + 
  theme_bw() + 
  # geom_pointrange(aes(ymin=mean.Power-sd.Power, ymax=mean.Power+sd.Power), position = position_dodge(width = 0.5)) +
  geom_point(position = position_dodge(width = 0.01)) +
  geom_line(position = position_dodge(width = 0.01)) +
  facet_grid(~nQTL, scales = "free_x", space = "free_x") + 
  scale_colour_manual(values = algorithm.palette, name = "Algorithm") + 
  ylim(c(0,1)) + 
  scale_y_continuous(breaks = seq(0,1,0.1), labels = seq(0,1,0.1)) + 
  theme(strip.text = element_text(size = 8),
        legend.position = "top", 
        panel.grid.minor = element_blank()) +
  labs(y = "Power",
       x = expression(italic(h^2)))
  


# plots <- cowplot::plot_grid(algorithm.A + theme(legend.position = "none"), 
#                                        algorithm.B + theme(legend.position = "none"), 
#                                        labels = "AUTO", 
#                                        rel_widths = c(0.75, 1), 
#                                        ncol = 2)
# legends <- cowplot::plot_grid(cowplot::get_legend(algorithm.A), 
#                               cowplot::get_legend(algorithm.B), 
#                               ncol = 2, rel_widths = c(1.75,1))
# algorithm.figure <- cowplot::plot_grid(plots, legends, ncol = 1, rel_heights = c(6,1))
ggsave(algorithm.A, width = 8, height = 5, filename = "output/prelim.fig.1.png")






# AF <- designations %>%
#   dplyr::mutate(Simulated = as.numeric(as.character(nQTL)),
#                 Power = Detected.CV/Simulated,
#                 Artefact.Rate = if_else(is.na(False.Discovery/Detected), 
#                                         true = 0, 
#                                         false = (False.Discovery/Detected)),
#                 Detected.CV.NS.Rate = if_else(is.na(CV.Not.Significant.In.Interval/Detected), 
#                                               true = 0, 
#                                               false = (CV.Not.Significant.In.Interval/Detected))) %>%
#   dplyr::ungroup() %>%
#   dplyr::group_by(h2, nQTL, algorithm) %>%
#   dplyr::summarise(mean.AR = mean(Artefact.Rate),
#                    sd.AR = sd(Artefact.Rate),
#                    n = n())
# AF$h2 <- as.numeric(as.character(AF$h2))
# AF %>%
#   ggplot(., mapping = aes(x = h2, y = mean.AR, colour = algorithm, group = algorithm )) + 
#   theme_bw() + 
#   # geom_pointrange(aes(ymin=mean.Power-sd.Power, ymax=mean.Power+sd.Power), position = position_dodge(width = 0.5)) +
#   geom_point(position = position_dodge(width = 0.05)) +
#   geom_line(position = position_dodge(width = 0.05)) +
#   facet_grid(~nQTL, scales = "free_x", space = "free_x") + 
#   scale_colour_manual(values = algorithm.palette, name = "Algorithm")
# 



```

```{r, stats}
power.stats.1QTL <- dat %>%
  dplyr::filter(nQTL == 1) %>%
  dplyr::group_by(h2, algorithm, Rep) %>%
  yardstick::sens(truth = Simulated, estimate = Detected) %>%
  dplyr::mutate(.metric = "Power") %>%
  dplyr::ungroup() %>%
  dplyr::group_by(h2) %>%
  tidyr::nest()
power.stats.5QTL <- dat %>%
  dplyr::filter(nQTL == 5) %>%
  dplyr::group_by(h2, algorithm, Rep) %>%
  yardstick::sens(truth = Simulated, estimate = Detected) %>%
  dplyr::mutate(.metric = "Power") %>%
  dplyr::ungroup() %>%
  dplyr::group_by(h2) %>%
  tidyr::nest()
kruskal.h2nested <- function(data){
  summary(kruskal.test(.estimate ~ algorithm, data = data))
  pairwise.wilcox.test(data$.estimate, data$algorithm, p.adjust.method = "BH")
}

oneQTL.PWRST <- purrr::map(power.stats.1QTL$data, kruskal.h2nested)
names(oneQTL.PWRST) <- power.stats.1QTL$h2
fiveQTL.PWRST <- purrr::map(power.stats.5QTL$data, kruskal.h2nested)
names(fiveQTL.PWRST) <- power.stats.5QTL$h2
print("Power, 1 QTL, Pairwise Wilcoxon Rank Sum Test + BH Correction")
oneQTL.PWRST
print("Power, 5 QTL, Pairwise Wilcoxon Rank Sum Test + BH Correction")
fiveQTL.PWRST


# Individual replicate bootstrapping
pr.auc.bootstrap.1QTL <- function(data, perc){
  bootstrapped.reps <- sample(as.numeric(unique(data$Rep)), 
                              length(as.numeric(unique(data$Rep)))*perc, 
                              replace = T)
  bootstraps <- list()
  for(i in 1:length(bootstrapped.reps)){
    bootstraps[[i]] <- data[which(data$Rep %in% as.character(bootstrapped.reps[i])),] %>%
      dplyr::filter(nQTL == 1)
  }
  Reduce(rbind, bootstraps)
}
pr.auc.bootstrap.5QTL <- function(data, perc){
  bootstrapped.reps <- sample(as.numeric(unique(data$Rep)), 
                              length(as.numeric(unique(data$Rep)))*perc, 
                              replace = T)
  bootstraps <- list()
  for(i in 1:length(bootstrapped.reps)){
    bootstraps[[i]] <- data[which(data$Rep %in% as.character(bootstrapped.reps[i])),] %>%
      dplyr::filter(nQTL == 5)
  }
  Reduce(rbind, bootstraps)
}
# Bootstrap some number of folds and calculate PR-AUC
replicate.bootstraps <- list()
for(j in seq(1:500)){
  replicate.bootstraps[[j]] <- pr.auc.bootstrap.1QTL(data = dat, perc = 1) %>%
    dplyr::group_by(h2, algorithm) %>%
    yardstick::pr_auc(Simulated, log10p) %>%
    dplyr::mutate(.metric = "Precision-Recall_AUC",
                  fold = j)
}
bootstrapped.PRAUC.1QTL <- Reduce(rbind, replicate.bootstraps)

replicate.bootstraps <- list()
for(j in seq(1:500)){
  replicate.bootstraps[[j]] <- pr.auc.bootstrap.5QTL(data = dat, perc = 1) %>%
    dplyr::group_by(h2, algorithm) %>%
    yardstick::pr_auc(Simulated, log10p) %>%
    dplyr::mutate(.metric = "Precision-Recall_AUC",
                  fold = j)
}
bootstrapped.PRAUC.5QTL <- Reduce(rbind, replicate.bootstraps)

print("Bootstrapped PR-AUC, 1 QTL, Pairwise Wilcoxon Rank Sum Test + BH Correction")
prAUC.stats.1QTL <- bootstrapped.PRAUC.1QTL %>%
  dplyr::group_by(h2) %>%
  tidyr::nest()
oneQTL.PRAUC.PWRST <- purrr::map(prAUC.stats.1QTL$data, kruskal.h2nested)
names(oneQTL.PRAUC.PWRST) <- prAUC.stats.1QTL$h2
bootstrapped.PRAUC.1QTL %>%
  dplyr::group_by(h2, algorithm) %>%
  dplyr::summarise(median = median(.estimate)) %>%
  tidyr::pivot_wider(names_from = h2, values_from = median)
oneQTL.PRAUC.PWRST

print("Bootstrapped PR-AUC, 5 QTL, Pairwise Wilcoxon Rank Sum Test + BH Correction")
prAUC.stats.5QTL <- bootstrapped.PRAUC.5QTL %>%
  dplyr::group_by(h2) %>%
  tidyr::nest()
fiveQTL.PRAUC.PWRST <- purrr::map(prAUC.stats.5QTL$data, kruskal.h2nested)
names(fiveQTL.PRAUC.PWRST) <- prAUC.stats.5QTL$h2
bootstrapped.PRAUC.5QTL %>%
  dplyr::group_by(h2, algorithm) %>%
  dplyr::summarise(median = median(.estimate)) %>%
  tidyr::pivot_wider(names_from = h2, values_from = median)
fiveQTL.PRAUC.PWRST
```


```{r eval=FALSE, include=FALSE}
EIGEN <- data.table::fread("data/dauer_total_independent_tests.txt") %>%
  as.numeric()
dauer.INBRED <- data.table::fread("data/processed_median.dauer.fraction_LMM_EXACT_INBRED_mapping.tsv") %>%
  dplyr::filter(!duplicated(marker),
                CHROM != 7) %>%
  dplyr::mutate(aboveBF = as.numeric(aboveBF),
                aboveBF = if_else(condition = (aboveBF < 1 & log10p > -log10((0.05/EIGEN))), 
                                  true = 2, 
                                  false = aboveBF),
                CHROM = as.factor(CHROM))
levels(dauer.INBRED$CHROM) <- c("I","II","III","IV","V","X")

dauer.LOCO <- data.table::fread("data/processed_median.dauer.fraction_LMM_EXACT_LOCO_mapping.tsv") %>%
  dplyr::filter(!duplicated(marker),
                CHROM != 7) %>%
  dplyr::mutate(aboveBF = as.numeric(aboveBF),
                aboveBF = if_else(condition = (aboveBF < 1 & log10p > -log10((0.05/EIGEN))), 
                                  true = 2, 
                                  false = aboveBF),
                CHROM = as.factor(CHROM))
levels(dauer.LOCO$CHROM) <- c("I","II","III","IV","V","X")

dauer.EMMA <- data.table::fread("data/median.dauer.fraction_processed_mapping.tsv") %>%
  dplyr::filter(!duplicated(marker)) %>%
  dplyr::mutate(aboveBF = as.numeric(aboveBF),
                aboveBF = if_else(condition = (aboveBF < 1 & log10p > -log10((0.05/EIGEN))), 
                                  true = 2, 
                                  false = aboveBF))

man.plot.LOCO <- dauer.LOCO %>%
  ggplot(.,aes(x = POS/1000000, 
             y = log10p,
             colour = as.factor(aboveBF))) + 
  theme_bw() + 
  geom_point(size = 1, alpha = 0.5) +
  scale_colour_manual(values = c("black","red","red")) +
  scale_y_continuous(expand = c(0,0), limits = c(0,9)) +
  geom_hline(yintercept = unique(dauer.LOCO$BF), linetype = 2) +
  geom_hline(yintercept = -log10((0.05/EIGEN)), linetype = 3) +
  labs(x = "Genomic position (Mb)",
       y = expression(-log[10](italic(p)))) +
  theme(legend.position = "none",
        panel.grid = element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank()) + 
  facet_grid(.~CHROM, scales = "free_x", space = "free")

man.plot.inbred <- dauer.INBRED %>%
  ggplot(.,aes(x = POS/1000000, 
             y = log10p,
             colour = as.factor(aboveBF))) + 
  theme_bw() + 
  geom_point(size = 1, alpha = 0.5) +
  scale_colour_manual(values = c("black","red","red")) +
  scale_y_continuous(expand = c(0,0), limits = c(0,9)) +
  geom_hline(yintercept = unique(dauer.INBRED$BF), linetype = 2) +
  geom_hline(yintercept = -log10((0.05/EIGEN)), linetype = 3) +
  labs(x = "Genomic position (Mb)",
       y = expression(-log[10](italic(p)))) +
  theme(legend.position = "none",
        panel.grid = element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank()) + 
  facet_grid(.~CHROM, scales = "free_x", space = "free")

man.plot.LOCO <- dauer.LOCO %>%
  ggplot(.,aes(x = POS/1000000, 
             y = log10p,
             colour = as.factor(aboveBF))) + 
  theme_bw() + 
  geom_point(size = 1, alpha = 0.5) +
  scale_colour_manual(values = c("black","red","red")) +
  scale_y_continuous(expand = c(0,0), limits = c(0,9)) +
  geom_hline(yintercept = unique(dauer.LOCO$BF), linetype = 2) +
  geom_hline(yintercept = -log10((0.05/EIGEN)), linetype = 3) +
  labs(x = "Genomic position (Mb)",
       y = expression(-log[10](italic(p)))) +
  theme(legend.position = "none",
        panel.grid = element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank()) + 
  facet_grid(.~CHROM, scales = "free_x", space = "free")

man.plot.inbred <- dauer.EMMA %>%
  ggplot(.,aes(x = POS/1000000, 
             y = log10p,
             colour = as.factor(aboveBF))) + 
  theme_bw() + 
  geom_point(size = 1, alpha = 0.5) +
  scale_colour_manual(values = c("black","red","red")) +
  scale_y_continuous(expand = c(0,0), limits = c(0,9)) +
  geom_hline(yintercept = unique(dauer.EMMA$BF), linetype = 2) +
  geom_hline(yintercept = -log10((0.05/EIGEN)), linetype = 3) +
  labs(x = "Genomic position (Mb)",
       y = expression(-log[10](italic(p)))) +
  theme(legend.position = "none",
        panel.grid = element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank()) + 
  facet_grid(.~CHROM, scales = "free_x", space = "free")

power.plot <- summarized %>%
  dplyr::filter(algorithm == "EMMA",
                nQTL == "1 Simulated QTL") %>%
  dplyr::mutate(h2 = as.numeric(levels(h2))[h2]) %>%
  ggplot(., mapping = aes(x = h2, y = Power)) +
  theme_bw() +
  geom_jitter(width = 0.01, height = 0) +
  geom_line() +
  theme(strip.text = element_text(size = 8),
        legend.position = "top", 
        panel.grid.minor = element_blank()) +
  labs(y = "Power",
       x = expression(italic(h^2)))

WB.plots <- cowplot::plot_grid(power.plot,
                               man.plot,
                               labels = "AUTO", 
                               rel_widths = c(1.5, 4),
                               align = "h", axis = "b",
                               ncol = 2)
ggsave(WB.plots, width = 7.5, height = 2.5, filename = "output/WB.figure.png")

cowplot::plot_grid(man.plot.inbred, man.plot.LOCO, ncol = 1)
```
