---
title: "Exact Linear Mixed Model Approaches are Most Effective for C. elegans GWAS"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
editor_options:
  chunk_output_type: console
---
```{r setup, message=FALSE, warning=FALSE, include=FALSE}
require(tidyverse)
require(tidymodels)
require(nationalparkcolors)
require(RColorBrewer)
require(workflowr)
require(viridis)
require(wesanderson)
require(data.table)
```

### Analysis date: `r format(Sys.time(), '%B %d, %Y')`

```{r message=FALSE, warning=FALSE, include=FALSE}
setwd("~/Documents/projects/NemaScan_Performance/") # Lab Computer
# setwd("~/Documents/AndersenLab/NemaScan_Performance/") # Laptop
load(file = "data/NemaScan_Performance.Algorithm.Sims.20201110_EMMA.RData") # EMMA, Algorithm Sims 
dat <- simulation.metrics.df %>%
  tidyr::separate(col = sim,
                  into = c("nQTL","Rep","h2","MAF","effect_range","strain_set"), 
                  sep = "_", remove = F) %>%
  tidyr::separate(col = QTL,
                  into = c("CHROM","POS"), 
                  sep = ":", remove = F) %>%
  dplyr::mutate(h2 = as.factor(h2),
                POS = as.numeric(POS),
                startPOS = as.numeric(startPOS),
                peakPOS = as.numeric(peakPOS),
                endPOS = as.numeric(endPOS),
                var.exp  = as.numeric(var.exp), # watch
                Simulated.QTL.VarExp = as.numeric(Simulated.QTL.VarExp), # watch
                peak_id = as.numeric(peak_id),
                BETA = as.numeric(BETA),
                Effect = as.numeric(Effect),
                Frequency = as.numeric(Frequency),
                log10p = as.numeric(log10p),
                interval_size = as.numeric(interval_size)
                #, algorithm = if_else(condition = algorithm == "EMMA",
                #                     true = "EMMAx",
                #                     false = algorithm)
                ) %>%
  dplyr::filter(!algorithm %in% c("No Successful Mappings",
                                  "LMM-EXACT-LOCO-INBRED"),
                CHROM != 7)
```

### NemaScan simulation performance was assessed with the following experimental parameters:
* ###### Number of Simulated QTL: ```r levels(as.factor(dat$nQTL))```
* ###### Sample Population(s): ```r levels(as.factor(dat$strain_set))```
* ###### Heritability(ies): ```r levels(as.factor(dat$h2))```
* ###### MAF(s): ```r levels(as.factor(dat$MAF))```
* ###### Number of Replicates per Regime: ```r max(as.numeric(dat$Rep))```
* ###### QTL Effect Range: ```r levels(as.factor(dat$effect_range))```

```{r simulated QTL locations, echo=FALSE}
dat %>%
  dplyr::filter(Simulated == TRUE,
                !duplicated(QTL)) %>%
  ggplot(mapping = aes(x = POS/1000000, y = Effect, fill = nQTL)) + 
  theme_bw() + 
  geom_point(shape = 21, alpha = 0.75) + 
  facet_grid(.~CHROM, drop = TRUE) + 
  scale_fill_manual(values = park_palettes$Hawaii[c(2,3)]) + 
  labs(x = "Genomic position (Mb)",
       y = "Causal QTL Effect",
       title = "Where were QTL Simulated?")

dat %>%
  dplyr::filter(Simulated == TRUE,
                !duplicated(QTL)) %>%
  ggplot(., mapping = aes(x = Frequency, fill = nQTL)) +
  theme_bw() + 
  geom_density() + 
  scale_fill_manual(values = nationalparkcolors::park_palettes$RockyMountains[c(1,2)]) + 
  theme(legend.position = "none") + 
  facet_grid(nQTL~.) + 
  xlim(c(0.05,0.5)) + 
  labs(x = "Minor Allele Frequency",
       y = "Frequency of Simulated QTL (Smoothed Density)")

dat %>%
  dplyr::filter(Simulated == TRUE,
                !duplicated(QTL)) %>%
  ggplot(., mapping = aes(x = Simulated.QTL.VarExp, fill = nQTL)) +
  theme_minimal() + 
  geom_density() + 
  scale_fill_manual(values = nationalparkcolors::park_palettes$RockyMountains[c(1,2)]) + 
  facet_grid(nQTL~.) + 
  theme(legend.position = "none")+
  labs(x = "Variance Explained by Simulated QTL",
       y = "Frequency of Simulated QTL (Smoothed Density)")
```

```{r PR curve, echo=FALSE, fig.height=6, fig.width=9}
pr.curve <- dat %>%
  dplyr::group_by(h2, algorithm, nQTL) %>%
  yardstick::pr_curve(Simulated, log10p) %>%
  dplyr::filter(!is.infinite(.threshold)) %>%
  dplyr::mutate(nQTL = as.factor(nQTL))
levels(pr.curve$nQTL) <- c("1 Simulated QTL", "5 Simulated QTL")

h2.pal <- wes_palette("BottleRocket2", 9, type = "continuous")
algorithm.B <- ggplot(pr.curve, mapping = aes(x = recall, y = precision, group = h2, colour = h2)) + 
  theme_bw() + 
  geom_path() +
  theme(strip.text = element_text(size = 8),
        axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
        panel.grid.minor = element_blank()) +
  theme(legend.position = "top") + 
  ylim(c(0,1)) + 
  scale_colour_manual(values = h2.pal, name = expression(italic(h^2))) +
  facet_grid(nQTL~algorithm, scales = "free", space = "free") + 
  labs(x = "Recall", y = "Precision")
algorithm.B
```

```{r metrics v heritability, echo=FALSE}
algorithm.palette <- c("#E3B505", # EMMAx
                       "#DCB8CB", # LMM EXACT
                       "#5B2E48", # LMM EXACT INBRED
                       "#87E5FD") # LMM EXACT LOCO

power <- dat %>%
  dplyr::group_by(h2, algorithm, nQTL) %>%
  yardstick::sens(truth = Simulated, estimate = Detected) %>%
  dplyr::mutate(.metric = "Power")
precision <- dat %>%
  dplyr::group_by(h2, algorithm, nQTL) %>%
  yardstick::precision(truth = Simulated, estimate = Detected) %>%
  dplyr::mutate(.metric = "Precision")
pr_auc <- dat %>%
  dplyr::group_by(h2, algorithm, nQTL) %>%
  yardstick::pr_auc(Simulated, log10p) %>%
  dplyr::mutate(.metric = "Precision-Recall_AUC")
resolution <- dat %>%
  dplyr::filter(Simulated == TRUE) %>%
  dplyr::filter(!is.na(var.exp)) %>%
  dplyr::select(CHROM, POS, Simulated, Frequency, interval_size, algorithm, h2, Simulated.QTL.VarExp, var.exp, nQTL) %>%
  dplyr::group_by(h2, algorithm, nQTL) %>%
  dplyr::summarise(mean(interval_size), median(interval_size), sd(interval_size), 
                   mean(Simulated.QTL.VarExp), median(Simulated.QTL.VarExp), sd(Simulated.QTL.VarExp))
summarized <- power %>%
  dplyr::full_join(., precision) %>%
  dplyr::full_join(., pr_auc) %>%
  tidyr::pivot_wider(names_from = .metric, values_from = .estimate) %>%
  dplyr::mutate(F1score = 2*((Precision*Power)/(Precision+Power)),
                FPR = 1-Precision) %>%
  dplyr::left_join(.,resolution) %>%
  dplyr::mutate(mean.interval.Mb = `mean(interval_size)`/1000000,
                median.interval.Mb = `median(interval_size)`/1000000,
                sd.interval.Mb = `sd(interval_size)`/1000000, 
                mean.PVE = `mean(Simulated.QTL.VarExp)`*100,
                median.PVE = `median(Simulated.QTL.VarExp)`*100,
                sd.PVE = `sd(Simulated.QTL.VarExp)`*100, 
                nQTL = as.factor(nQTL),
                .keep = "unused")
levels(summarized$nQTL) <- c("1 Simulated QTL", "5 Simulated QTL")


## PLOTS
power.difference.makers <- dat %>%
  dplyr::group_by(h2, algorithm, nQTL, Rep) %>%
  yardstick::sens(truth = Simulated, estimate = Detected) %>%
  dplyr::mutate(.metric = "Power") %>%
  dplyr::filter(algorithm %in% c("EMMA","LMM-EXACT-LOCO"),
                nQTL == 5) %>%
  tidyr::pivot_wider(names_from = algorithm, values_from = .estimate) %>%
  dplyr::mutate(DELTA = `LMM-EXACT-LOCO` - EMMA)

algorithm.A <- summarized %>%
  dplyr::mutate(h2 = as.numeric(levels(h2))[h2]) %>%
  ggplot(., mapping = aes(x = h2, y = Power, colour = algorithm)) +
  theme_bw() +
  geom_jitter(width = 0.01, height = 0) +
  geom_line(aes(group = algorithm)) +
  scale_colour_manual(values = algorithm.palette, name = "Algorithm") + 
  facet_grid(.~nQTL, scales = "free", space = "free") +  
  theme(strip.text = element_text(size = 8),
        legend.position = "top", 
        panel.grid.minor = element_blank()) +
  labs(y = "Power",
       x = expression(italic(h^2)))


plots <- cowplot::plot_grid(algorithm.A + theme(legend.position = "none"), 
                                       algorithm.B + theme(legend.position = "none"), 
                                       labels = "AUTO", 
                                       rel_widths = c(0.75, 1), 
                                       ncol = 2)
legends <- cowplot::plot_grid(cowplot::get_legend(algorithm.A), 
                              cowplot::get_legend(algorithm.B), 
                              ncol = 2, rel_widths = c(1.75,1))
algorithm.figure <- cowplot::plot_grid(plots, legends, ncol = 1, rel_heights = c(6,1))
ggsave(algorithm.figure, width = 10, height = 5, filename = "output/prelim.fig.1.png")

summarized %>%
  dplyr::mutate(h2 = as.numeric(levels(h2))[h2]) %>%
  ggplot(., mapping = aes(x = h2, y = Precision, colour = algorithm)) +
  theme_bw() +
  geom_jitter(width = 0.01, height = 0) +
  geom_line(aes(group = algorithm)) +
  scale_colour_manual(values = algorithm.palette, name = "Algorithm") + 
  facet_grid(.~nQTL, scales = "free", space = "free") +  
  # ylim(c(0,1)) + 
  labs(title = "Precision v. Heritability",
       x = expression(italic(h^2)))

summarized %>%
  dplyr::mutate(h2 = as.numeric(levels(h2))[h2]) %>%
  ggplot(., mapping = aes(x = h2, y = `Precision-Recall_AUC`, colour = algorithm)) +
  theme_bw() +
  geom_jitter(width = 0.01, height = 0) +
  geom_line(aes(group = algorithm)) +
  scale_colour_manual(values = algorithm.palette, name = "Algorithm") + 
  facet_grid(.~nQTL, scales = "free", space = "free") +  
  # ylim(c(0,1)) + 
  labs(title = "Precision-Recall AUC v. Heritability",
       x = expression(italic(h^2)))

summarized %>%
  dplyr::mutate(h2 = as.numeric(levels(h2))[h2]) %>%
  ggplot(., mapping = aes(x = h2, y = F1score, colour = algorithm)) +
  theme_bw() +
  geom_jitter(width = 0.01, height = 0) +
  geom_line(aes(group = algorithm)) +
  scale_colour_manual(values = algorithm.palette, name = "Algorithm") + 
  facet_grid(.~nQTL, scales = "free", space = "free") +  
  # ylim(c(0,1)) + 
  labs(title = "F-score v. Heritability",
       x = expression(italic(h^2)))

summarized %>%
  dplyr::mutate(h2 = as.numeric(levels(h2))[h2]) %>%
  ggplot(., mapping = aes(x = h2, y = FPR, colour = algorithm)) +
  theme_bw() +
  geom_jitter(width = 0.01, height = 0) +
  geom_line(aes(group = algorithm)) +
  scale_colour_manual(values = algorithm.palette, name = "Algorithm") + 
  facet_grid(.~nQTL, scales = "free", space = "free") +  
  ylim(c(0,1)) + 
  labs(title = "FPR v. Heritability",
       x = expression(italic(h^2)))
```

```{r MAF+VE, echo=FALSE, message=FALSE, warning=FALSE}
MAF.cuts <- seq(0, 1, by = 0.025)
var.exp.cuts <- seq(0, 1, by = 0.1)

binned.dat <- dat %>%
  dplyr::mutate(MAF.bin = as.factor(cut(Frequency, breaks = MAF.cuts, include.lowest = F)),
                VE.bin = as.factor(cut(Simulated.QTL.VarExp, breaks = var.exp.cuts, include.lowest = F)))

MAF.power <- binned.dat %>%
  dplyr::group_by(MAF.bin, algorithm, nQTL) %>%
  yardstick::sens(truth = Simulated, estimate = Detected) %>%
  dplyr::mutate(.metric = "Power")
MAF.resolution <- binned.dat %>%
  dplyr::filter(Simulated == TRUE) %>%
  dplyr::filter(!is.na(var.exp)) %>%
  dplyr::select(CHROM, POS, Simulated, Frequency, interval_size, algorithm, h2, Simulated.QTL.VarExp, var.exp, nQTL, MAF.bin) %>%
  dplyr::group_by(MAF.bin, algorithm, nQTL) %>%
  dplyr::summarise(mean(interval_size), median(interval_size), sd(interval_size))
MAF.summarized <- MAF.power %>%
  tidyr::pivot_wider(names_from = .metric, values_from = .estimate) %>%
  dplyr::left_join(.,MAF.resolution) %>%
  dplyr::mutate(mean.interval.Mb = `mean(interval_size)`/1000000,
                median.interval.Mb = `median(interval_size)`/1000000,
                sd.interval.Mb = `sd(interval_size)`/1000000, 
                .keep = "unused")

VE.nest <- binned.dat %>%
  dplyr::filter(Simulated == TRUE) %>%
  dplyr::group_by(VE.bin, algorithm, nQTL) %>%
  tidyr::nest()
VE.power <- binned.dat %>%
  dplyr::filter(Simulated == TRUE) %>%
  dplyr::group_by(VE.bin, algorithm, nQTL) %>%
  yardstick::sens(truth = Simulated, estimate = Detected) %>%
  dplyr::mutate(.metric = "Power")
VE.resolution <- binned.dat %>%
  dplyr::filter(Simulated == TRUE) %>%
  dplyr::filter(!is.na(var.exp)) %>%
  dplyr::select(CHROM, POS, Simulated, Frequency, interval_size, algorithm, h2, Simulated.QTL.VarExp, var.exp, nQTL, VE.bin) %>%
  dplyr::group_by(VE.bin, algorithm, nQTL) %>%
  dplyr::summarise(mean(interval_size), median(interval_size), sd(interval_size))
VE.summarized <- VE.power %>%
  tidyr::pivot_wider(names_from = .metric, values_from = .estimate) %>%
  dplyr::left_join(., VE.resolution) %>%
  dplyr::mutate(mean.interval.Mb = `mean(interval_size)`/1000000,
                median.interval.Mb = `median(interval_size)`/1000000,
                sd.interval.Mb = `sd(interval_size)`/1000000, 
                .keep = "unused")


## PLOTS
MAF.summarized %>%
  # dplyr::mutate(h2 = as.numeric(levels(h2))[h2]) %>%
  ggplot(., mapping = aes(x = MAF.bin, y = Power, colour = algorithm)) +
  theme_bw() +
  geom_jitter(width = 0.01, height = 0) +
  geom_line(aes(group = algorithm)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1)) + 
  scale_colour_manual(values = algorithm.palette, name = "Algorithm") + 
  facet_grid(.~nQTL, scales = "free", space = "free") +  
  ylim(c(0,1)) + 
  labs(title = "Power v. QTL MAF",
       x = "MAF Bin")

VE.summarized %>%
  # dplyr::mutate(h2 = as.numeric(levels(h2))[h2]) %>%
  ggplot(., mapping = aes(x = VE.bin, y = Power, colour = algorithm)) +
  theme_bw() +
  geom_jitter(width = 0.01, height = 0) +
  geom_line(aes(group = algorithm)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1)) + 
  scale_colour_manual(values = algorithm.palette, name = "Algorithm") + 
  facet_grid(.~nQTL, scales = "free", space = "free") +  
  ylim(c(0,1)) + 
  labs(title = "Power v. QTL VE",
       x = "VE Bin")

MAF.summarized %>%
  ggplot(., mapping = aes(x = MAF.bin, y = mean.interval.Mb, colour = algorithm)) +
  theme_bw() +
  geom_jitter(width = 0.01, height = 0) +
  geom_line(aes(group = algorithm)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1)) + 
  scale_colour_manual(values = algorithm.palette, name = "Algorithm") + 
  facet_grid(.~nQTL, scales = "free", space = "free") +  
  labs(title = "Resolution v. QTL MAF",
       y = "Mean CI Width",
       x = "MAF Bin")

VE.summarized %>%
  ggplot(., mapping = aes(x = VE.bin, y = mean.interval.Mb, colour = algorithm)) +
  theme_bw() +
  geom_jitter(width = 0.01, height = 0) +
  geom_line(aes(group = algorithm)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1)) + 
  scale_colour_manual(values = algorithm.palette, name = "Algorithm") + 
  facet_grid(.~nQTL, scales = "free", space = "free") +  
  labs(title = "Resolution v. QTL VE",
       y = "Mean CI Width",
       x = "VE Bin")
```

```{r eval=FALSE, include=FALSE}
EIGEN <- data.table::fread("data/dauer_total_independent_tests.txt") %>%
  as.numeric()
dauer.INBRED <- data.table::fread("data/processed_median.dauer.fraction_LMM_EXACT_INBRED_mapping.tsv") %>%
  dplyr::filter(!duplicated(marker),
                CHROM != 7) %>%
  dplyr::mutate(aboveBF = as.numeric(aboveBF),
                aboveBF = if_else(condition = (aboveBF < 1 & log10p > -log10((0.05/EIGEN))), 
                                  true = 2, 
                                  false = aboveBF),
                CHROM = as.factor(CHROM))
levels(dauer.INBRED$CHROM) <- c("I","II","III","IV","V","X")

dauer.LOCO <- data.table::fread("data/processed_median.dauer.fraction_LMM_EXACT_LOCO_mapping.tsv") %>%
  dplyr::filter(!duplicated(marker),
                CHROM != 7) %>%
  dplyr::mutate(aboveBF = as.numeric(aboveBF),
                aboveBF = if_else(condition = (aboveBF < 1 & log10p > -log10((0.05/EIGEN))), 
                                  true = 2, 
                                  false = aboveBF),
                CHROM = as.factor(CHROM))
levels(dauer.LOCO$CHROM) <- c("I","II","III","IV","V","X")

dauer.EMMA <- data.table::fread("data/median.dauer.fraction_processed_mapping.tsv") %>%
  dplyr::filter(!duplicated(marker)) %>%
  dplyr::mutate(aboveBF = as.numeric(aboveBF),
                aboveBF = if_else(condition = (aboveBF < 1 & log10p > -log10((0.05/EIGEN))), 
                                  true = 2, 
                                  false = aboveBF))

man.plot.LOCO <- dauer.LOCO %>%
  ggplot(.,aes(x = POS/1000000, 
             y = log10p,
             colour = as.factor(aboveBF))) + 
  theme_bw() + 
  geom_point(size = 1, alpha = 0.5) +
  scale_colour_manual(values = c("black","red","red")) +
  scale_y_continuous(expand = c(0,0), limits = c(0,9)) +
  geom_hline(yintercept = unique(dauer.LOCO$BF), linetype = 2) +
  geom_hline(yintercept = -log10((0.05/EIGEN)), linetype = 3) +
  labs(x = "Genomic position (Mb)",
       y = expression(-log[10](italic(p)))) +
  theme(legend.position = "none",
        panel.grid = element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank()) + 
  facet_grid(.~CHROM, scales = "free_x", space = "free")

man.plot.inbred <- dauer.INBRED %>%
  ggplot(.,aes(x = POS/1000000, 
             y = log10p,
             colour = as.factor(aboveBF))) + 
  theme_bw() + 
  geom_point(size = 1, alpha = 0.5) +
  scale_colour_manual(values = c("black","red","red")) +
  scale_y_continuous(expand = c(0,0), limits = c(0,9)) +
  geom_hline(yintercept = unique(dauer.INBRED$BF), linetype = 2) +
  geom_hline(yintercept = -log10((0.05/EIGEN)), linetype = 3) +
  labs(x = "Genomic position (Mb)",
       y = expression(-log[10](italic(p)))) +
  theme(legend.position = "none",
        panel.grid = element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank()) + 
  facet_grid(.~CHROM, scales = "free_x", space = "free")

man.plot.LOCO <- dauer.LOCO %>%
  ggplot(.,aes(x = POS/1000000, 
             y = log10p,
             colour = as.factor(aboveBF))) + 
  theme_bw() + 
  geom_point(size = 1, alpha = 0.5) +
  scale_colour_manual(values = c("black","red","red")) +
  scale_y_continuous(expand = c(0,0), limits = c(0,9)) +
  geom_hline(yintercept = unique(dauer.LOCO$BF), linetype = 2) +
  geom_hline(yintercept = -log10((0.05/EIGEN)), linetype = 3) +
  labs(x = "Genomic position (Mb)",
       y = expression(-log[10](italic(p)))) +
  theme(legend.position = "none",
        panel.grid = element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank()) + 
  facet_grid(.~CHROM, scales = "free_x", space = "free")

man.plot.inbred <- dauer.EMMA %>%
  ggplot(.,aes(x = POS/1000000, 
             y = log10p,
             colour = as.factor(aboveBF))) + 
  theme_bw() + 
  geom_point(size = 1, alpha = 0.5) +
  scale_colour_manual(values = c("black","red","red")) +
  scale_y_continuous(expand = c(0,0), limits = c(0,9)) +
  geom_hline(yintercept = unique(dauer.EMMA$BF), linetype = 2) +
  geom_hline(yintercept = -log10((0.05/EIGEN)), linetype = 3) +
  labs(x = "Genomic position (Mb)",
       y = expression(-log[10](italic(p)))) +
  theme(legend.position = "none",
        panel.grid = element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank()) + 
  facet_grid(.~CHROM, scales = "free_x", space = "free")

power.plot <- summarized %>%
  dplyr::filter(algorithm == "EMMA",
                nQTL == "1 Simulated QTL") %>%
  dplyr::mutate(h2 = as.numeric(levels(h2))[h2]) %>%
  ggplot(., mapping = aes(x = h2, y = Power)) +
  theme_bw() +
  geom_jitter(width = 0.01, height = 0) +
  geom_line() +
  theme(strip.text = element_text(size = 8),
        legend.position = "top", 
        panel.grid.minor = element_blank()) +
  labs(y = "Power",
       x = expression(italic(h^2)))

WB.plots <- cowplot::plot_grid(power.plot,
                               man.plot,
                               labels = "AUTO", 
                               rel_widths = c(1.5, 4),
                               align = "h", axis = "b",
                               ncol = 2)
ggsave(WB.plots, width = 7.5, height = 2.5, filename = "output/WB.figure.png")

cowplot::plot_grid(man.plot.inbred, man.plot.LOCO, ncol = 1)
```
