---
title: "CeNDR Strain Set Power Analysis"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: true
    toc_float: true
    toc_collapsed: true
    toc_depth: 4
editor_options:
  chunk_output_type: console
---
```{r setup, message=FALSE, warning=FALSE, include=FALSE}
require(tidyverse)
require(tidymodels)
require(nationalparkcolors)
require(RColorBrewer)
require(GenomicRanges)
```

```{r echo=FALSE, message=FALSE, warning=FALSE}
setwd("~/Documents/projects/NemaScan_Performance/")
load(file = "data/NemaScan_Performance.CeNDR2020_cendr_sets.20200929.RData") # EMMAx, hard-filtered
dat <- simulation.metrics.df %>%
  tidyr::separate(col = sim,
                  into = c("nQTL","Rep","h2","MAF","effect_range","strain_set"), 
                  sep = "_", remove = F) %>%
  tidyr::separate(col = QTL,
                  into = c("CHROM","POS"), 
                  sep = ":", remove = F) %>%
  dplyr::mutate(h2 = as.factor(h2),
                POS = as.numeric(POS),
                startPOS = as.numeric(startPOS),
                peakPOS = as.numeric(peakPOS),
                endPOS = as.numeric(endPOS),
                var.exp  = as.numeric(var.exp),
                peak_id = as.numeric(peak_id),
                BETA = as.numeric(BETA),
                Effect = as.numeric(Effect),
                Frequency = as.numeric(Frequency),
                log10p = as.numeric(log10p),
                interval_size = as.numeric(interval_size)) %>%
  dplyr::filter(algorithm == "LMM-EXACT")
```

## Simulation Parameters

NemaScan simulation performance was assessed with the following experimental parameters:

* ###### Number of Simulated QTL: ```r levels(as.factor(dat$nQTL))```
* ###### Sample Population(s): ```r levels(as.factor(dat$strain_set))```
* ###### Heritability(ies): ```r levels(as.factor(dat$h2))```
* ###### MAF(s): ```r levels(as.factor(dat$MAF))```
* ###### Number of Replicates per Regime: ```r max(as.numeric(dat$Rep))```
* ###### QTL Effect Range: ```r levels(as.factor(dat$effect_range))```

## Simulated QTL Locations

```{r echo=FALSE}
dat %>%
  dplyr::filter(Simulated == TRUE, 
                !is.na(CHROM)) %>%
  ggplot(mapping = aes(x = POS/1000000, y = Effect)) + 
  theme_bw() + 
  geom_point(alpha = 0.5) + 
  facet_grid(.~CHROM, drop = TRUE) + 
  labs(x = "Genomic position (Mb)",
       y = "Causal QTL Effect",
       title = "Where were QTL Simulated?")

dat %>%
  dplyr::filter(Simulated == TRUE, 
                !is.na(CHROM)) %>%
  ggplot(mapping = aes(x = Frequency, fill = strain_set)) +
  theme_minimal() + 
  geom_density(position = "stack") + 
  # scale_fill_manual(values = c(brewer.pal(5, "Greens"),"blue","orange"), name = "Strain Set") + 
  xlim(c(0.05,0.5)) + 
  labs(x = "Minor Allele Frequency",
       y = "Frequency of Simulated QTL (Smoothed Density)")

dat %>%
  dplyr::filter(Simulated == TRUE, 
                !is.na(CHROM)) %>%
  ggplot(., mapping = aes(x = abs(Effect), fill = strain_set)) +
  theme_minimal() + 
  geom_density(position = "stack") + 
  # scale_fill_manual(values = c(brewer.pal(5, "Greens"),"blue","orange"), name = "Strain Set") + 
  labs(x = "abs(QTL Effect)",
       y = "Frequency of Simulated QTL (Smoothed Density)")
```

## Power and Precision

Through [CeNDR](https://www.elegansvariation.org/), we offer a ready-made sampling of strain sets. Roughly 48 strains comprise each set, and sample the range of genetic variation present in the full set (403) of CeNDR strains. This is more amenable to collaboration: to exhaustively sample 403 strains in a GWA framework is not always tenable or necessary for all analyses. On the flip side, a 100 strain GWA mapping is expected to .ack power to detect certain genetic architectures. We sought to quantify this, as well as evaluate the performance of various strain sets provided a 2 strain set cap for each mapping. We compared the performance of these panels to a random 96 strain sample drawn from proportions of swept and unswept strains present in the 403 strain set.

```{r include=FALSE}
power <- dat %>%
  dplyr::group_by(h2, strain_set) %>%
  yardstick::sens(truth = Simulated, estimate = Detected) %>%
  dplyr::mutate(.metric = "Power")

precision <- dat %>%
  dplyr::group_by(h2, strain_set) %>%
  yardstick::precision(truth = Simulated, estimate = Detected) %>%
  dplyr::mutate(.metric = "Precision")

resolution <- dat %>%
  dplyr::filter(Detected == TRUE) %>%
  dplyr::filter(!is.na(var.exp)) %>%
  dplyr::select(CHROM, POS, Simulated, Frequency, interval_size, h2, var.exp, strain_set) %>%
  dplyr::group_by(h2, strain_set) %>%
  dplyr::summarise(mean(interval_size), median(interval_size), sd(interval_size), 
                   mean(var.exp), median(var.exp), sd(var.exp))

summarized <- power %>%
  dplyr::full_join(., precision) %>%
  tidyr::pivot_wider(names_from = .metric, values_from = .estimate) %>%
  dplyr::left_join(.,resolution) %>%
  dplyr::mutate(mean.interval.Mb = `mean(interval_size)`/1000000,
                median.interval.Mb = `median(interval_size)`/1000000,
                sd.interval.Mb = `sd(interval_size)`/1000000, 
                mean.PVE = `mean(var.exp)`*100,
                median.PVE = `median(var.exp)`*100,
                sd.PVE = `sd(var.exp)`*100, 
                .keep = "unused") %>%
  tidyr::separate(col = strain_set,
                  into = c("blah1","blah2","set1","blah3","set2"), 
                  sep = "[.]", remove = F) %>%
  dplyr::select(-c(blah1,blah2,blah3)) %>%
  dplyr::mutate(set.category = if_else(condition = is.na(set2), 
                                       true = "Full Subset", 
                                       false = "CeNDR Subset"),
                set2 = if_else(condition = is.na(set2),
                               true = "96",
                               false = set2))
```

```{r power precision plots, echo=FALSE, fig.height=6, fig.width=7}
# Ranked Power
# h2 = 0.5
set.a.pal <- c(brewer.pal(n = 9, name = "Set1")[1:7],"gray")
set.b.pal <- c(brewer.pal(n = 9, name = "Set1")[2:8],"gray")
power.seta.fill <- summarized %>%
  dplyr::mutate(h2 = as.numeric(levels(h2))[h2]) %>%
  dplyr::filter(h2 == 0.5) %>%
  ggplot(., mapping = aes(x = reorder(strain_set,Power), y = Power, fill = set1)) +
  theme_bw() +
  geom_point(size = 3, shape = 21) +
  facet_grid(.~set.category, scales = "free", space = "free") +
  scale_fill_manual(values = set.a.pal, name = "Set A") + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  labs(x = "Strain Set")
power.setb.fill <- summarized %>%
  dplyr::mutate(h2 = as.numeric(levels(h2))[h2]) %>%
  dplyr::filter(h2 == 0.5) %>%
  ggplot(., mapping = aes(x = reorder(strain_set,Power), y = Power, fill = set2)) +
  theme_bw() +
  geom_point(size = 3, shape = 21) +
  facet_grid(.~set.category, scales = "free", space = "free") +
  scale_fill_manual(values = set.b.pal, name = "Set B") + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  labs(x = "Strain Set")
cowplot::plot_grid(power.seta.fill, power.setb.fill)

# h2 = 0.8
power.seta.fill <- summarized %>%
  dplyr::mutate(h2 = as.numeric(levels(h2))[h2]) %>%
  dplyr::filter(h2 == 0.8) %>%
  ggplot(., mapping = aes(x = reorder(strain_set,Power), y = Power, fill = set1)) +
  theme_bw() +
  geom_point(size = 3, shape = 21) +
  facet_grid(.~set.category, scales = "free", space = "free") +
  scale_fill_manual(values = set.a.pal, name = "Set A") + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  labs(x = "Strain Set")
power.setb.fill <- summarized %>%
  dplyr::mutate(h2 = as.numeric(levels(h2))[h2]) %>%
  dplyr::filter(h2 == 0.8) %>%
  ggplot(., mapping = aes(x = reorder(strain_set,Power), y = Power, fill = set2)) +
  theme_bw() +
  geom_point(size = 3, shape = 21) +
  facet_grid(.~set.category, scales = "free", space = "free") +
  scale_fill_manual(values = set.b.pal, name = "Set B") + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  labs(x = "Strain Set")
cowplot::plot_grid(power.seta.fill, power.setb.fill)



# Ranked Precision
precision.seta.fill <- summarized %>%
  dplyr::mutate(h2 = as.numeric(levels(h2))[h2]) %>%
  dplyr::filter(h2 == 0.5) %>%
  ggplot(., mapping = aes(x = reorder(strain_set,Precision), y = Precision, fill = set1)) +
  theme_bw() +
  geom_point(size = 3, shape = 21) +
  facet_grid(.~set.category, scales = "free", space = "free") +
  scale_fill_manual(values = set.a.pal, name = "Set A") + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  labs(x = "Strain Set")
precision.setb.fill <- summarized %>%
  dplyr::mutate(h2 = as.numeric(levels(h2))[h2]) %>%
  dplyr::filter(h2 == 0.5) %>%
  ggplot(., mapping = aes(x = reorder(strain_set,Precision), y = Precision, fill = set2)) +
  theme_bw() +
  geom_point(size = 3, shape = 21) +
  facet_grid(.~set.category, scales = "free", space = "free") +
  scale_fill_manual(values = set.b.pal, name = "Set B") + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  labs(x = "Strain Set")
cowplot::plot_grid(precision.seta.fill, precision.setb.fill)

# h2 = 0.8
precision.seta.fill <- summarized %>%
  dplyr::mutate(h2 = as.numeric(levels(h2))[h2]) %>%
  dplyr::filter(h2 == 0.8) %>%
  ggplot(., mapping = aes(x = reorder(strain_set,Precision), y = Precision, fill = set1)) +
  theme_bw() +
  geom_point(size = 3, shape = 21) +
  facet_grid(.~set.category, scales = "free", space = "free") +
  scale_fill_manual(values = set.a.pal, name = "Set A") + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  labs(x = "Strain Set")
precision.setb.fill <- summarized %>%
  dplyr::mutate(h2 = as.numeric(levels(h2))[h2]) %>%
  dplyr::filter(h2 == 0.8) %>%
  ggplot(., mapping = aes(x = reorder(strain_set,Precision), y = Precision, fill = set2)) +
  theme_bw() +
  geom_point(size = 3, shape = 21) +
  facet_grid(.~set.category, scales = "free", space = "free") +
  scale_fill_manual(values = set.b.pal, name = "Set B") + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  labs(x = "Strain Set")
cowplot::plot_grid(precision.seta.fill, precision.setb.fill)



# Set Heatmap
summarized %>%
  dplyr::mutate(h2 = as.numeric(levels(h2))[h2]) %>%
  dplyr::filter(set.category == "CeNDR Subset") %>%
  ggplot(., mapping = aes(x = set1, y = set2, fill = Power)) +
  theme_classic() +
  geom_tile() + 
  facet_grid(.~h2, scales = "free", space = "free") + 
  scale_fill_distiller(type = "seq", direction = 1, palette = "YlGnBu") + 
  labs(y = "Set 2",
       x = "Set 1")

summarized %>%
  dplyr::mutate(h2 = as.numeric(levels(h2))[h2]) %>%
  dplyr::filter(set.category == "CeNDR Subset") %>%
  ggplot(., mapping = aes(x = set1, y = set2, fill = Precision)) +
  theme_classic() +
  geom_tile() + 
  facet_grid(.~h2, scales = "free", space = "free") + 
  scale_fill_distiller(type = "seq", direction = 1, palette = "YlOrBr") + 
  labs(y = "Set 2",
       x = "Set 1")
```

## QTL Resolution

```{r resolution plots, echo=FALSE, fig.height=6, fig.width=7}
# h2 = 0.5
power.0.5 <- summarized %>%
  dplyr::mutate(h2 = as.numeric(levels(h2))[h2]) %>%
  dplyr::filter(h2 == 0.5) %>%
  ggplot(., mapping = aes(x = reorder(strain_set,Power), y = mean.interval.Mb, colour = Power)) +
  theme_bw() +
  geom_pointrange(aes(ymin = mean.interval.Mb-sd.interval.Mb,
                      ymax = mean.interval.Mb+sd.interval.Mb), alpha = 0.7) +
  scale_colour_distiller(type = "seq", direction = 1, palette = "YlGnBu") + 
  facet_grid(.~set.category, scales = "free", space = "free") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  labs(title = "GWA Mapping Resolution: h2 = 0.5",
       x = "Strain Set",
       y = "Mean QTL Interval Width (Mb)")

precision.0.5 <- summarized %>%
  dplyr::mutate(h2 = as.numeric(levels(h2))[h2]) %>%
  dplyr::filter(h2 == 0.5) %>%
  ggplot(., mapping = aes(x = reorder(strain_set,Precision), y = mean.interval.Mb, colour = Precision)) +
  theme_bw() +
  geom_pointrange(aes(ymin = mean.interval.Mb-sd.interval.Mb,
                      ymax = mean.interval.Mb+sd.interval.Mb), alpha = 0.7) +
  scale_colour_distiller(type = "seq", direction = 1, palette = "YlOrBr") + 
  facet_grid(.~set.category, scales = "free", space = "free") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  labs(title = "GWA Mapping Resolution: h2 = 0.5",
       x = "Strain Set",
       y = "Mean QTL Interval Width (Mb)")

# h2 = 0.8
power.0.8 <- summarized %>%
  dplyr::mutate(h2 = as.numeric(levels(h2))[h2]) %>%
  dplyr::filter(h2 == 0.8) %>%
  ggplot(., mapping = aes(x = reorder(strain_set,Power), y = mean.interval.Mb, colour = Power)) +
  theme_bw() +
  geom_pointrange(aes(ymin = mean.interval.Mb-sd.interval.Mb,
                      ymax = mean.interval.Mb+sd.interval.Mb), alpha = 0.7) +
  scale_colour_distiller(type = "seq", direction = 1, palette = "YlGnBu") + 
  facet_grid(.~set.category, scales = "free", space = "free") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  labs(title = "GWA Mapping Resolution: h2 = 0.8",
       x = "Strain Set",
       y = "Mean QTL Interval Width (Mb)")

precision.0.8 <- summarized %>%
  dplyr::mutate(h2 = as.numeric(levels(h2))[h2]) %>%
  dplyr::filter(h2 == 0.8) %>%
  ggplot(., mapping = aes(x = reorder(strain_set,Precision), y = mean.interval.Mb, colour = Precision)) +
  theme_bw() +
  geom_pointrange(aes(ymin = mean.interval.Mb-sd.interval.Mb,
                      ymax = mean.interval.Mb+sd.interval.Mb), alpha = 0.7) +
  scale_colour_distiller(type = "seq", direction = 1, palette = "YlOrBr") + 
  facet_grid(.~set.category, scales = "free", space = "free") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  labs(title = "GWA Mapping Resolution: h2 = 0.8",
       x = "Strain Set",
       y = "Mean QTL Interval Width (Mb)")

cowplot::plot_grid(power.0.5, precision.0.5, ncol = 1)
cowplot::plot_grid(power.0.8, precision.0.8, ncol = 1)
```

## Percent of Phenotypic Variance Explained by QTL

```{r pve plots, echo=FALSE, fig.height=6, fig.width=7}
# h2 = 0.5
power.0.5 <- summarized %>%
  dplyr::mutate(h2 = as.numeric(levels(h2))[h2]) %>%
  dplyr::filter(h2 == 0.5) %>%
  ggplot(., mapping = aes(x = reorder(strain_set,Power), y = mean.PVE, colour = Power)) +
  theme_bw() +
  geom_pointrange(aes(ymin = mean.PVE-sd.PVE,
                      ymax = mean.PVE+sd.PVE), alpha = 0.7) +
  scale_colour_distiller(type = "seq", direction = 1, palette = "YlGnBu") + 
  facet_grid(.~set.category, scales = "free", space = "free") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  labs(title = "Variance Explained: h2 = 0.5",
       x = "Strain Set",
       y = "PVE (%)")

precision.0.5 <- summarized %>%
  dplyr::mutate(h2 = as.numeric(levels(h2))[h2]) %>%
  dplyr::filter(h2 == 0.5) %>%
  ggplot(., mapping = aes(x = reorder(strain_set,Precision), y = mean.PVE, colour = Precision)) +
  theme_bw() +
  geom_pointrange(aes(ymin = mean.PVE-sd.PVE,
                      ymax = mean.PVE+sd.PVE), alpha = 0.7) +
  scale_colour_distiller(type = "seq", direction = 1, palette = "YlOrBr") + 
  facet_grid(.~set.category, scales = "free", space = "free") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  labs(title = "Variance Explained: h2 = 0.5",
       x = "Strain Set",
       y = "PVE (%)")

# h2 = 0.8
power.0.8 <- summarized %>%
  dplyr::mutate(h2 = as.numeric(levels(h2))[h2]) %>%
  dplyr::filter(h2 == 0.8) %>%
  ggplot(., mapping = aes(x = reorder(strain_set,Power), y = mean.PVE, colour = Power)) +
  theme_bw() +
  geom_pointrange(aes(ymin = mean.PVE-sd.PVE,
                      ymax = mean.PVE+sd.PVE), alpha = 0.7) +
  scale_colour_distiller(type = "seq", direction = 1, palette = "YlGnBu") + 
  facet_grid(.~set.category, scales = "free", space = "free") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  labs(title = "Variance Explained: h2 = 0.8",
       x = "Strain Set",
       y = "PVE (%)")

precision.0.8 <- summarized %>%
  dplyr::mutate(h2 = as.numeric(levels(h2))[h2]) %>%
  dplyr::filter(h2 == 0.8) %>%
  ggplot(., mapping = aes(x = reorder(strain_set,Precision), y = mean.PVE, colour = Precision)) +
  theme_bw() +
  geom_pointrange(aes(ymin = mean.PVE-sd.PVE,
                      ymax = mean.PVE+sd.PVE), alpha = 0.7) +
  scale_colour_distiller(type = "seq", direction = 1, palette = "YlOrBr") + 
  facet_grid(.~set.category, scales = "free", space = "free") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  labs(title = "Variance Explained: h2 = 0.8",
       x = "Strain Set",
       y = "PVE (%)")

cowplot::plot_grid(power.0.5, precision.0.5, ncol = 1)
cowplot::plot_grid(power.0.8, precision.0.8, ncol = 1)
```

## Population Genomic Differences Between CeNDR Strain Sets

```{r sweptness, echo=FALSE, fig.height=6, fig.width=7}
# Lifted from generate_strain_lists_swept.R (10/02/2020)
# For two-set combinations only so far
strain.data <- data.table::fread("data/CelegansStrainData.tsv") 
sweeps <- data.table::fread("data/sweep_summary.tsv")
metadata <- sweeps %>%
  dplyr::full_join(., strain.data) %>%
  dplyr::mutate(I = dplyr::if_else(condition = I == TRUE, 
                                   true = "Swept", 
                                   false = "Unswept"),
                II = dplyr::if_else(condition = II == TRUE, 
                                    true = "Swept", 
                                    false = "Unswept"),
                III = dplyr::if_else(condition = III == TRUE, 
                                     true = "Swept", 
                                     false = "Unswept"),
                IV = dplyr::if_else(condition = IV == TRUE, 
                                    true = "Swept", 
                                    false = "Unswept"),
                V = dplyr::if_else(condition = V == TRUE, 
                                   true = "Swept", 
                                   false = "Unswept"),
                X = dplyr::if_else(condition = X == TRUE, 
                                   true = "Swept", 
                                   false = "Unswept"))

cendr.strain.sets.nested <- strain.data %>%
  dplyr::select(isotype, strain_set) %>%
  dplyr::filter(strain_set %in% c(1:8)) %>%
  dplyr::arrange(strain_set) %>%
  dplyr::group_by(strain_set) %>%
  tidyr::nest()

strain.sets <- c(c(1:8),c(1:8))
strain.sets <- unique(strain.sets)
gather.strain.sets <- t(combn(strain.sets, 2)) %>%
  as.data.frame() %>%
  `colnames<-`(c("set1","set2"))

cendr.strain.set.sweptness <- list()
for(i in 1:nrow(gather.strain.sets)){
  combo <- gather.strain.sets[i,]
  
  set1 <- as.numeric(combo$set1)
  set2 <- as.numeric(combo$set2)
  combo.name <- paste("cendr.set",set1,"set",set2,sep = ".")

  combined.strain.sets <- data.frame(cendr.strain.sets.nested$data[set1]) %>%
    dplyr::full_join(.,data.frame(cendr.strain.sets.nested$data[set2])) %>%
    dplyr::left_join(.,metadata) %>%
    dplyr::select(isotype, I, II, III, IV, V, X, swept_chroms) %>%
    dplyr::mutate(strain_set = combo.name)
  
  cendr.strain.set.sweptness[[i]] <- combined.strain.sets
}
cendr.strain.sets.df <- do.call(rbind, cendr.strain.set.sweptness) %>%
  dplyr::full_join(., summarized) %>%
  dplyr::select(-c(II, III)) %>%
  tidyr::pivot_longer(cols = c(I, IV, V, X), names_to = c("Swept.Chr"), values_to = c("Status")) 

swept.power.0.5 <- cendr.strain.sets.df %>%
  dplyr::group_by(strain_set, h2) %>%
  dplyr::summarise(mean(swept_chroms), sd(swept_chroms)) %>%
  dplyr::filter(h2 == 0.5) %>%
  dplyr::left_join(., summarized) %>%
  dplyr::filter(set.category == "CeNDR Subset") %>%
  ggplot(., mapping = aes(x = reorder(strain_set, Power), y = `mean(swept_chroms)`, colour = Power)) +
  theme_bw() + 
  geom_pointrange(aes(ymin = `mean(swept_chroms)`-`sd(swept_chroms)`,
                      ymax = `mean(swept_chroms)`+`sd(swept_chroms)`)) + 
  scale_colour_distiller(type = "seq", direction = 1, palette = "YlGnBu") + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  labs(y = "Average Number of Swept Chromosomes per Strain",
       x = "Strain Set",
       title = "Swept Chromosomes: h2 = 0.5")

swept.precision.0.5 <- cendr.strain.sets.df %>%
  dplyr::group_by(strain_set, h2) %>%
  dplyr::summarise(mean(swept_chroms), sd(swept_chroms)) %>%
  dplyr::filter(h2 == 0.5) %>%
  dplyr::left_join(., summarized) %>%
  dplyr::filter(set.category == "CeNDR Subset") %>%
  ggplot(., mapping = aes(x = reorder(strain_set, Precision), y = `mean(swept_chroms)`, colour = Precision)) +
  theme_bw() + 
  geom_pointrange(aes(ymin = `mean(swept_chroms)`-`sd(swept_chroms)`,
                      ymax = `mean(swept_chroms)`+`sd(swept_chroms)`)) + 
  scale_colour_distiller(type = "seq", direction = 1, palette = "YlOrBr") + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  labs(y = "Average Number of Swept Chromosomes per Strain",
       x = "Strain Set",
       title = "Swept Chromosomes: h2 = 0.5")



swept.power.0.8 <- cendr.strain.sets.df %>%
  dplyr::group_by(strain_set, h2) %>%
  dplyr::summarise(mean(swept_chroms), sd(swept_chroms)) %>%
  dplyr::filter(h2 == 0.8) %>%
  dplyr::left_join(., summarized) %>%
  dplyr::filter(set.category == "CeNDR Subset") %>%
  ggplot(., mapping = aes(x = reorder(strain_set, Power), y = `mean(swept_chroms)`, colour = Power)) +
  theme_bw() + 
  geom_pointrange(aes(ymin = `mean(swept_chroms)`-`sd(swept_chroms)`,
                      ymax = `mean(swept_chroms)`+`sd(swept_chroms)`)) + 
  scale_colour_distiller(type = "seq", direction = 1, palette = "YlGnBu") + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  labs(y = "Average Number of Swept Chromosomes per Strain",
       x = "Strain Set",
       title = "Swept Chromosomes: h2 = 0.8")

swept.precision.0.8 <- cendr.strain.sets.df %>%
  dplyr::group_by(strain_set, h2) %>%
  dplyr::summarise(mean(swept_chroms), sd(swept_chroms)) %>%
  dplyr::filter(h2 == 0.8) %>%
  dplyr::left_join(., summarized) %>%
  dplyr::filter(set.category == "CeNDR Subset") %>%
  ggplot(., mapping = aes(x = reorder(strain_set, Precision), y = `mean(swept_chroms)`, colour = Precision)) +
  theme_bw() + 
  geom_pointrange(aes(ymin = `mean(swept_chroms)`-`sd(swept_chroms)`,
                      ymax = `mean(swept_chroms)`+`sd(swept_chroms)`)) + 
  scale_colour_distiller(type = "seq", direction = 1, palette = "YlOrBr") + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  labs(y = "Average Number of Swept Chromosomes per Strain",
       x = "Strain Set",
       title = "Swept Chromosomes: h2 = 0.8")
  
cowplot::plot_grid(swept.power.0.5, swept.precision.0.5, ncol = 1)
cowplot::plot_grid(swept.precision.0.5, swept.precision.0.8, ncol = 1)
```