---
title: "Population Differences in Mapping Performance"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: true
    toc_float: true
    toc_collapsed: true
    toc_depth: 4
editor_options:
  chunk_output_type: console
---
```{r setup, message=FALSE, warning=FALSE, include=FALSE}
require(tidyverse)
require(tidymodels)
require(nationalparkcolors)
require(RColorBrewer)
require(GenomicRanges)
require(cowplot)
require(workflowr)
require(ggbeeswarm)
require(wesanderson)
```

## Analysis date: `r format(Sys.time(), '%B %d, %Y')`

```{r message=FALSE, warning=FALSE, include=FALSE}
setwd("~/Documents/projects/NemaScan_Performance/")
# setwd("~/Documents/AndersenLab/NemaScan_Performance/")
############################
### COMPLETE SUBSAMPLING ###
############################

# load(file = "data/NemaScan_Performance.CeNDR2020_population_subsampling.20200925.RData")
load(file = "data/NemaScan_Performance.Demography.SampleSize.Complete.Subsampled.20201230.RData")
dat.complete <- simulation.metrics.df %>%
  tidyr::separate(col = sim,
                  into = c("nQTL","Rep","h2","MAF","effect_range","strain_set"), 
                  sep = "_", remove = F) %>%
  tidyr::separate(col = QTL,
                  into = c("CHROM","POS"), 
                  sep = ":", remove = F) %>%
  dplyr::mutate(h2 = as.factor(h2),
                nQTL = as.factor(nQTL),
                POS = as.numeric(POS),
                startPOS = as.numeric(startPOS),
                peakPOS = as.numeric(peakPOS),
                endPOS = as.numeric(endPOS),
                var.exp  = as.numeric(var.exp), # watch
                Simulated.QTL.VarExp = as.numeric(Simulated.QTL.VarExp), # watch
                peak_id = as.numeric(peak_id),
                BETA = as.numeric(BETA),
                Effect = as.numeric(Effect),
                Frequency = as.numeric(Frequency),
                log10p = as.numeric(log10p),
                interval_size = as.numeric(interval_size)) %>%
  dplyr::filter(!algorithm %in% c("No Successful Mappings"),
                algorithm == "MIXED",
                CHROM != 7) %>%
  dplyr::mutate(population.group = "Complete Subsampling")


###########################
### UNSWEPT SUBSAMPLING ###
###########################
load(file = "data/NemaScan_Performance.Demography.SampleSize.Unswept.Subsampled.20210111.RData")
dat.unswept <- simulation.metrics.df %>%
  tidyr::separate(col = sim,
                  into = c("nQTL","Rep","h2","MAF","effect_range","strain_set"), 
                  sep = "_", remove = F) %>%
  tidyr::separate(col = QTL,
                  into = c("CHROM","POS"), 
                  sep = ":", remove = F) %>%
  dplyr::mutate(h2 = as.factor(h2),
                nQTL = as.factor(nQTL),
                POS = as.numeric(POS),
                startPOS = as.numeric(startPOS),
                peakPOS = as.numeric(peakPOS),
                endPOS = as.numeric(endPOS),
                var.exp  = as.numeric(var.exp), # watch
                Simulated.QTL.VarExp = as.numeric(Simulated.QTL.VarExp), # watch
                peak_id = as.numeric(peak_id),
                BETA = as.numeric(BETA),
                Effect = as.numeric(Effect),
                Frequency = as.numeric(Frequency),
                log10p = as.numeric(log10p),
                interval_size = as.numeric(interval_size)) %>%
  dplyr::filter(!algorithm %in% c("No Successful Mappings"),
                algorithm == "MIXED",
                CHROM != 7) %>%
  dplyr::mutate(population.group = "Unswept Subsampling")

dat <- dat.complete %>%
  dplyr::full_join(., dat.unswept)

finished.pops <- dat %>%
  dplyr::filter(Simulated == TRUE) %>%
  dplyr::group_by(strain_set, population.group) %>%
  dplyr::summarise(n()) %>%
  dplyr::filter(`n()` == 30) %>%
  dplyr::select(strain_set)

dat.curated <- dat %>%
  dplyr::filter(strain_set %in% finished.pops$strain_set)

  

#########################
## POPULATION FEATURES ##
#########################

load("data/NemaScan_Population_Metrics.Demography.SampleSize.Complete.Subsampled.20210111.RData")
population.features.complete <- population.metrics.df %>%
  dplyr::filter(!duplicated(strain_set)) %>%
  dplyr::mutate(population.group = "Complete Subsampling")

load("data/NemaScan_Population_Metrics.Demography.SampleSize.Unswept.Subsampled.20210111.RData")
population.features.unswept <- population.metrics.df %>%
  dplyr::filter(!duplicated(strain_set)) %>%
  dplyr::mutate(population.group = "Unswept Subsampling")

population.features <- population.features.complete %>%
  dplyr::full_join(., population.features.unswept) %>%
  dplyr::filter(strain_set %in% dat.curated$strain_set)

#######################
## Haplotype Sharing ##
#######################

hapshare.nested <- population.features %>%
  dplyr::select(strain_set, population.group, contains("hapshare")) %>%
  tidyr::pivot_longer(cols = contains("hapshare"), names_to = "CHROM.HAPSHARE") %>%
  tidyr::separate(col = CHROM.HAPSHARE, c("stat","chrom","hapshare")) %>%
  dplyr::filter(stat == "median") %>%
  dplyr::select(-hapshare) %>%
  dplyr::group_by(chrom) %>%
  tidyr::nest()
hapshare.quantiles <- function(data, chrom){
  data %>%
    dplyr::mutate(quintile = cut(value, breaks = 7, include.lowest = T),
                  chrom = chrom)
}
population.hapshare.quantiles <- pmap(.l = list(hapshare.nested$data, 
                                                hapshare.nested$chrom), 
                                      .f = hapshare.quantiles) %>%
  Reduce(rbind,.) %>%
  tidyr::pivot_wider(names_from = c(stat, chrom), values_from = c(value, quintile)) %>%
  dplyr::select(strain_set, population.group, contains("quintile"))

###################################
## Chromosome Sweptness Metadata ##
###################################

chrom.swept.nested <- population.features %>%
  dplyr::select(strain_set, population.group, I, IV, V, X) %>%
    tidyr::pivot_longer(cols = c(I, IV, V, X), 
                        names_to = "CHROM.SWEPT", 
                        values_to = "PCT.SWEPT") %>%
  dplyr::group_by(CHROM.SWEPT) %>%
  tidyr::nest()
chrom.swept.quantiles <- function(data, CHROM.SWEPT){
  data %>%
    dplyr::mutate(quintile = cut(PCT.SWEPT, breaks = 7),
                  CHROM.SWEPT = CHROM.SWEPT)
}
population.swept.quantiles <- pmap(.l = list(chrom.swept.nested$data, 
                                             chrom.swept.nested$CHROM.SWEPT), 
                                      .f = chrom.swept.quantiles) %>%
  Reduce(rbind,.) %>%
  tidyr::pivot_wider(names_from = CHROM.SWEPT, values_from = c(PCT.SWEPT, quintile)) %>%
  dplyr::select(strain_set, population.group, contains("quintile"))


pop.size <- population.features %>%
  dplyr::select(strain_set, population.group, pop.size)
population.quantiles <- population.hapshare.quantiles %>%
  dplyr::full_join(., population.swept.quantiles)


dat.population.features <- dat.curated %>%
  dplyr::left_join(., population.quantiles) %>%
  dplyr::full_join(., pop.size)



#######################
## Divergent Regions ##
#######################
common.divergent.regions <- data.table::fread("data/Common_divergent_regions_clustered.tsv") %>%
  dplyr::mutate(chr = as.factor(chr))
levels(common.divergent.regions$chr) <- c("1","2","3","4","5","6")
common.div.regions.gr <- GenomicRanges::GRanges(seqnames = common.divergent.regions$chr, 
                                         ranges = IRanges(start = common.divergent.regions$start, 
                                                          end = common.divergent.regions$stop))

intermediate.divergent.regions <- data.table::fread("data/Intermediate_divergent_regions_clustered.tsv") %>%
  dplyr::mutate(chr = as.factor(chr))
levels(intermediate.divergent.regions$chr) <- c("1","2","3","4","5","6")
intermediate.div.regions.gr <- GenomicRanges::GRanges(seqnames = intermediate.divergent.regions$chr, 
                                         ranges = IRanges(start = intermediate.divergent.regions$start, 
                                                          end = intermediate.divergent.regions$stop))

rare.divergent.regions <- data.table::fread("data/Rare_divergent_regions_clustered.tsv") %>%
  dplyr::mutate(chr = as.factor(chr))
levels(rare.divergent.regions$chr) <- c("1","2","3","4","5","6")
rare.div.regions.gr <- GenomicRanges::GRanges(seqnames = rare.divergent.regions$chr, 
                                         ranges = IRanges(start = rare.divergent.regions$start, 
                                                          end = rare.divergent.regions$stop))

all.gr <- GenomicRanges::GRanges(seqnames = dat.population.features$CHROM, 
                                 ranges = IRanges(start = dat.population.features$POS, 
                                                  end = dat.population.features$POS),
                                 QTL = dat.population.features$QTL)

common.divergent.QTL <- IRanges::findOverlapPairs(all.gr, common.div.regions.gr) %>%
  data.frame() %>%
  dplyr::select(first.QTL, first.X.seqnames, first.X.start, second.start, second.end) %>%
  `colnames<-`(c("QTL","CHROM","POS","div.start","div.end"))

intermediate.divergent.QTL <- IRanges::findOverlapPairs(all.gr, intermediate.div.regions.gr) %>%
  data.frame() %>%
  dplyr::select(first.QTL, first.X.seqnames, first.X.start, second.start, second.end) %>%
  `colnames<-`(c("QTL","CHROM","POS","div.start","div.end"))

rare.divergent.QTL <- IRanges::findOverlapPairs(all.gr, rare.div.regions.gr) %>%
  data.frame() %>%
  dplyr::select(first.QTL, first.X.seqnames, first.X.start, second.start, second.end) %>%
  `colnames<-`(c("QTL","CHROM","POS","div.start","div.end"))

dat.population.features.divergence <- dat.population.features %>%
  dplyr::mutate(common.divergent = ifelse(test = QTL %in% common.divergent.QTL$QTL, 
                                   yes = "Divergent", 
                                   no = "Non-Divergent"),
                intermediate.divergent = ifelse(test = QTL %in% intermediate.divergent.QTL$QTL, 
                                   yes = "Divergent", 
                                   no = "Non-Divergent"),
                rare.divergent = ifelse(test = QTL %in% rare.divergent.QTL$QTL,
                                   yes = "Divergent", 
                                   no = "Non-Divergent"))


####################
## ARMS & CENTERS ##
####################
genome.borders <- data.table::fread("data/genome.bed.tsv")
genome.borders$chr <- seq(1,6,1)
genome.borders <- genome.borders %>%
  dplyr::rename(CHROM = chr, startPOS = start, endPOS = stop)

arms.centers <- data.table::fread("data/ARMS_CENTERS.tsv") %>%
  dplyr::rename(CHROM = V1, startPOS = V2, endPOS = V3, chrom.region = V4) %>%
  dplyr::mutate(orientation = rep(c("LEFT","CENTER","RIGHT"), 6),
                CHROM = as.factor(CHROM))
levels(arms.centers$CHROM) <- c(seq(1,6))
for(i in 1:nrow(arms.centers)){
  if(arms.centers[i,]$orientation == "LEFT"){
    arms.centers[i,]$startPOS <- 0
  } else if(arms.centers[i,]$orientation == "RIGHT"){
    arms.centers[i,]$endPOS <- genome.borders %>%
      dplyr::filter(CHROM == arms.centers[i,]$CHROM) %>%
      dplyr::select(endPOS) %>% as.numeric()
  } else {
    next
  }
}
arms.centers.clean <- arms.centers %>%
  tidyr::unite("CHROM.REGION",c(orientation, chrom.region)) %>%
  dplyr::mutate(CHROM.REGION = if_else(CHROM.REGION == "CENTER_CENTER",
                                       true = "CENTER", 
                                       false = CHROM.REGION))

arms.centers.gr <- GenomicRanges::GRanges(seqnames = arms.centers.clean$CHROM, 
                                         ranges = IRanges(start = arms.centers.clean$startPOS, 
                                                          end = arms.centers.clean$endPOS),
                                         CHROM.REGION = arms.centers.clean$CHROM.REGION)

arm.center.QTL <- IRanges::findOverlapPairs(all.gr, arms.centers.gr) %>%
  data.frame() %>%
  dplyr::select(first.QTL, first.X.seqnames,  second.CHROM.REGION) %>%
  `colnames<-`(c("QTL","CHROM","CHROM.REGION"))


dat.population.features.divergence <- dat.population.features.divergence %>%
  dplyr::left_join(., arm.center.QTL) %>%
  dplyr::mutate(CHROM.REGION = as.factor(CHROM.REGION))
levels(dat.population.features.divergence$CHROM.REGION) <- c("Center","Left Arm", "Right Arm")
dat.population.features.divergence$CHROM.REGION <- factor(dat.population.features.divergence$CHROM.REGION, levels = c("Left Arm","Center","Right Arm"))
```

## Simulation Parameters

NemaScan simulation performance was assessed with the following experimental parameters:

* ###### Number of Simulated QTL: ```r levels(as.factor(dat.population.features$nQTL))```
* ###### Sample Population(s): ```r length(levels(as.factor(dat.population.features$strain_set)))```
* ###### Number of Replicates per Regime: ```r max(as.numeric(dat.population.features$Rep))```
* ###### QTL Effect Range: ```r levels(as.factor(dat.population.features$effect_range))```

## Plots

```{r echo=FALSE}
pop.group.pal <- wes_palette(name = "Royal1", n = 2)

dat.population.features.divergence %>%
  dplyr::filter(Simulated == TRUE,
                !duplicated(QTL), 
                !is.na(CHROM)) %>%
  ggplot(mapping = aes(x = POS/1000000, y = Effect)) + 
  theme_bw() + 
  geom_point(alpha = 0.5) + 
  facet_grid(population.group~CHROM, drop = TRUE, scales = "free") + 
  labs(x = "Genomic position (Mb)",
       y = "Causal QTL Effect",
       title = "Where were QTL Simulated?")

dat.population.features.divergence %>%
  dplyr::filter(Simulated == TRUE,
                !is.na(CHROM),
                !duplicated(QTL)) %>%
  ggplot(mapping = aes(x = Frequency, fill = population.group)) +
  geom_density(alpha = 0.6) + 
  theme_bw() +
  facet_wrap(.~pop.size) + 
  scale_fill_manual(values = pop.group.pal, name = "Population Group") + 
  theme(legend.position = "top") + 
  labs(x = "Minor Allele Frequency",
       y = "Frequency of Simulated QTL (Smoothed Density)")

dat.population.features.divergence %>%
  dplyr::filter(Simulated == TRUE,
                !is.na(CHROM),
                !duplicated(QTL)) %>%
  ggplot(., mapping = aes(x = Simulated.QTL.VarExp*100, fill = population.group)) +
  theme_bw() + 
  geom_density(alpha = 0.6) + 
  theme_bw() +
  facet_wrap(.~pop.size) + 
  scale_fill_manual(values = pop.group.pal, name = "Population Group") + 
  theme(legend.position = "top") + 
  labs(x = "Variance Explained by Simulated QTL (%)",
       y = "Frequency of Simulated QTL (Smoothed Density)")

dat.population.features.divergence %>%
  dplyr::filter(Simulated == TRUE,
                !is.na(CHROM),
                !duplicated(QTL), 
                !is.na(CHROM.REGION)) %>%
  dplyr::group_by(CHROM.REGION, pop.size, population.group, CHROM) %>%
  dplyr::summarise(n()) %>%
  ggplot(., mapping = aes(x = as.factor(pop.size), y = `n()`, fill = population.group)) + 
  theme_bw() + 
  geom_col() + 
  facet_grid(CHROM ~ CHROM.REGION, scales = "free") + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
        legend.position = "top") + 
  scale_fill_manual(values = pop.group.pal, name = "Population Group") + 
  labs(x = "Sample Size",
       y = "Number of Simulated QTL")
  


dat.population.features.divergence %>%
  dplyr::select(strain_set, population.group, pop.size) %>%
  dplyr::filter(!duplicated(.)) %>%
  dplyr::group_by(population.group, pop.size) %>%
  dplyr::summarise(n()) %>%
  dplyr::arrange(pop.size) %>%
  tidyr::pivot_wider(names_from = pop.size, values_from = `n()`)

dat.population.features.divergence %>%
  dplyr::select(strain_set, quintile_I) %>%
  dplyr::filter(!duplicated(.)) %>%
  dplyr::group_by(quintile_I) %>%
  dplyr::summarise(n())

dat.population.features.divergence %>%
  dplyr::select(strain_set, quintile_IV) %>%
  dplyr::filter(!duplicated(.)) %>%
  dplyr::group_by(quintile_IV) %>%
  dplyr::summarise(n())

dat.population.features.divergence %>%
  dplyr::select(strain_set, quintile_V) %>%
  dplyr::filter(!duplicated(.)) %>%
  dplyr::group_by(quintile_V) %>%
  dplyr::summarise(n())

dat.population.features.divergence %>%
  dplyr::select(strain_set, quintile_X) %>%
  dplyr::filter(!duplicated(.)) %>%
  dplyr::group_by(quintile_X) %>%
  dplyr::summarise(n())

population.features %>%
  tidyr::pivot_longer(cols = c(I, IV, V, X), names_to = "CHROM.SWEPT", values_to = "PCT.SWEPT") %>%
  ggplot(., mapping = aes(x = PCT.SWEPT*100, fill = population.group)) +
  theme_bw() + 
  geom_density() +
  facet_grid(pop.size~CHROM.SWEPT, scales = "free_y") + 
  scale_fill_manual(values = pop.group.pal, name = "Population Group") + 
  theme(legend.position = "top") + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  labs(x = "Percent of Strains Containing 'Swept' Chromosome",
       y = "Density")

population.features %>%
  tidyr::pivot_longer(cols = contains("hapshare"), names_to = "CHROM.HAPSHARE") %>%
  tidyr::separate(col = CHROM.HAPSHARE, c("stat","chrom","hapshare")) %>%
  dplyr::select(-hapshare) %>%
  dplyr::filter(stat == "mean") %>%
  ggplot(., mapping = aes(x = value*100, fill = population.group)) +
  theme_bw() + 
  geom_density() + 
  scale_fill_manual(values = pop.group.pal, name = "Population Group") + 
  facet_grid(pop.size~chrom, scales = "free") + 
  theme(legend.position = "top") + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  labs(x = "Percent of Chromosome Containing Swept Haplotype",
       y = "Density")

A <- dat.population.features.divergence %>%
  dplyr::select(QTL, strain_set, Simulated, Detected, population.group, pop.size, common.divergent, intermediate.divergent, rare.divergent) %>%
  tidyr::pivot_longer(cols = contains("divergent"), names_to = "Divergent.Region.Type", values_to = "Divergence") %>%
  dplyr::group_by(strain_set, pop.size, population.group, Divergent.Region.Type, Divergence) %>%
  dplyr::summarize(n()) %>%
  tidyr::pivot_wider(names_from = Divergence, values_from = `n()`) %>%
  dplyr::mutate(Percent.Divergent.QTL = Divergent/(Divergent +`Non-Divergent`),
                Divergent.Region.Type = as.factor(Divergent.Region.Type)) %>%
  dplyr::select(-c(Divergent,`Non-Divergent`))
levels(A$Divergent.Region.Type) <- c("Common","Intermediate","Rare")
A[is.na(A$Percent.Divergent.QTL),]$Percent.Divergent.QTL <- 0
ggplot(A, mapping = aes(x = Percent.Divergent.QTL*100, fill = population.group)) + 
  theme_bw() + 
  geom_density(alpha = 0.7) + 
  scale_fill_manual(values = pop.group.pal, name = "Population Group") + 
  facet_grid(pop.size~Divergent.Region.Type, scales = "free") + 
  theme(legend.position = "top") + 
  labs(x = "Percent of QTL Within Divergent Regions Per Population",
       y = "Density")
  
B <- dat.population.features.divergence %>%
  dplyr::select(QTL, strain_set, Simulated, Detected, population.group, pop.size, common.divergent, intermediate.divergent, rare.divergent) %>%
  tidyr::pivot_longer(cols = contains("divergent"), names_to = "Divergent.Region.Type", values_to = "Divergence") %>%
  dplyr::group_by(pop.size, population.group, Divergent.Region.Type, Divergence) %>%
  dplyr::summarize(n()) %>%
  tidyr::pivot_wider(names_from = Divergence, values_from = `n()`) %>%
  dplyr::mutate(Percent.Divergent.QTL = Divergent/(Divergent +`Non-Divergent`),
                Divergent.Region.Type = as.factor(Divergent.Region.Type)) %>%
  dplyr::select(-c(Divergent,`Non-Divergent`))

levels(B$Divergent.Region.Type) <- c("Common","Intermediate","Rare")
ggplot(B, mapping = aes(x = as.factor(pop.size), y = Percent.Divergent.QTL, 
                          fill = population.group,
                          group  = population.group)) + 
  theme_bw() + 
  geom_line(mapping = aes(colour = population.group)) +
  geom_point(shape = 21) + 
  scale_fill_manual(values = pop.group.pal, name = "Population Group") + 
  scale_colour_manual(values = pop.group.pal, name = "Population Group") + 
  theme(legend.position = "top") + 
  facet_grid(.~Divergent.Region.Type, scales = "free") +
  labs(x = "Sample Size",
       y = "Percent of QTL Within Divergent Regions")

```

## Performance

```{r echo=FALSE, warning=FALSE}
pop.size <- dat.population.features.divergence %>%
  dplyr::select(strain_set, population.group, pop.size) %>%
  dplyr::filter(!duplicated(.))


power.popgroup.samplesize <- dat.population.features.divergence %>%
  dplyr::group_by(strain_set, population.group) %>%
  yardstick::sens(truth = Simulated, estimate = Detected) %>%
  dplyr::mutate(.metric = "Power") %>%
  dplyr::left_join(., pop.size) %>%
  dplyr::group_by(population.group, pop.size) %>%
  dplyr::summarise(mean = mean(.estimate),
                   sd = sd(.estimate),
                   n = n()) %>%
  dplyr::mutate(b = mean + sd,
                a = mean - sd,
                ymax = if_else(condition = b > 1, 
                             true = 1-mean, 
                             false = sd),
                ymin = if_else(condition = a < 0, 
                             true = (0-mean)*-1, 
                             false = sd)) %>%
  dplyr::select(-a,-b)
ggplot(power.popgroup.samplesize, mapping = aes(x = as.factor(pop.size), y = mean, 
                                                fill = population.group,
                                                group = population.group)) + 
  theme_bw() + 
  geom_point(position = position_dodge(width = 0.5), shape = 21) +
  geom_errorbar(data = power.popgroup.samplesize,
                width = 0.5,
                mapping = aes(y = mean, ymax = mean+ymax, ymin = mean-ymin, colour = population.group),
                position=position_dodge(width=0.5)) + 
  geom_line(position=position_dodge(width=0.5), mapping = aes(colour = population.group)) +
  scale_fill_manual(values = pop.group.pal, name = "Population Group") + 
  scale_colour_manual(values = pop.group.pal, name = "Population Group") + 
  theme(legend.position = "top") + 
  ylim(c(0,1)) + 
  labs(x = "Sample Size",
       y = "Power")



power.popgroup.samplesize.divergence <- dat.population.features.divergence %>%
  dplyr::group_by(strain_set, population.group, common.divergent) %>%
  yardstick::sens(truth = Simulated, estimate = Detected) %>%
  dplyr::mutate(.metric = "Power") %>%
  dplyr::left_join(., pop.size) %>%
  dplyr::filter(!is.na(.estimate)) %>%
  dplyr::group_by(population.group, common.divergent, pop.size) %>%
  dplyr::summarise(mean = mean(.estimate),
                   sd = sd(.estimate),
                   n()) %>%
  dplyr::mutate(b = mean + sd,
                a = mean - sd,
                ymax = if_else(condition = b > 1, 
                             true = 1-mean, 
                             false = sd),
                ymin = if_else(condition = a < 0, 
                             true = (0-mean)*-1, 
                             false = sd)) %>%
  dplyr::select(-a,-b)
ggplot(power.popgroup.samplesize.divergence, mapping = aes(x = as.factor(pop.size), y = mean, 
                                                fill = population.group,
                                                group = interaction(population.group, common.divergent))) + 
  theme_bw() + 
  geom_point(position = position_dodge(width = 0.5), shape = 21) +
  geom_errorbar(data = power.popgroup.samplesize.divergence,
                width = 0.5,
                mapping = aes(y = mean, ymax = mean+ymax, ymin = mean-ymin, colour = population.group),
                position=position_dodge(width=0.5)) + 
  geom_line(position=position_dodge(width=0.5), mapping = aes(colour = population.group)) +
  facet_grid(.~common.divergent) + 
  scale_fill_manual(values = pop.group.pal, name = "Population Group") + 
  scale_colour_manual(values = pop.group.pal, name = "Population Group") + 
  theme(legend.position = "top") + 
  labs(x = "Sample Size",
       y = "Power")




power.popgroup.quintileI <- dat.population.features.divergence %>%
  dplyr::filter(CHROM == 1) %>%
  dplyr::group_by(strain_set, population.group, quintile_I) %>%
  yardstick::sens(truth = Simulated, estimate = Detected) %>%
  dplyr::mutate(.metric = "Power") %>%
  dplyr::filter(!is.na(.estimate)) %>%
  dplyr::mutate(quintile_I = gsub(as.character(quintile_I), pattern = "\\(", replacement = "") %>%
                  gsub(., pattern = "\\]", replacement = "") %>%
                  gsub(., pattern = ",", replacement = "-"))
ggplot(power.popgroup.quintileI, mapping = aes(x = quintile_I, y = .estimate, 
                                                fill = quintile_I)) + 
  theme_bw() + 
  geom_violin(draw_quantiles = 0.5, scale = "count") + 
  scale_fill_brewer(palette = "RdBu", name = "Population Composition") +
  theme(legend.position = "top",
        axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  labs(x = "Proportion of Population with Swept Chromosome I",
       y = "Power")


power.popgroup.quintileIV <- dat.population.features.divergence %>%
  dplyr::filter(CHROM == 4) %>%
  dplyr::group_by(strain_set, population.group, quintile_IV) %>%
  yardstick::sens(truth = Simulated, estimate = Detected) %>%
  dplyr::mutate(.metric = "Power") %>%
  dplyr::filter(!is.na(.estimate)) %>%
  dplyr::mutate(quintile_IV = gsub(as.character(quintile_IV), pattern = "\\(", replacement = "") %>%
                  gsub(., pattern = "\\]", replacement = "") %>%
                  gsub(., pattern = ",", replacement = "-"))
ggplot(power.popgroup.quintileIV, mapping = aes(x = quintile_IV, y = .estimate, 
                                                fill = quintile_IV)) + 
  theme_bw() + 
  geom_violin(draw_quantiles = 0.5, scale = "count") + 
  scale_fill_brewer(palette = "RdBu", name = "Population Composition") +
  theme(legend.position = "top",
        axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  labs(x = "Proportion of Population with Swept Chromosome IV",
       y = "Power")



power.popgroup.quintileV <- dat.population.features.divergence %>%
  dplyr::filter(CHROM == 5) %>%
  dplyr::group_by(strain_set, population.group, quintile_V) %>%
  yardstick::sens(truth = Simulated, estimate = Detected) %>%
  dplyr::mutate(.metric = "Power") %>%
  dplyr::filter(!is.na(.estimate)) %>%
  dplyr::mutate(quintile_V = gsub(as.character(quintile_V), pattern = "\\(", replacement = "") %>%
                  gsub(., pattern = "\\]", replacement = "") %>%
                  gsub(., pattern = ",", replacement = "-"))
ggplot(power.popgroup.quintileV, mapping = aes(x = quintile_V, y = .estimate, 
                                                fill = quintile_V)) + 
  theme_bw() + 
  geom_violin(draw_quantiles = 0.5, scale = "count") + 
  scale_fill_brewer(palette = "RdBu", name = "Population Composition") +
  theme(legend.position = "top",
        axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  labs(x = "Proportion of Population with Swept Chromosome V",
       y = "Power")


power.popgroup.quintileX <- dat.population.features.divergence %>%
  dplyr::filter(CHROM == 6) %>%
  dplyr::group_by(strain_set, population.group, quintile_X) %>%
  yardstick::sens(truth = Simulated, estimate = Detected) %>%
  dplyr::mutate(.metric = "Power") %>%
  dplyr::filter(!is.na(.estimate)) %>%
  dplyr::mutate(quintile_X = gsub(as.character(quintile_X), pattern = "\\(", replacement = "") %>%
                  gsub(., pattern = "\\]", replacement = "") %>%
                  gsub(., pattern = ",", replacement = "-"))
ggplot(power.popgroup.quintileX, mapping = aes(x = quintile_X, y = .estimate, 
                                                fill = quintile_X)) + 
  theme_bw() + 
  geom_violin(draw_quantiles = 0.5) + 
  scale_fill_brewer(palette = "RdBu", name = "Population Composition") +
  theme(legend.position = "top",
        axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + 
  labs(x = "Proportion of Population with Swept Chromosome X",
       y = "Power")



dat.population.features.divergence %>%
  dplyr::group_by(pop.size, population.group) %>%
  yardstick::pr_curve(Simulated, log10p) %>%
  dplyr::filter(!is.infinite(.threshold)) %>%
  ggplot(., mapping = aes(x = recall, y = precision, colour = population.group)) + 
  theme_bw() + 
  geom_path() +
  theme(strip.text = element_text(size = 8),
        axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
        panel.grid.minor = element_blank()) +
  theme(legend.position = "top") + 
  scale_colour_manual(values = pop.group.pal, name = "Population Group") + 
  facet_wrap(pop.size~.) + 
  ylim(c(0,1)) + 
  labs(x = "Recall", y = "Precision")


dat.population.features.divergence %>%
  dplyr::filter(pop.size %in% c(48,96,144,192)) %>%
  dplyr::group_by(common.divergent, population.group, pop.size) %>%
  yardstick::pr_curve(Simulated, log10p) %>%
  dplyr::filter(!is.infinite(.threshold)) %>%
  ggplot(., mapping = aes(x = recall, y = precision, 
                          colour = population.group, 
                          alpha = common.divergent)) + 
  theme_bw() + 
  geom_path() +
  theme(strip.text = element_text(size = 8),
        axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
        panel.grid.minor = element_blank()) +
  theme(legend.position = "right") + 
  facet_grid(.~pop.size) + 
  scale_colour_manual(values = pop.group.pal, name = "Population Group") + 
  scale_alpha_manual(values = c(0.6,1), name = "Common Divergence") + 
  ylim(c(0,1)) + 
  labs(x = "Recall", y = "Precision")

dat.population.features.divergence %>%
  dplyr::filter(pop.size %in% c(48,96,144,192)) %>%
  dplyr::group_by(rare.divergent, population.group, pop.size) %>%
  yardstick::pr_curve(Simulated, log10p) %>%
  dplyr::filter(!is.infinite(.threshold)) %>%
  ggplot(., mapping = aes(x = recall, y = precision, 
                          colour = population.group, 
                          alpha = rare.divergent)) + 
  theme_bw() + 
  geom_path() +
  theme(strip.text = element_text(size = 8),
        axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
        panel.grid.minor = element_blank()) +
  theme(legend.position = "right") + 
  facet_grid(.~pop.size) + 
  scale_colour_manual(values = pop.group.pal, name = "Population Group") + 
  scale_alpha_manual(values = c(0.6,1), name = "Rare Divergence") + 
  ylim(c(0,1)) + 
  labs(x = "Recall", y = "Precision")


pr_auc_common.divergent <- dat.population.features.divergence %>%
  dplyr::filter(pop.size %in% c(48,96,144,192)) %>%
  dplyr::group_by(common.divergent, population.group, pop.size, strain_set) %>%
  yardstick::pr_auc(Simulated, log10p) %>%
  dplyr::filter(!is.na(.estimate)) %>%
  dplyr::rename(PR_AUC = .estimate) %>%
  dplyr::select(-c(.metric, .estimator)) %>%
  dplyr::group_by(common.divergent, population.group, pop.size) %>%
  dplyr::summarise(mean = mean(PR_AUC),
                   sd = sd(PR_AUC)) %>%
  dplyr::mutate(b = mean + sd,
                a = mean - sd,
                ymax = if_else(condition = b > 1, 
                             true = 1-mean, 
                             false = sd),
                ymin = if_else(condition = a < 0, 
                             true = (0-mean)*-1, 
                             false = sd)) %>%
  dplyr::select(-a,-b) %>%
  dplyr::filter(pop.size %in% c(48, 96, 144, 192))

ggplot(pr_auc_common.divergent, mapping = aes(x = population.group, 
                                            y = mean, 
                                            colour = population.group,
                                            alpha = common.divergent,
                                            group = common.divergent)) + 
  theme_bw() +
  geom_point(position = position_dodge(width=0.2)) + 
  geom_errorbar(data = pr_auc_common.divergent,
                width = 0.2,
                mapping = aes(y = mean, ymax = mean+ymax, ymin = mean-ymin, colour = population.group),
                position=position_dodge(width=0.2))  + 
  geom_line(position=position_dodge(width=0.2), colour = "black") +
  scale_colour_manual(values = pop.group.pal, name = "Population Group") + 
  scale_alpha_manual(values = c(0.4,1), name = "Common Divergence") +  
  facet_grid(.~pop.size) + 
  scale_y_continuous(breaks = seq(0,1,0.1), limits = c(0,1)) +
  theme(legend.position = "right",
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank()) + 
  labs(x = "Population Size",
       y = "Precision-Recall AUC")






pr_auc_rare.divergent <- dat.population.features.divergence %>%
  dplyr::filter(pop.size %in% c(48,96,144,192)) %>%
  dplyr::group_by(rare.divergent, population.group, pop.size, strain_set) %>%
  yardstick::pr_auc(Simulated, log10p) %>%
  dplyr::filter(!is.na(.estimate)) %>%
  dplyr::rename(PR_AUC = .estimate) %>%
  dplyr::select(-c(.metric, .estimator)) %>%
  dplyr::group_by(rare.divergent, population.group, pop.size) %>%
  dplyr::summarise(mean = mean(PR_AUC),
                   sd = sd(PR_AUC)) %>%
  dplyr::mutate(b = mean + sd,
                a = mean - sd,
                ymax = if_else(condition = b > 1, 
                             true = 1-mean, 
                             false = sd),
                ymin = if_else(condition = a < 0, 
                             true = (0-mean)*-1, 
                             false = sd)) %>%
  dplyr::select(-a,-b) %>%
  dplyr::filter(pop.size %in% c(48, 96, 144, 192))

ggplot(pr_auc_rare.divergent, mapping = aes(x = population.group, 
                                            y = mean, 
                                            colour = population.group,
                                            alpha = rare.divergent,
                                            group = rare.divergent)) + 
  theme_bw() +
  geom_point(position = position_dodge(width=0.2)) + 
  geom_errorbar(data = pr_auc_rare.divergent,
                width = 0.2,
                mapping = aes(y = mean, ymax = mean+ymax, ymin = mean-ymin, colour = population.group),
                position=position_dodge(width=0.2))  + 
  geom_line(position=position_dodge(width=0.2), colour = "black") +
  scale_colour_manual(values = pop.group.pal, name = "Population Group") + 
  scale_alpha_manual(values = c(0.4,1), name = "Rare Divergence") +  
  facet_grid(.~pop.size) + 
  scale_y_continuous(breaks = seq(0,1,0.1), limits = c(0,1)) +
  theme(legend.position = "right",
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank()) + 
  labs(x = "Population Size",
       y = "Precision-Recall AUC")



  
################
## STRAIN SET ##
################

power <- dat.population.features.divergence %>%
  dplyr::filter(!is.na(CHROM.REGION)) %>%
  dplyr::group_by(pop.size, population.group, strain_set) %>%
  yardstick::sens(Simulated, Detected) %>%
  dplyr::rename(Power = .estimate) %>%
  dplyr::select(-c(.metric,.estimator))

precision <- dat.population.features.divergence %>%
  dplyr::filter(!is.na(CHROM.REGION)) %>%
  dplyr::group_by(pop.size, population.group, strain_set) %>%
  yardstick::precision(Simulated, Detected) %>%
  dplyr::rename(Precision = .estimate)%>%
  dplyr::select(-c(.metric,.estimator))

pr_auc <- dat.population.features.divergence %>%
  dplyr::filter(!is.na(CHROM.REGION)) %>%
  dplyr::group_by(pop.size, population.group, strain_set) %>%
  yardstick::pr_auc(Simulated, log10p) %>%
  dplyr::rename(PR_AUC = .estimate)%>%
  dplyr::select(-c(.metric,.estimator))

resolution <- dat.population.features.divergence  %>%
  dplyr::filter(!is.na(CHROM.REGION)) %>%
  dplyr::filter(Detected == TRUE) %>%
  dplyr::select(CHROM, POS, Simulated, Frequency, interval_size, 
                Simulated.QTL.VarExp, var.exp, pop.size, population.group, strain_set) %>%
  dplyr::group_by(pop.size, population.group, strain_set) %>%
  dplyr::summarise(mean.interval.Mb = mean(interval_size/1000000), 
                   median.interval.Mb = median(interval_size/1000000), 
                   sd.interval.Mb = sd(interval_size/1000000))

summarized <- power %>%
  dplyr::full_join(., precision) %>%
  dplyr::full_join(., pr_auc) %>%
  dplyr::mutate(F1score = 2*((Precision*Power)/(Precision+Power)),
                FDR = 1-Precision) %>%
  dplyr::left_join(.,resolution) %>%
  dplyr::left_join(., pop.size)

summarized %>%
  dplyr::filter(pop.size %in% c(48,96,144,192)) %>%
  ggplot(., mapping = aes(x = as.factor(pop.size), 
                          y = mean.interval.Mb, 
                          fill = population.group)) + 
  theme_bw() +
  geom_violin(draw_quantiles = 0.5) +
  scale_fill_manual(values = pop.group.pal, name = "Population Group") +
  labs(x = "Sample Size",
       y = "Mean Region of Interest Width (Mb)")



#######################
## CHROMOSOME REGION ##
#######################

power <- dat.population.features.divergence %>%
  dplyr::filter(!is.na(CHROM.REGION)) %>%
  dplyr::group_by(pop.size, population.group, CHROM.REGION, strain_set, CHROM) %>%
  yardstick::sens(Simulated, Detected) %>%
  dplyr::rename(Power = .estimate) %>%
  dplyr::select(-c(.metric,.estimator))

power.chrom.region <- power %>%
  dplyr::filter(!is.na(Power)) %>%
  dplyr::group_by(pop.size, population.group, CHROM.REGION, CHROM) %>%
  dplyr::summarise(mean = mean(Power),
                   sd = sd(Power)) %>%
  dplyr::mutate(b = mean + sd,
                a = mean - sd,
                ymax = if_else(condition = b > 1, 
                             true = 1-mean, 
                             false = sd),
                ymin = if_else(condition = a < 0, 
                             true = (0-mean)*-1, 
                             false = sd)) %>%
  dplyr::select(-a,-b) %>%
  dplyr::filter(pop.size %in% c(48, 96, 144, 192))
ggplot(power.chrom.region, mapping = aes(x = CHROM.REGION, 
                                         y = mean, 
                                         colour = population.group)) + 
  theme_bw() +
  geom_point(position = position_dodge(width=0.5)) + 
  geom_errorbar(data = power.chrom.region,
                width = 0.5,
                mapping = aes(y = mean, ymax = mean+ymax, ymin = mean-ymin, colour = population.group),
                position=position_dodge(width=0.5))  + 
  # geom_line(mapping = aes(group = population.group), position = position_dodge(width = 0.5)) + 
  scale_colour_manual(values = pop.group.pal, name = "Population Group") + 
  facet_grid(CHROM~pop.size) + 
  theme(legend.position = "top") + 
  labs(x = "Population Size",
       y = "Power")






resolution <- dat.population.features.divergence  %>%
  dplyr::filter(!is.na(CHROM.REGION)) %>%
  dplyr::filter(Detected == TRUE) %>%
  dplyr::group_by(pop.size, population.group, CHROM.REGION, strain_set, CHROM) %>%
  dplyr::summarise(mean.interval.Mb = mean(interval_size/1000000)) 

resolution.chrom.region <- resolution %>% 
  dplyr::group_by(pop.size, population.group, CHROM.REGION, CHROM) %>%
  dplyr::summarise(mean = mean(mean.interval.Mb),
                   sd = sd(mean.interval.Mb)) %>%
  dplyr::mutate(b = mean + sd,
                a = mean - sd,
                ymin = if_else(condition = a < 0, 
                             true = (0-mean)*-1, 
                             false = sd)) %>%
  dplyr::select(-a,-b) %>%
  dplyr::filter(pop.size %in% c(48, 96, 144, 192))

ggplot(resolution.chrom.region) + 
  theme_bw() +
  geom_point(mapping=aes(x=CHROM.REGION, 
                         y=mean, 
                         colour = population.group), position = position_dodge(width = 0.5)) + 
  geom_pointrange(data=resolution.chrom.region, 
                  mapping=aes(x=CHROM.REGION, 
                              y=mean, 
                              ymin=mean-ymin, ymax=mean+sd, colour = population.group), 
                  position = position_dodge(width = 0.5))  +
  scale_colour_manual(values = pop.group.pal, name = "Population Group") + 
  facet_grid(CHROM~pop.size) + 
  theme(legend.position = "top") + 
  labs(x = "Population Size",
       y = "Mean ROI Size (Mb)")

```
